<!DOCTYPE html>
<html lang="en"><head>
<script src="Bioc2023_presentation_files/libs/clipboard/clipboard.min.js"></script>
<script src="Bioc2023_presentation_files/libs/quarto-html/tabby.min.js"></script>
<script src="Bioc2023_presentation_files/libs/quarto-html/popper.min.js"></script>
<script src="Bioc2023_presentation_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="Bioc2023_presentation_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Bioc2023_presentation_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="Bioc2023_presentation_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="Bioc2023_presentation_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.353">

  <meta name="author" content="Stevie Pederson (They/Them)">
  <meta name="dcterms.date" content="2023-08-02">
  <title>extraChIPs</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="Bioc2023_presentation_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="Bioc2023_presentation_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="Bioc2023_presentation_files/libs/revealjs/dist/theme/quarto.css">
  <link href="Bioc2023_presentation_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="Bioc2023_presentation_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="Bioc2023_presentation_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="Bioc2023_presentation_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" data-background-color="#3c3c44" data-background-image="https://raw.githubusercontent.com/Bioconductor/BiocStickers/devel/extraChIPs/extraChIPs.png" data-background-opacity="20%" data-background-position="50% 50%" data-background-size="35%" class="quarto-title-block center">
  <h1 class="title">extraChIPs</h1>
  <p class="subtitle">Bioc 2023 Package Demonstration</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Stevie Pederson (They/Them) 
</div>
</div>
</div>

  <p class="date">August 2, 2023</p>
</section>
<section id="section" class="slide level2">
<h2></h2>
<p>https://smped.github.io/Bioc2023_extraChIPs/</p>
</section>
<section>
<section id="introduction" class="title-slide slide level1 center" data-background-color="#3c3c44">
<h1>Introduction</h1>

</section>
<section id="who-am-i" class="slide level2">
<h2>Who Am I?</h2>
<ul>
<li>Post-doctoral Bioinformatician at Black Ochre Data Labs
<ul>
<li>Telethon Kids Institute, Adelaide, Australia</li>
<li>Traditional lands of the Kaurna nation</li>
</ul></li>
<li>Improving health outcomes for Indigenous Australians
<ul>
<li>Type 2 Diabetes, Cardiovascular Disease, Chronic Kidney Disease</li>
</ul></li>
<li>1400 participants in multi-omics study
<ul>
<li>Genomics, Epigenomics, Transcriptomics, Proteomics, Lipidomics etc</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Also the developer/maintainer of <code>ngsReports</code> <br> <img data-src="https://raw.githubusercontent.com/Bioconductor/BiocStickers/devel/ngsReports/ngsReports.png" style="width:20.0%"></li>
</ul>
</div>
</section>
<section id="why-extrachips" class="slide level2">
<h2>Why <code>extraChIPs</code>?</h2>
<!-- - Bioinformatics Hub at University of Adelaide (2014-2020) -->
<ul>
<li>Dame Roma Mitchell Cancer Research Laboratories (UofA, 2020-2022)
<ul>
<li>Integrating multiple ChIP targets across multiple cell lines</li>
<li>Regulatory <em>dynamics</em> of AR activation with DHT (or SARMs)</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Built to enable the GRAVI workflow:
<ul>
<li>Gene Regulatory Analysis using Variable Inputs (Poster Thursday)</li>
<li>Integration of multiple ChIP targets with RNA-Seq + HiC</li>
<li>Wanted to standardise ‘best practice’ across multiple datasets</li>
<li>Design evolved alongside GRAVI</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Filled gaps between <code>DiffBind</code> <span class="citation" data-cites="DiffBind2012">(<a href="#/references" role="doc-biblioref" onclick="">Ross-Innes et al. 2012</a>)</span> and <code>csaw</code> <span class="citation" data-cites="csaw2016">(<a href="#/references" role="doc-biblioref" onclick="">Lun and Smyth 2016</a>)</span></li>
</ul>
<aside class="notes">
<ul>
<li>Regulatory dynamics we were interested in wree primarily about which genes were being impacted by changes in AR/ER binding. Was AR omving ER away from proliferative genes</li>
<li>Not intended to be a strict “tidy” implementation of ChIP-Seq</li>
<li>GRAVI itself is a snakemake workflow</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="why-extrachips-1" class="slide level2">
<h2>Why <code>extraChIPs</code>?</h2>
<div style="font-size: 90%;">
<ul>
<li>Simplify common processes:
<ul>
<li>Flexible detection of Differential Signal (Fixed-Width or Sliding Windows)</li>
<li>Maintaining <code>mcols</code> throughout an analysis</li>
<li>Easier comparison <em>across multiple targets or treatments</em> <!-- + Mapping to genes & pre-defined genomic regions/features --></li>
<li>Enable common visualisations</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Integration with the <code>tidyverse</code> <span class="citation" data-cites="tidyverse2019">(<a href="#/references" role="doc-biblioref" onclick="">Wickham et al. 2019</a>)</span> + <code>plyranges</code> <span class="citation" data-cites="plyranges">(<a href="#/references" role="doc-biblioref" onclick="">Lee et al. 2019</a>)</span></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Use existing object classes:
<ul>
<li><code>SummarizedExperiment</code>, <code>GRanges</code>, <code>GRangesList</code>, <code>List</code> <!-- + `BigWigFileList`, `BamFileList` etc --></li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Applicable well beyond ChIP-Seq (DNA-methylation, ATAC-Seq etc)</li>
</ul>
</div>
</div>
</section>
<section id="key-functions" class="slide level2">
<h2>Key Functions</h2>
<div style="font-size: 90%;">
<div class="columns">
<div class="column">
<h4 id="core-infrastructure">Core Infrastructure</h4>
<ul>
<li><code>reduceMC</code>, <code>intersectMC</code>, <code>setdiffMC</code>, <code>unionMC</code></li>
<li><code>chopMC</code>, <code>distinctMC</code></li>
<li><code>as_tibble</code>, <code>colToRanges</code></li>
</ul>
<p><br></p>
<h4 id="working-with-peaksgranges">Working With Peaks/GRanges</h4>
<ul>
<li><code>importPeaks</code>, <code>makeConsensus</code>, <code>plotOverlaps</code></li>
<li><code>bestOverlap</code>, <code>propOverlap</code></li>
<li><code>defineRegions</code>, <code>mapByFeature</code></li>
<li><code>mapGrlCols</code> <sup>1</sup></li>
</ul>
</div><div class="column">
<h4 id="differential-signal-analysis">Differential Signal Analysis</h4>
<ul>
<li><code>fitAssayDiff</code>, <code>addDiffStatus</code></li>
<li><code>dualFilter</code>,</li>
<li><code>mergeByHMP</code>, <code>mergeByCol</code>, <code>mergeBySig</code></li>
</ul>
<p><br></p>
<h4 id="visualisations">Visualisations</h4>
<ul>
<li><code>plotAssayDensities</code>, <code>plotAssayPCA</code>, <code>plotAssayRle</code></li>
<li><code>plotPie</code>, <code>plotSplitDonut</code></li>
<li><code>plotPairwise</code> <sup>1</sup></li>
<li><code>plotProfileHeatmap</code>, <code>plotHFGC</code></li>
</ul>
</div><aside class="notes">
<p>Functions highlighted with an asterisk are only in <code>devel</code></p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</div>
<aside><ol class="aside-footnotes"><li id="fn1"><p>Only available in <code>devel</code></p></li></ol></aside></section></section>
<section>
<section id="core-infrastructure-1" class="title-slide slide level1 center" data-background-color="#3c3c44" data-background-image="https://raw.githubusercontent.com/Bioconductor/BiocStickers/devel/extraChIPs/extraChIPs.png" data-background-size="35%" data-background-opacity="0.2">
<h1>Core Infrastructure</h1>

</section>
<section id="mc-functions" class="slide level2">
<h2>*MC Functions</h2>
<ul>
<li>Existing functions <code>reduce</code>, <code>setdiff</code>, <code>intersect</code>, <code>union</code>
<ul>
<li>All will drop information in the <code>mcols</code> element</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li><code>reduceMC</code>, <code>setdiffMC</code>, <code>intersectMC</code> and <code>unionMC</code>
<ul>
<li>Retain all information in <code>mcols</code> element</li>
<li>Returned as <code>CompressedList</code> or atomic vectors</li>
<li>Heavily used internally</li>
</ul></li>
</ul>
<!-- . . .     -->
<!-- - Addition of `chopMC` and `distinctMC` -->
<!--     + Based on `tidyverse` functions  -->
<!--     + Helpful after multiple sequential joins -->
</div>
</section>
<section id="reduce-vs.-reducemc" class="slide level2">
<h2><code>reduce</code> Vs. <code>reduceMC</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>x</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 2 ranges and 2 metadata columns:
      seqnames    ranges strand |          id        gene
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
  [1]     chr1      1-10      * |      range1       gene1
  [2]     chr1      6-12      * |      range2       gene1
  -------
  seqinfo: 1 sequence from example genome</code></pre>
</div>
</div>
<div class="fragment">
<p><code>reduce</code> will simply combine (i.e.&nbsp;reduce) overlapping ranges</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">reduce</span>(x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 1 range and 0 metadata columns:
      seqnames    ranges strand
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
  [1]     chr1      1-12      *
  -------
  seqinfo: 1 sequence from example genome</code></pre>
</div>
</div>
<aside class="notes">
<ul>
<li>Here we have two partially overlapping ranges</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="reduce-vs.-reducemc-1" class="slide level2">
<h2><code>reduce</code> Vs. <code>reduceMC</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>x</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 2 ranges and 2 metadata columns:
      seqnames    ranges strand |          id        gene
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
  [1]     chr1      1-10      * |      range1       gene1
  [2]     chr1      6-12      * |      range2       gene1
  -------
  seqinfo: 1 sequence from example genome</code></pre>
</div>
</div>
<p><code>reduceMC</code> will also combine values in the <code>mcols</code> element</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">reduceMC</span>(x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 1 range and 2 metadata columns:
      seqnames    ranges strand |              id        gene
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;CharacterList&gt; &lt;character&gt;
  [1]     chr1      1-12      * |   range1,range2       gene1
  -------
  seqinfo: 1 sequence from example genome</code></pre>
</div>
</div>
<div class="fragment">
<p><code>mcols</code> are simplified by default (<code>simplify = TRUE</code>)</p>
</div>
</section>
<section id="tibble-coercion" class="slide level2">
<h2><code>tibble</code> Coercion</h2>
<ul>
<li><code>GRanges</code> can be coerced to <code>tibble</code> objects</li>
<li>Ranges are coerced to a <code>character</code> column</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>tbl <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(x)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>tbl</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 3
  range     id     gene 
  &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;
1 chr1:1-10 range1 gene1
2 chr1:6-12 range2 gene1</code></pre>
</div>
</div>
<div class="fragment">
<ul>
<li><code>rangeAsChar = FALSE</code> will produce similar to <code>as.data.frame</code></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(x, <span class="at">rangeAsChar =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 7
  seqnames start   end width strand id     gene 
  &lt;fct&gt;    &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;fct&gt;  &lt;chr&gt;  &lt;chr&gt;
1 chr1         1    10    10 *      range1 gene1
2 chr1         6    12     7 *      range2 gene1</code></pre>
</div>
</div>
</div>
</section>
<section id="coercing-tibble-objects-back-to-granges" class="slide level2">
<h2>Coercing <code>tibble</code> objects back to <code>GRanges</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>tbl</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 3
  range     id     gene 
  &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;
1 chr1:1-10 range1 gene1
2 chr1:6-12 range2 gene1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colToRanges</span>(tbl, <span class="at">var =</span> <span class="st">"range"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 2 ranges and 2 metadata columns:
      seqnames    ranges strand |          id        gene
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
  [1]     chr1      1-10      * |      range1       gene1
  [2]     chr1      6-12      * |      range2       gene1
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<div class="fragment">
<p>Seqinfo objects can also be provided</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colToRanges</span>(tbl, <span class="at">var =</span> <span class="st">"range"</span>, <span class="at">seqinfo =</span> <span class="fu">seqinfo</span>(x))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 2 ranges and 2 metadata columns:
      seqnames    ranges strand |          id        gene
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
  [1]     chr1      1-10      * |      range1       gene1
  [2]     chr1      6-12      * |      range2       gene1
  -------
  seqinfo: 1 sequence from example genome</code></pre>
</div>
</div>
</div>
</section>
<section id="coltoranges" class="slide level2">
<h2><code>colToRanges</code></h2>
<ul>
<li><code>colToRanges</code> can also move <code>GRanges</code> objects in <code>mcols</code> to the ‘backbone’ ranges</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb19" data-code-line-numbers="1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>x<span class="sc">$</span>centre <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="fu">paste0</span>(<span class="st">"chr1:"</span>, <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">10</span>)), <span class="at">seqinfo =</span> <span class="fu">seqinfo</span>(x))</span>
<span id="cb19-2"><a href="#cb19-2"></a>x</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 2 ranges and 3 metadata columns:
      seqnames    ranges strand |          id        gene    centre
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;GRanges&gt;
  [1]     chr1      1-10      * |      range1       gene1    chr1:5
  [2]     chr1      6-12      * |      range2       gene1   chr1:10
  -------
  seqinfo: 1 sequence from example genome</code></pre>
</div>
</div>
<div class="fragment">
<p><br></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colToRanges</span>(x, <span class="at">var =</span> <span class="st">"centre"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 2 ranges and 2 metadata columns:
      seqnames    ranges strand |          id        gene
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
  [1]     chr1         5      * |      range1       gene1
  [2]     chr1        10      * |      range2       gene1
  -------
  seqinfo: 1 sequence from example genome</code></pre>
</div>
</div>
<aside class="notes">
<ul>
<li>Here I’m first adding the peak centre as a new mcols column, then shifting this to be the core range</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section></section>
<section>
<section id="working-with-peaks" class="title-slide slide level1 center" data-background-color="#3c3c44" data-background-image="https://raw.githubusercontent.com/Bioconductor/BiocStickers/devel/extraChIPs/extraChIPs.png" data-background-size="35%" data-background-opacity="0.2">
<h1>Working With Peaks</h1>

</section>
<section id="demonstration-data" class="slide level2">
<h2>Demonstration Data <sup>1</sup></h2>
<div class="cell columns column-output-location">
<div class="column">
<div class="sourceCode cell-code" id="cb23" data-code-line-numbers="1-3|4-15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="do">## Define Samples</span></span>
<span id="cb23-2"><a href="#cb23-2"></a>targets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ER"</span>, <span class="st">"H3K27ac"</span>)</span>
<span id="cb23-3"><a href="#cb23-3"></a>treat_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E2DHT"</span>)</span>
<span id="cb23-4"><a href="#cb23-4"></a>samples <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb23-5"><a href="#cb23-5"></a>  <span class="at">accession =</span> <span class="fu">paste0</span>(<span class="st">"SRR83151"</span>, <span class="dv">80</span><span class="sc">:</span><span class="dv">91</span>),</span>
<span id="cb23-6"><a href="#cb23-6"></a>  <span class="at">target =</span> <span class="fu">rep</span>(targets, <span class="at">each =</span> <span class="dv">6</span>),</span>
<span id="cb23-7"><a href="#cb23-7"></a>  <span class="at">treatment =</span> treat_levels <span class="sc">%&gt;%</span> </span>
<span id="cb23-8"><a href="#cb23-8"></a>    <span class="fu">rep</span>(<span class="at">each =</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-9"><a href="#cb23-9"></a>    <span class="fu">rep_len</span>(<span class="dv">12</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-10"><a href="#cb23-10"></a>    <span class="fu">factor</span>(<span class="at">levels =</span> treat_levels),</span>
<span id="cb23-11"><a href="#cb23-11"></a>  <span class="at">input =</span> <span class="st">"SRR8315192"</span></span>
<span id="cb23-12"><a href="#cb23-12"></a>)</span>
<span id="cb23-13"><a href="#cb23-13"></a><span class="do">## Split for easier wrangling</span></span>
<span id="cb23-14"><a href="#cb23-14"></a>samples <span class="ot">&lt;-</span> <span class="fu">split</span>(samples, samples<span class="sc">$</span>target)</span>
<span id="cb23-15"><a href="#cb23-15"></a>samples</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div><div class="column">
<div class="cell-output cell-output-stdout">
<pre><code>$ER
# A tibble: 6 × 4
  accession  target treatment input     
  &lt;chr&gt;      &lt;chr&gt;  &lt;fct&gt;     &lt;chr&gt;     
1 SRR8315180 ER     E2        SRR8315192
2 SRR8315181 ER     E2        SRR8315192
3 SRR8315182 ER     E2        SRR8315192
4 SRR8315183 ER     E2DHT     SRR8315192
5 SRR8315184 ER     E2DHT     SRR8315192
6 SRR8315185 ER     E2DHT     SRR8315192

$H3K27ac
# A tibble: 6 × 4
  accession  target  treatment input     
  &lt;chr&gt;      &lt;chr&gt;   &lt;fct&gt;     &lt;chr&gt;     
1 SRR8315186 H3K27ac E2        SRR8315192
2 SRR8315187 H3K27ac E2        SRR8315192
3 SRR8315188 H3K27ac E2        SRR8315192
4 SRR8315189 H3K27ac E2DHT     SRR8315192
5 SRR8315190 H3K27ac E2DHT     SRR8315192
6 SRR8315191 H3K27ac E2DHT     SRR8315192</code></pre>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>2x ChIP targets (ER + H3K27ac)</li>
<li>2x treatments (E2, E2DHT)</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<aside><ol class="aside-footnotes"><li id="fn2"><p>Taken from SRA BioProject PRJNA509779 <span class="citation" data-cites="Hickey2021-mz">(<a href="#/references" role="doc-biblioref" onclick="">Hickey et al. 2021</a>)</span> and subset to a region of chr10</p></li></ol></aside></section>
<section id="data-directory-contents" class="slide level2">
<h2>Data Directory Contents</h2>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>-- annotations
   |__chr10_blacklist.bed
   |__chr10_greylist.bed
   |__gencode.v43lift37.chr10.annotation.gtf.gz
-- ER
   |__E2_cov_chr10.bw
   |__E2DHT_cov_chr10.bw
   |__SRR8315180_peaks.narrowPeak
   |__SRR8315180.bam
   |__SRR8315180.bam.bai
   |__SRR8315181_peaks.narrowPeak
   |__SRR8315181.bam
   |__SRR8315181.bam.bai
   |__SRR8315182_peaks.narrowPeak
   |__SRR8315182.bam
   |__SRR8315182.bam.bai
   |__SRR8315183_peaks.narrowPeak
   |__SRR8315183.bam
   |__SRR8315183.bam.bai
   |__SRR8315184_peaks.narrowPeak
   |__SRR8315184.bam
   |__SRR8315184.bam.bai
   |__SRR8315185_peaks.narrowPeak
   |__SRR8315185.bam
   |__SRR8315185.bam.bai
-- H3K27ac
   |__E2_cov_chr10.bw
   |__E2DHT_cov_chr10.bw
   |__H3K27ac_chr10.bed
   |__SRR8315186.bam
   |__SRR8315186.bam.bai
   |__SRR8315187.bam
   |__SRR8315187.bam.bai
   |__SRR8315188.bam
   |__SRR8315188.bam.bai
   |__SRR8315189.bam
   |__SRR8315189.bam.bai
   |__SRR8315190.bam
   |__SRR8315190.bam.bai
   |__SRR8315191.bam
   |__SRR8315191.bam.bai
   |__SRR8315192.bam
   |__SRR8315192.bam.bai</code></pre>
</div>
</div>
</section>
<section id="additional-objects" class="slide level2">
<h2>Additional Objects</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb26" data-code-line-numbers="1-4|6-13|14-20|22-27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="do">## A consistent Seqinfo object</span></span>
<span id="cb26-2"><a href="#cb26-2"></a>sq <span class="ot">&lt;-</span> <span class="fu">Seqinfo</span>(</span>
<span id="cb26-3"><a href="#cb26-3"></a>  <span class="at">seqnames =</span> <span class="st">"chr10"</span>, <span class="at">seqlengths =</span> <span class="dv">135534747</span>, <span class="at">isCircular =</span> <span class="cn">FALSE</span>,  <span class="at">genome =</span> <span class="st">"hg19"</span></span>
<span id="cb26-4"><a href="#cb26-4"></a>)</span>
<span id="cb26-5"><a href="#cb26-5"></a></span>
<span id="cb26-6"><a href="#cb26-6"></a><span class="do">## Gene Annotations as a GTF, split by feature type</span></span>
<span id="cb26-7"><a href="#cb26-7"></a>gtf <span class="ot">&lt;-</span> <span class="fu">file.path</span>(</span>
<span id="cb26-8"><a href="#cb26-8"></a>  <span class="st">"data"</span>, <span class="st">"annotations"</span>, <span class="st">"gencode.v43lift37.chr10.annotation.gtf.gz"</span></span>
<span id="cb26-9"><a href="#cb26-9"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-10"><a href="#cb26-10"></a>  <span class="fu">import.gff</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-11"><a href="#cb26-11"></a>  <span class="fu">split</span>(.<span class="sc">$</span>type)</span>
<span id="cb26-12"><a href="#cb26-12"></a><span class="fu">seqinfo</span>(gtf) <span class="ot">&lt;-</span> sq</span>
<span id="cb26-13"><a href="#cb26-13"></a></span>
<span id="cb26-14"><a href="#cb26-14"></a><span class="do">## Black &amp; grey-lists (From ENCODE &amp; Input Sample)</span></span>
<span id="cb26-15"><a href="#cb26-15"></a>bg_list <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="st">"annotations"</span>, <span class="fu">c</span>(<span class="st">"chr10_blacklist.bed"</span>, <span class="st">"chr10_greylist.bed"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-16"><a href="#cb26-16"></a>  <span class="fu">lapply</span>(import.bed, <span class="at">seqinfo =</span> sq) <span class="sc">%&gt;%</span> </span>
<span id="cb26-17"><a href="#cb26-17"></a>  <span class="fu">lapply</span>(granges) <span class="sc">%&gt;%</span> </span>
<span id="cb26-18"><a href="#cb26-18"></a>  <span class="fu">GRangesList</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-19"><a href="#cb26-19"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-20"><a href="#cb26-20"></a>  <span class="fu">sort</span>()</span>
<span id="cb26-21"><a href="#cb26-21"></a></span>
<span id="cb26-22"><a href="#cb26-22"></a><span class="do">## Setup colours for consistent plotting</span></span>
<span id="cb26-23"><a href="#cb26-23"></a>treat_colours <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">E2 =</span> <span class="st">"steelblue"</span>, <span class="at">E2DHT =</span> <span class="st">"red3"</span>)</span>
<span id="cb26-24"><a href="#cb26-24"></a>sample_colours <span class="ot">&lt;-</span> <span class="fu">setNames</span>(</span>
<span id="cb26-25"><a href="#cb26-25"></a>  treat_colours[<span class="fu">bind_rows</span>(samples)<span class="sc">$</span>treatment], </span>
<span id="cb26-26"><a href="#cb26-26"></a>  <span class="fu">bind_rows</span>(samples)<span class="sc">$</span>accession</span>
<span id="cb26-27"><a href="#cb26-27"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="working-with-peaks-1" class="slide level2">
<h2>Working With Peaks</h2>
<p>Key Steps:</p>
<ol type="1">
<li>Load peaks identified in individual samples (e.g.&nbsp;<code>macs2 callpeak</code>)</li>
</ol>
<div class="fragment">
<ol start="2" type="1">
<li>Find overlapping regions from <span class="math inline">\(&gt; p\times n\)</span> replicates <span class="math inline">\(\implies\)</span> <em>consensus peaks</em>
<ul>
<li>Usually formed within treatment groups</li>
<li>Merged across treatment groups</li>
</ul></li>
</ol>
</div>
<div class="fragment">
<p>Peak callers often produce narrowPeak/broadPeak files</p>
</div>
</section>
<section id="importing-peaks" class="slide level2">
<h2>Importing Peaks</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb27" data-code-line-numbers="1-5|10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a><span class="fu">importPeaks</span>(</span>
<span id="cb27-2"><a href="#cb27-2"></a>  x,</span>
<span id="cb27-3"><a href="#cb27-3"></a>  <span class="at">type =</span> <span class="fu">c</span>(<span class="st">"narrow"</span>, <span class="st">"broad"</span>),</span>
<span id="cb27-4"><a href="#cb27-4"></a>  blacklist,</span>
<span id="cb27-5"><a href="#cb27-5"></a>  seqinfo,</span>
<span id="cb27-6"><a href="#cb27-6"></a>  <span class="at">pruning.mode =</span> <span class="fu">c</span>(<span class="st">"coarse"</span>, <span class="st">"error"</span>),</span>
<span id="cb27-7"><a href="#cb27-7"></a>  <span class="at">sort =</span> <span class="cn">TRUE</span>,</span>
<span id="cb27-8"><a href="#cb27-8"></a>  <span class="at">setNames =</span> <span class="cn">TRUE</span>,</span>
<span id="cb27-9"><a href="#cb27-9"></a>  <span class="at">glueNames =</span> <span class="st">"{basename(x)}"</span>,</span>
<span id="cb27-10"><a href="#cb27-10"></a>  <span class="at">centre =</span> <span class="cn">FALSE</span>,</span>
<span id="cb27-11"><a href="#cb27-11"></a>  <span class="at">nameRanges =</span> <span class="cn">TRUE</span>,</span>
<span id="cb27-12"><a href="#cb27-12"></a>  ...</span>
<span id="cb27-13"><a href="#cb27-13"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Import broad/narrowPeak files <span class="math inline">\(\implies\)</span> <code>GRangesList</code></li>
<li>Blacklisted regions can be omitted during parsing</li>
<li><strong>Only chromosomes/scaffolds within the <code>seqinfo</code> object will be returned</strong></li>
<li>Peak Centres can be automatically added for <code>narrowPeak</code> files</li>
</ul>
</section>
<section id="importing-peaks-1" class="slide level2">
<h2>Importing Peaks</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb28" data-code-line-numbers="2|1-4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>fl <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="st">"ER"</span>, <span class="fu">paste0</span>(samples<span class="sc">$</span>ER<span class="sc">$</span>accession, <span class="st">"_peaks.narrowPeak"</span>))</span>
<span id="cb28-2"><a href="#cb28-2"></a>peaks <span class="ot">&lt;-</span> <span class="fu">importPeaks</span>(fl, <span class="at">blacklist =</span> bg_list, <span class="at">seqinfo =</span> sq, <span class="at">centre =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-3"><a href="#cb28-3"></a><span class="fu">names</span>(peaks) <span class="ot">&lt;-</span> samples<span class="sc">$</span>ER<span class="sc">$</span>accession</span>
<span id="cb28-4"><a href="#cb28-4"></a>peaks</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell fragment">
<div class="cell-output cell-output-stdout">
<pre><code>GRangesList object of length 6:
$SRR8315180
GRanges object with 123 ranges and 6 metadata columns:
                      seqnames            ranges strand |     score signalValue
                         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;   &lt;numeric&gt;
  SRR8315180_peak_698    chr10 43048224-43048487      * |       251    13.91781
  SRR8315180_peak_699    chr10 43521883-43522126      * |       223    12.88686
  SRR8315180_peak_700    chr10 43540200-43540384      * |        58     6.18569
  SRR8315180_peak_701    chr10 43606308-43606517      * |        92     7.73212
  SRR8315180_peak_702    chr10 44092187-44092576      * |       251    13.91781
                  ...      ...               ...    ... .       ...         ...
  SRR8315180_peak_819    chr10 95796039-95796377      * |       434    18.67842
  SRR8315180_peak_820    chr10 96089595-96089749      * |        25     4.32096
  SRR8315180_peak_821    chr10 99097150-99097384      * |       196     9.02984
  SRR8315180_peak_822    chr10 99168353-99168578      * |        48     5.67022
  SRR8315180_peak_823    chr10 99331430-99331686      * |       210    10.70668
                         pValue    qValue      peak    centre
                      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
  SRR8315180_peak_698  29.15643  25.19601       117  43048341
  SRR8315180_peak_699  26.25515  22.33688       125  43522008
  SRR8315180_peak_700   9.37883   5.87296        98  43540298
  SRR8315180_peak_701  12.90188   9.27326       110  43606418
  SRR8315180_peak_702  29.15643  25.19601       214  44092401
                  ...       ...       ...       ...       ...
  SRR8315180_peak_819  47.62759  43.44466       186  95796225
  SRR8315180_peak_820   5.88899   2.58967         6  96089601
  SRR8315180_peak_821  23.51470  19.64532       109  99097259
  SRR8315180_peak_822   8.26998   4.82385       153  99168506
  SRR8315180_peak_823  24.91666  21.02332       191  99331621
  -------
  seqinfo: 1 sequence from hg19 genome

...
&lt;5 more elements&gt;</code></pre>
</div>
</div>
</section>
<section id="comparing-replicates" class="slide level2">
<h2>Comparing Replicates</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotOverlaps</span>(peaks, <span class="at">set_col =</span> sample_colours)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Bioc2023_presentation_files/figure-revealjs/plot-overlaps-1.png" width="960" class="r-stretch"><ul>
<li>Calls <code>ComplexUpset::upset()</code> if n <span class="math inline">\(\geq\)</span> 3</li>
</ul>
<aside class="notes">
<ul>
<li>If a <code>GRangesList</code> is input, consensus peaks are formed for the plot</li>
<li><code>...</code> is passed to <code>upset()</code></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>

<section id="makeconsensus" class="slide level2">
<h2>makeConsensus</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">makeConsensus</span>(peaks[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">var =</span> <span class="st">"centre"</span>, <span class="at">p =</span> <span class="dv">2</span><span class="sc">/</span><span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 164 ranges and 5 metadata columns:
        seqnames            ranges strand |                     centre
           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |              &lt;NumericList&gt;
    [1]    chr10 43048195-43048529      * | 43048341,43048357,43048389
    [2]    chr10 43521739-43522260      * | 43522008,43522012,43522039
    [3]    chr10 43540042-43540390      * |          43540298,43540245
    [4]    chr10 43606238-43606573      * | 43606418,43606408,43606421
    [5]    chr10 43851214-43851989      * |          43851672,43851766
    ...      ...               ...    ... .                        ...
  [160]    chr10 99096784-99097428      * | 99097259,99097253,99097249
  [161]    chr10 99168353-99168649      * | 99168506,99168486,99168513
  [162]    chr10 99207868-99208156      * |          99207995,99208002
  [163]    chr10 99331363-99331730      * | 99331621,99331585,99331580
  [164]    chr10 99621632-99621961      * |          99621809,99621828
        SRR8315180 SRR8315181 SRR8315182         n
         &lt;logical&gt;  &lt;logical&gt;  &lt;logical&gt; &lt;numeric&gt;
    [1]       TRUE       TRUE       TRUE         3
    [2]       TRUE       TRUE       TRUE         3
    [3]       TRUE      FALSE       TRUE         2
    [4]       TRUE       TRUE       TRUE         3
    [5]      FALSE       TRUE       TRUE         2
    ...        ...        ...        ...       ...
  [160]       TRUE       TRUE       TRUE         3
  [161]       TRUE       TRUE       TRUE         3
  [162]      FALSE       TRUE       TRUE         2
  [163]       TRUE       TRUE       TRUE         3
  [164]      FALSE       TRUE       TRUE         2
  -------
  seqinfo: 1 sequence from hg19 genome</code></pre>
</div>
</div>
<div class="fragment">
<p><br></p>
<ul>
<li>All/any/no mcols can be returned</li>
</ul>
</div>
</section>
<section id="makeconsensus-1" class="slide level2">
<h2>makeConsensus</h2>

<img data-src="Bioc2023_presentation_files/figure-revealjs/olap-ranges-1.png" width="960" class="r-stretch"><ul>
<li>By default, the <em>union</em> of overlapping ranges will be returned
<ul>
<li>between black lines</li>
</ul></li>
</ul>
<p><br></p>
</section>
<section id="makeconsensus-2" class="slide level2">
<h2>makeConsensus</h2>

<img data-src="Bioc2023_presentation_files/figure-revealjs/olap-ranges-red-1.png" width="960" class="r-stretch"><ul>
<li>Can also return only ranges <em>covered</em> by <span class="math inline">\(&gt;p\times n\)</span> samples
<ul>
<li>between red lines</li>
</ul></li>
<li><code>method = "coverage"</code></li>
</ul>
</section>
<section id="consensus-peaks" class="slide level2">
<h2>Consensus Peaks</h2>
<p>To find peaks identified in 2/3 samples in either condition:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34" data-code-line-numbers="2-3|5|1-7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>treatment_peaks <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb34-2"><a href="#cb34-2"></a>    <span class="at">E2 =</span> <span class="fu">makeConsensus</span>(peaks[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">var =</span> <span class="st">"centre"</span>, <span class="at">p =</span> <span class="dv">2</span><span class="sc">/</span><span class="dv">3</span>),</span>
<span id="cb34-3"><a href="#cb34-3"></a>    <span class="at">E2DHT =</span> <span class="fu">makeConsensus</span>(peaks[<span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>], <span class="at">var =</span> <span class="st">"centre"</span>, <span class="at">p =</span> <span class="dv">2</span><span class="sc">/</span><span class="dv">3</span>)</span>
<span id="cb34-4"><a href="#cb34-4"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-5"><a href="#cb34-5"></a>    <span class="fu">lapply</span>(select, centre) <span class="sc">%&gt;%</span> </span>
<span id="cb34-6"><a href="#cb34-6"></a>    <span class="fu">GRangesList</span>()</span>
<span id="cb34-7"><a href="#cb34-7"></a>treatment_peaks</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell fragment">
<div class="cell-output cell-output-stdout">
<pre><code>GRangesList object of length 2:
$E2
GRanges object with 164 ranges and 1 metadata column:
        seqnames            ranges strand |                     centre
           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |              &lt;NumericList&gt;
    [1]    chr10 43048195-43048529      * | 43048341,43048357,43048389
    [2]    chr10 43521739-43522260      * | 43522008,43522012,43522039
    [3]    chr10 43540042-43540390      * |          43540298,43540245
    [4]    chr10 43606238-43606573      * | 43606418,43606408,43606421
    [5]    chr10 43851214-43851989      * |          43851672,43851766
    ...      ...               ...    ... .                        ...
  [160]    chr10 99096784-99097428      * | 99097259,99097253,99097249
  [161]    chr10 99168353-99168649      * | 99168506,99168486,99168513
  [162]    chr10 99207868-99208156      * |          99207995,99208002
  [163]    chr10 99331363-99331730      * | 99331621,99331585,99331580
  [164]    chr10 99621632-99621961      * |          99621809,99621828
  -------
  seqinfo: 1 sequence from hg19 genome

$E2DHT
GRanges object with 160 ranges and 1 metadata column:
        seqnames            ranges strand |                     centre
           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |              &lt;NumericList&gt;
    [1]    chr10 43048235-43048520      * | 43048383,43048373,43048388
    [2]    chr10 43521795-43522191      * | 43521992,43522008,43522026
    [3]    chr10 43540202-43540457      * | 43540318,43540364,43540361
    [4]    chr10 43606184-43606572      * |          43606426,43606411
    [5]    chr10 44092131-44092563      * | 44092370,44092321,44092385
    ...      ...               ...    ... .                        ...
  [156]    chr10 99168349-99168690      * |          99168488,99168465
  [157]    chr10 99207862-99208157      * | 99207990,99207997,99208044
  [158]    chr10 99331323-99331741      * | 99331567,99331605,99331594
  [159]    chr10 99340107-99340354      * |          99340208,99340289
  [160]    chr10 99621638-99621988      * |          99621822,99621733
  -------
  seqinfo: 1 sequence from hg19 genome</code></pre>
</div>
</div>
</section>
<section id="consensus-peaks-1" class="slide level2">
<h2>Consensus Peaks</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotOverlaps</span>(treatment_peaks, <span class="at">set_col =</span> treat_colours)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Bioc2023_presentation_files/figure-revealjs/plot-overlap-treat-1.png" width="576" class="r-stretch"><ul>
<li>Produces a VennDiagram when n = 2</li>
</ul>
</section>
<section id="consensus-peaks-2" class="slide level2">
<h2>Consensus Peaks</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb37" data-code-line-numbers="1|2-6|1-7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>union_peaks <span class="ot">&lt;-</span> <span class="fu">makeConsensus</span>(treatment_peaks, <span class="at">var =</span> <span class="st">"centre"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb37-2"><a href="#cb37-2"></a>  <span class="fu">mutate</span>(</span>
<span id="cb37-3"><a href="#cb37-3"></a>    <span class="at">centre =</span>  <span class="fu">round</span>(<span class="fu">vapply</span>(centre, mean, <span class="fu">numeric</span>(<span class="dv">1</span>)), <span class="dv">0</span>),</span>
<span id="cb37-4"><a href="#cb37-4"></a>    <span class="at">centre =</span> <span class="fu">paste0</span>(seqnames, <span class="st">":"</span>, centre) <span class="sc">%&gt;%</span> <span class="fu">GRanges</span>(<span class="at">seqinfo =</span> sq)</span>
<span id="cb37-5"><a href="#cb37-5"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb37-6"><a href="#cb37-6"></a>  <span class="fu">select</span>(centre)</span>
<span id="cb37-7"><a href="#cb37-7"></a>union_peaks</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell fragment">
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 188 ranges and 1 metadata column:
        seqnames            ranges strand |         centre
           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |      &lt;GRanges&gt;
    [1]    chr10 43048195-43048529      * | chr10:43048372
    [2]    chr10 43521739-43522260      * | chr10:43522014
    [3]    chr10 43540042-43540457      * | chr10:43540317
    [4]    chr10 43606184-43606573      * | chr10:43606417
    [5]    chr10 43851214-43851989      * | chr10:43851719
    ...      ...               ...    ... .            ...
  [184]    chr10 99168349-99168690      * | chr10:99168492
  [185]    chr10 99207862-99208157      * | chr10:99208006
  [186]    chr10 99331323-99331741      * | chr10:99331592
  [187]    chr10 99340107-99340354      * | chr10:99340248
  [188]    chr10 99621632-99621988      * | chr10:99621798
  -------
  seqinfo: 1 sequence from hg19 genome</code></pre>
</div>
</div>
<div class="fragment">
<ul>
<li>Peak centres are the mean from all replicates with an overlapping peak</li>
</ul>
</div>
</section>
<section id="mapping-peaks" class="slide level2">
<h2>Mapping Peaks</h2>
<ul>
<li>After finding consensus/union peaks:
<ul>
<li>What type of regulatory regions?</li>
<li>What genes are likely targets?</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li><code>defineRegions()</code> uses gene, transcript and exon-level information
<ul>
<li><code>GRangesList</code> of <em>non-overlapping regions</em> (promoter, intron etc)</li>
<li>Map peaks to these regions using <code>bestOverlap()</code></li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li><code>mapByFeature()</code> assigns peaks/ranges to <em>most likely genes</em>
<ul>
<li>Incorporates promoters, enhancers and long-range interactions</li>
</ul></li>
</ul>
</div>
</section>
<section id="defineregions" class="slide level2">
<h2><code>defineRegions</code></h2>
<div class="sourceCode" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">defineRegions</span>(</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>  genes, transcripts, exons,</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">promoter =</span> <span class="fu">c</span>(<span class="dv">2500</span>, <span class="dv">500</span>),</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">upstream =</span> <span class="dv">5000</span>,</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">intron =</span> <span class="cn">TRUE</span>,</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">proximal =</span> <span class="dv">10000</span>,</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">simplify =</span> <span class="cn">FALSE</span>,</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"gene_id"</span>, <span class="st">"gene_name"</span>, <span class="st">"transcript_id"</span>, <span class="st">"transcript_name"</span>),</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>  ...</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Unique Genomic Regions are defined <em>hierarchically</em> using a supplied gtf</p>
<div>
<ol type="1">
<li class="fragment">Promoters</li>
<li class="fragment">Upstream Promoters</li>
<li class="fragment">Exons</li>
<li class="fragment">Introns</li>
<li class="fragment">Proximal Intergenic</li>
<li class="fragment">Distal Intergenic</li>
</ol>
</div>
<aside class="notes">
<ul>
<li>Defined such that every base belongs to one and only one region</li>
<li>All distances are able to be set manually</li>
<li>Exons/Introns can be collapsed into ‘Gene Body’ by setting <code>intron = FALSE</code></li>
<li>Intergenic Regions can also be collapsed by setting <code>proximal = 0</code></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="defineregions-1" class="slide level2">
<h2><code>defineRegions</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>regions <span class="ot">&lt;-</span> <span class="fu">defineRegions</span>(<span class="at">genes =</span> gtf<span class="sc">$</span>gene, <span class="at">transcripts =</span> gtf<span class="sc">$</span>transcript, <span class="at">exons =</span> gtf<span class="sc">$</span>exon)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(regions)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "promoter"            "upstream_promoter"   "exon"               
[4] "intron"              "proximal_intergenic" "distal_intergenic"  </code></pre>
</div>
</div>
<div class="fragment">
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>regions<span class="sc">$</span>promoter</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 3359 ranges and 5 metadata columns:
         seqnames              ranges strand |                region
            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |           &lt;character&gt;
     [1]    chr10         61989-64988      * | Promoter (-2500/+500)
     [2]    chr10         66360-69392      * | Promoter (-2500/+500)
     [3]    chr10         88152-91151      * | Promoter (-2500/+500)
     [4]    chr10         94710-97961      * | Promoter (-2500/+500)
     [5]    chr10       119604-122603      * | Promoter (-2500/+500)
     ...      ...                 ...    ... .                   ...
  [3355]    chr10 135488075-135491074      * | Promoter (-2500/+500)
  [3356]    chr10 135491384-135494383      * | Promoter (-2500/+500)
  [3357]    chr10 135494684-135497683      * | Promoter (-2500/+500)
  [3358]    chr10 135503135-135506134      * | Promoter (-2500/+500)
  [3359]    chr10 135513071-135518323      * | Promoter (-2500/+500)
                                                                 gene_id
                                                         &lt;CharacterList&gt;
     [1]                                            ENSG00000260370.2_11
     [2]  ENSG00000260370.2_11,ENSG00000260370.2_11,ENSG00000260370.2_11
     [3]                                             ENSG00000237297.1_7
     [4] ENSG00000261456.6_7,ENSG00000261456.6_7,ENSG00000261456.6_7,...
     [5]                                             ENSG00000261456.6_7
     ...                                                             ...
  [3355]                                             ENSG00000276046.1_5
  [3356]                                             ENSG00000276904.1_5
  [3357]                                             ENSG00000278641.1_5
  [3358]                                               ENSG00000226880.1
  [3359] ENSG00000213147.3_3,ENSG00000261914.2_3,ENSG00000282875.1_8,...
                                               gene_name
                                         &lt;CharacterList&gt;
     [1]                                 ENSG00000260370
     [2] ENSG00000260370,ENSG00000260370,ENSG00000260370
     [3]                                 ENSG00000237297
     [4]                           TUBB8,TUBB8,TUBB8,...
     [5]                                           TUBB8
     ...                                             ...
  [3355]                                         DUX4L13
  [3356]                                         DUX4L14
  [3357]                                         DUX4L15
  [3358]                                    XX-2136C48.7
  [3359]         RPL23AP60,RPL23AP84,ENSG00000282875,...
                                                           transcript_id
                                                         &lt;CharacterList&gt;
     [1]                                             ENST00000562162.2_4
     [2]     ENST00000682078.1_1,ENST00000566940.2_4,ENST00000683798.1_1
     [3]                                             ENST00000416477.1_2
     [4] ENST00000568584.6_6,ENST00000568866.5_4,ENST00000561967.1_4,...
     [5]                                             ENST00000564130.2_4
     ...                                                             ...
  [3355]                                             ENST00000554103.2_2
  [3356]                                             ENST00000621227.1_2
  [3357]                                             ENST00000611059.1_2
  [3358]                                               ENST00000411632.1
  [3359] ENST00000450011.1_2,ENST00000571193.2_2,ENST00000635574.1_4,...
                                         transcript_name
                                         &lt;CharacterList&gt;
     [1]                                 ENST00000562162
     [2] ENST00000682078,ENST00000566940,ENST00000683798
     [3]                                 ENST00000416477
     [4]               TUBB8-206,TUBB8-207,TUBB8-201,...
     [5]                                       TUBB8-204
     ...                                             ...
  [3355]                                     DUX4L13-201
  [3356]                                     DUX4L14-201
  [3357]                                     DUX4L15-201
  [3358]                                XX-2136C48.7-001
  [3359] RPL23AP60-201,RPL23AP84-201,ENST00000635574,...
  -------
  seqinfo: 1 sequence from hg19 genome</code></pre>
</div>
</div>
<aside class="notes">
<ul>
<li>Distances are included in the region <code>mcol</code> when appropriate</li>
<li>All genes/transcripts these regions derive from as included</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="bestoverlap" class="slide level2">
<h2><code>bestOverlap</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb44" data-code-line-numbers="2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a>region_levels <span class="ot">&lt;-</span> <span class="fu">vapply</span>(regions, <span class="cf">function</span>(x) x<span class="sc">$</span>region[<span class="dv">1</span>], <span class="fu">character</span>(<span class="dv">1</span>))</span>
<span id="cb44-2"><a href="#cb44-2"></a>union_peaks<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">bestOverlap</span>(union_peaks, <span class="fu">unlist</span>(regions), <span class="at">var =</span> <span class="st">"region"</span>)</span>
<span id="cb44-3"><a href="#cb44-3"></a>union_peaks<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">factor</span>(union_peaks<span class="sc">$</span>region, <span class="at">levels =</span> region_levels)</span>
<span id="cb44-4"><a href="#cb44-4"></a>union_peaks</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell fragment">
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 188 ranges and 2 metadata columns:
        seqnames            ranges strand |         centre
           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |      &lt;GRanges&gt;
    [1]    chr10 43048195-43048529      * | chr10:43048372
    [2]    chr10 43521739-43522260      * | chr10:43522014
    [3]    chr10 43540042-43540457      * | chr10:43540317
    [4]    chr10 43606184-43606573      * | chr10:43606417
    [5]    chr10 43851214-43851989      * | chr10:43851719
    ...      ...               ...    ... .            ...
  [184]    chr10 99168349-99168690      * | chr10:99168492
  [185]    chr10 99207862-99208157      * | chr10:99208006
  [186]    chr10 99331323-99331741      * | chr10:99331592
  [187]    chr10 99340107-99340354      * | chr10:99340248
  [188]    chr10 99621632-99621988      * | chr10:99621798
                          region
                        &lt;factor&gt;
    [1]    Promoter (-2500/+500)
    [2]    Intergenic (&gt;10kb)   
    [3]    Intergenic (&gt;10kb)   
    [4]    Intron               
    [5]    Intergenic (&lt;10kb)   
    ...                      ...
  [184] Intron                  
  [185] Promoter (-2500/+500)   
  [186] Promoter (-2500/+500)   
  [187] Upstream Promoter (-5kb)
  [188] Upstream Promoter (-5kb)
  -------
  seqinfo: 1 sequence from hg19 genome</code></pre>
</div>
</div>
<div class="fragment">
<p><br> Now we have peaks annotated to regions <span class="math inline">\(\implies\)</span> <code>plotPie()</code></p>
<aside class="notes">
<ul>
<li>If passing a <code>GRangesList</code> elements names will be used for the mapping</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>


<section id="plotpie-1" class="slide level2">
<h2><code>plotPie</code></h2>

<img data-src="Bioc2023_presentation_files/figure-revealjs/plot-pie-labels-1.png" class="r-stretch"><aside class="notes">
<p>Now that we have our peaks mapped to genomic regions, we can use <code>plotPie()</code></p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="plotpie-2" class="slide level2 scrollable">
<h2><code>plotPie</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb49" data-code-line-numbers="2|4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1"></a><span class="fu">plotPie</span>(</span>
<span id="cb49-2"><a href="#cb49-2"></a>  union_peaks, <span class="at">fill =</span> <span class="st">"region"</span>, </span>
<span id="cb49-3"><a href="#cb49-3"></a>  <span class="at">min_p =</span> <span class="fl">0.05</span>, <span class="at">total_size =</span> <span class="fl">4.5</span>, <span class="at">cat_alpha =</span> <span class="fl">0.7</span>, <span class="at">cat_adj =</span> <span class="fl">0.05</span>,</span>
<span id="cb49-4"><a href="#cb49-4"></a>  <span class="at">cat_glue =</span> <span class="st">"{str_wrap(.data[[fill]], 15)}</span><span class="sc">\n</span><span class="st">n = {n}</span><span class="sc">\n</span><span class="st">({percent(p, 0.1)})"</span></span>
<span id="cb49-5"><a href="#cb49-5"></a>) <span class="sc">+</span>  </span>
<span id="cb49-6"><a href="#cb49-6"></a>  <span class="fu">scale_fill_viridis_d</span>(<span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<!-- - Can hide labels for infrequent overlaps (`min_p = 0.05`) and: -->
<!--     + Adjust size of totals (`total_size = 4.5`) -->
<!--     + Transparency, position of category labels etc -->
<ul>
<li>Label text is highly customisable
<ul>
<li>Uses <code>glue</code> syntax <span class="math inline">\(\implies\)</span> can include function calls and line-breaks</li>
<li>Totals available as <code>n</code>, proportions as <code>p</code>, categories as <code>.data[[fill]]</code></li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Slices can also be scaled by width</li>
</ul>
<aside class="notes">
<p>Now that we have our peaks mapped to genomic regions, we can use <code>plotPie()</code> - Here, region labels will be wrapped at 15 characters (<code>str_wrap()</code>) - Can also scale segments by range width - Also able to take data.frame objects as input, with scaling by any column</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="mapbyfeature" class="slide level2">
<h2><code>mapByFeature</code></h2>
<div class="sourceCode" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mapByFeature</span>(</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>  gr,    <span class="co"># Ranges to be mapped</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  genes, <span class="co"># Gene-annotations</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>  prom,  <span class="co"># Ranges annotated as promoters</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>  enh,   <span class="co"># Ranges annotated as enhancers</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>  gi,    <span class="co"># GInteractions object</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"gene_id"</span>, <span class="st">"gene_name"</span>, <span class="st">"symbol"</span>), <span class="co"># Columns to return (if present)</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">gr2prom =</span> <span class="dv">0</span>, <span class="at">gr2enh =</span> <span class="dv">0</span>, <span class="at">gr2gi =</span> <span class="dv">0</span>, <span class="co"># Distances between ranges &amp; features</span></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">gr2gene =</span> <span class="fl">1e+05</span>, <span class="at">prom2gene =</span> <span class="dv">0</span>, <span class="at">enh2gene =</span> <span class="fl">1e+05</span>, <span class="at">gi2gene =</span> <span class="dv">0</span>, ...</span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Ranges overlapping:</p>
<div>
<ol type="1">
<li class="fragment">a promoter <span class="math inline">\(\implies\)</span> assigned to that gene</li>
<li class="fragment">an enhancer <span class="math inline">\(\implies\)</span> all genes within a specified distance</li>
<li class="fragment">a long-range interaction <span class="math inline">\(\implies\)</span> all genes connected by the interaction</li>
<li class="fragment">none-of the above:
<ol type="a">
<li class="fragment">all overlapping genes or</li>
<li class="fragment">the nearest gene within a specified distance</li>
</ol></li>
</ol>
</div>
<aside class="notes">
<ul>
<li>Map ranges to genes using additional annotated ranges (i.e.&nbsp;features)</li>
<li>If no genes fit criteria, no genes are mapped to that peak/range</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="mapbyfeature-1" class="slide level2">
<h2><code>mapByFeature</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>union_peaks <span class="ot">&lt;-</span> <span class="fu">mapByFeature</span>(union_peaks, <span class="at">genes =</span> gtf<span class="sc">$</span>gene, <span class="at">prom =</span> regions<span class="sc">$</span>promoter)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>union_peaks[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 5 ranges and 4 metadata columns:
      seqnames            ranges strand |         centre                region
         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |      &lt;GRanges&gt;              &lt;factor&gt;
  [1]    chr10 43048195-43048529      * | chr10:43048372 Promoter (-2500/+500)
  [2]    chr10 43521739-43522260      * | chr10:43522014 Intergenic (&gt;10kb)   
  [3]    chr10 43540042-43540457      * | chr10:43540317 Intergenic (&gt;10kb)   
  [4]    chr10 43606184-43606573      * | chr10:43606417 Intron               
  [5]    chr10 43851214-43851989      * | chr10:43851719 Intergenic (&lt;10kb)   
                                       gene_id       gene_name
                               &lt;CharacterList&gt; &lt;CharacterList&gt;
  [1] ENSG00000234420.9_10,ENSG00000270762.1_3 ZNF37BP,FXYD6P1
  [2]                        ENSG00000263795.1         MIR5100
  [3]                    ENSG00000165731.21_15             RET
  [4]                    ENSG00000165731.21_15             RET
  [5]                      ENSG00000285712.1_7 ENSG00000285712
  -------
  seqinfo: 1 sequence from hg19 genome</code></pre>
</div>
</div>
<aside class="notes">
<ul>
<li><p>Genes mapped to promoters/enhancers will be used preferentially</p></li>
<li><p>By this point, we have peaks mapped to regulatory regions and most likely target genes</p></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section></section>
<section>
<section id="differential-signal-analysis-1" class="title-slide slide level1 center" data-background-color="#3c3c44" data-background-image="https://raw.githubusercontent.com/Bioconductor/BiocStickers/devel/extraChIPs/extraChIPs.png" data-background-size="35%" data-background-opacity="0.2">
<h1>Differential Signal Analysis</h1>

</section>
<section id="two-approaches-to-differential-signal-analysis" class="slide level2">
<h2>Two Approaches to Differential Signal Analysis</h2>
<aside class="notes">
<ul>
<li><code>dualFilter()</code> wraps both <code>filterWindowsProportion()</code> and <code>filterWindowsControl()</code></li>
<li>If no input sample, only filters based on overall signal</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<div class="panel-tabset">
<ul id="tabset-1" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-1-1">Fixed-Width Analaysis</a></li><li><a href="#tabset-1-2">Sliding Windows</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1">
<p>Highly comparable to <code>DiffBind</code></p>
<ol type="1">
<li>Define a set of fixed-width ranges:
<ul>
<li>Centres from <code>importPeaks()</code> <span class="math inline">\(\implies\)</span> <code>makeConsensus()</code></li>
<li>Resize to fixed-width</li>
</ul></li>
<li>Count Reads using <code>csaw::regionCounts()</code> + <code>BamFileList</code>
<ul>
<li>Returns a <code>RangedSummarizedExperiment</code></li>
</ul></li>
<li>Perform analysis using <code>fitAssayDiff()</code></li>
</ol>
</div>
<div id="tabset-1-2">
<p>Extends <code>csaw</code> approaches</p>
<ol type="1">
<li>Define window size &amp; step</li>
<li>Count reads using <code>csaw::windowCounts()</code> + <code>BamFileList</code>
<ul>
<li>Returns a <code>RangedSummarizedExperiment</code></li>
</ul></li>
<li>Filter windows for those likely containing signal: <code>dualFilter()</code>
<ul>
<li>Uses a set of ‘guide ranges’ to set filtering thresholds</li>
</ul></li>
<li>Fit windows using <code>fitAssayDiff()</code></li>
<li><em>Merge neighbouring windows</em> for final results</li>
</ol>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>Conventionally, sliding windows don’t require pre-defined peaks</li>
<li>dualFilter sets thresholds for signal above input &amp; total signal using guide ranges</li>
<li>If no input provided, <code>dualFilter</code> only uses total signal</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="differential-signal-analysis-2" class="slide level2">
<h2>Differential Signal Analysis</h2>
<ul>
<li>Visualisation functions after counting reads
<ul>
<li><code>plotAssayPCA</code>, <code>plotAssayRle</code>, <code>plotAssayDensities</code></li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Differential Signal Analysis: <code>fitAssayDiff()</code>
<ul>
<li><code>edgeR::glmQLF()</code> or <code>limma-trend</code></li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Sliding window analysis <span class="math inline">\(\implies\)</span> merge neighboring windows
<ul>
<li><code>mergeByHMP()</code> <u><em>uses harmonic-mean p</em></u> <span class="citation" data-cites="Wilson2019-ln">(<a href="#/references" role="doc-biblioref" onclick="">Wilson 2019</a>)</span></li>
<li><code>mergeByCol()</code>, <code>mergeBySig()</code>: wraps <code>csaw</code> methods</li>
<li>All return the relevant underlying windows (<code>keyval_range</code>)</li>
</ul></li>
</ul>
<aside class="notes">
<ul>
<li>Sliding window analysis requires an extra step.</li>
<li>Both mergeByCol and mergeBySig are <code>csaw</code> methods.</li>
<li>mergeByHMP is unique to extraChIPs and uses asymptotically exact harmonic-mean p</li>
<li>Inclusion of the keyvalue range can be very helpful when plotting and for downstream analysis</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="fixed-width-windows" class="slide level2">
<h2>Fixed-Width Windows</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb53" data-code-line-numbers="3-7|8-11|12-14|15-18|1-23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a><span class="fu">library</span>(Rsamtools)</span>
<span id="cb53-2"><a href="#cb53-2"></a><span class="fu">library</span>(csaw)</span>
<span id="cb53-3"><a href="#cb53-3"></a><span class="do">## Centre &amp; resize the peaks</span></span>
<span id="cb53-4"><a href="#cb53-4"></a>centred_peaks <span class="ot">&lt;-</span> union_peaks <span class="sc">%&gt;%</span> </span>
<span id="cb53-5"><a href="#cb53-5"></a>  <span class="fu">mutate</span>(<span class="at">union_peak =</span> <span class="fu">granges</span>(.)) <span class="sc">%&gt;%</span>  <span class="co"># Add the original ranges to mcols</span></span>
<span id="cb53-6"><a href="#cb53-6"></a>  <span class="fu">colToRanges</span>(<span class="at">var =</span> <span class="st">"centre"</span>) <span class="sc">%&gt;%</span> <span class="co"># Move the centred range to the 'backbone'</span></span>
<span id="cb53-7"><a href="#cb53-7"></a>  <span class="fu">resize</span>(<span class="at">width =</span> <span class="dv">400</span>, <span class="at">fix =</span> <span class="st">"center"</span>) <span class="co"># Resize to a fixed with</span></span>
<span id="cb53-8"><a href="#cb53-8"></a><span class="do">## Define the BamFileList</span></span>
<span id="cb53-9"><a href="#cb53-9"></a>er_bfl <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="st">"ER"</span>, <span class="fu">paste0</span>(samples<span class="sc">$</span>ER<span class="sc">$</span>accession, <span class="st">".bam"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb53-10"><a href="#cb53-10"></a>  <span class="fu">BamFileList</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb53-11"><a href="#cb53-11"></a>  <span class="fu">setNames</span>(samples<span class="sc">$</span>ER<span class="sc">$</span>accession)</span>
<span id="cb53-12"><a href="#cb53-12"></a><span class="do">## Count Reads</span></span>
<span id="cb53-13"><a href="#cb53-13"></a>er_se <span class="ot">&lt;-</span> <span class="fu">regionCounts</span>(er_bfl, centred_peaks, <span class="at">ext =</span> <span class="dv">200</span>)</span>
<span id="cb53-14"><a href="#cb53-14"></a><span class="fu">seqinfo</span>(er_se) <span class="ot">&lt;-</span> sq</span>
<span id="cb53-15"><a href="#cb53-15"></a><span class="do">## Re-assign union peaks to core ranges after counting</span></span>
<span id="cb53-16"><a href="#cb53-16"></a><span class="fu">rowRanges</span>(er_se) <span class="ot">&lt;-</span> <span class="fu">rowRanges</span>(er_se) <span class="sc">%&gt;%</span> </span>
<span id="cb53-17"><a href="#cb53-17"></a>  <span class="fu">colToRanges</span>(<span class="at">var =</span> <span class="st">"union_peak"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb53-18"><a href="#cb53-18"></a>  <span class="fu">select</span>(region, <span class="fu">starts_with</span>(<span class="st">"gene"</span>))</span>
<span id="cb53-19"><a href="#cb53-19"></a><span class="do">## Add all sample annotations</span></span>
<span id="cb53-20"><a href="#cb53-20"></a><span class="fu">colData</span>(er_se) <span class="ot">&lt;-</span> <span class="fu">colData</span>(er_se) <span class="sc">%&gt;%</span> </span>
<span id="cb53-21"><a href="#cb53-21"></a>  <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="st">"accession"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb53-22"><a href="#cb53-22"></a>  <span class="fu">left_join</span>(samples<span class="sc">$</span>ER) <span class="sc">%&gt;%</span> </span>
<span id="cb53-23"><a href="#cb53-23"></a>  <span class="fu">as</span>(<span class="st">"DataFrame"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<aside class="notes">
<ul>
<li>The only real enhancements here are the shifting of ranges with <code>colToRanges</code> and the way centred peaks were formed</li>
<li>Remember fixed-width ranges are a proxy for the peaks detected by our peak caller</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="visualising-counts" class="slide level2">
<h2>Visualising Counts</h2>
<div class="panel-tabset">
<ul id="tabset-2" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-2-1"><code>plotAssayDensities</code></a></li><li><a href="#tabset-2-2"><code>plotAssayPCA</code></a></li></ul>
<div class="tab-content">
<div id="tabset-2-1">
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAssayDensities</span>(er_se, <span class="at">trans =</span> <span class="st">"log1p"</span>, <span class="at">colour =</span> <span class="st">"treatment"</span>) <span class="sc">+</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> treat_colours)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img data-src="Bioc2023_presentation_files/figure-revealjs/plot-er-densities-1.png" width="576"></p>
</div>
</div>
</div>
<div id="tabset-2-2">
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAssayPCA</span>(er_se, <span class="at">trans =</span> <span class="st">"log1p"</span>, <span class="at">colour =</span> <span class="st">"treatment"</span>, <span class="at">label =</span> <span class="st">"accession"</span>) <span class="sc">+</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> treat_colours)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img data-src="Bioc2023_presentation_files/figure-revealjs/plot-er-pca-1.png" width="576"></p>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="rle-plots" class="slide level2">
<h2>RLE Plots</h2>
<div class="panel-tabset">
<ul id="tabset-3" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-3-1">RLE By Sample</a></li><li><a href="#tabset-3-2">RLE By Treatment</a></li></ul>
<div class="tab-content">
<div id="tabset-3-1">
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAssayRle</span>(er_se, <span class="at">trans =</span> <span class="st">"log1p"</span>, <span class="at">fill =</span> <span class="st">"treatment"</span>) <span class="sc">+</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> treat_colours)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img data-src="Bioc2023_presentation_files/figure-revealjs/er-rle-by-sample-1.png" width="576"></p>
</div>
</div>
</div>
<div id="tabset-3-2">
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAssayRle</span>(er_se, <span class="at">trans =</span> <span class="st">"log1p"</span>, <span class="at">fill =</span> <span class="st">"treatment"</span>, <span class="at">by_x =</span> <span class="st">"treatment"</span>) <span class="sc">+</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> treat_colours)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img data-src="Bioc2023_presentation_files/figure-revealjs/er-rle-by-treatment-1.png" width="576"></p>
</div>
</div>
</div>
</div>
</div>
<aside class="notes">
<p>Standard plot uses median values across all samples</p>
<ul>
<li>Can also be set to se median values within a treatment group/passage etc</li>
<li>Can show RLE summarised by group</li>
</ul>
<p>This is helpful when amount of signal may vary between treatments, such as may be the case where treatment stimulates cytoplasmic to nuclear translocation</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="sliding-windows-1" class="slide level2">
<h2>Sliding Windows</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb58" data-code-line-numbers="1-6|7-16|1-17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1"></a><span class="do">## Define the BamFile List INCLUDING the Input</span></span>
<span id="cb58-2"><a href="#cb58-2"></a>acc <span class="ot">&lt;-</span> <span class="fu">c</span>(samples<span class="sc">$</span>H3K27ac<span class="sc">$</span>accession, samples<span class="sc">$</span>H3K27ac<span class="sc">$</span>input) <span class="sc">%&gt;%</span> </span>
<span id="cb58-3"><a href="#cb58-3"></a>  <span class="fu">unique</span>()</span>
<span id="cb58-4"><a href="#cb58-4"></a>h3k_bfl <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="st">"H3K27ac"</span>, <span class="fu">paste0</span>(acc, <span class="st">".bam"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb58-5"><a href="#cb58-5"></a>  <span class="fu">BamFileList</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb58-6"><a href="#cb58-6"></a>  <span class="fu">setNames</span>(acc)</span>
<span id="cb58-7"><a href="#cb58-7"></a><span class="do">## Setup the parameters for windows &amp; counts</span></span>
<span id="cb58-8"><a href="#cb58-8"></a>rp <span class="ot">&lt;-</span> <span class="fu">readParam</span>(<span class="at">pe =</span> <span class="st">"none"</span>, <span class="at">restrict =</span> <span class="st">"chr10"</span>, <span class="at">discard =</span> bg_list)</span>
<span id="cb58-9"><a href="#cb58-9"></a>win_size <span class="ot">&lt;-</span> <span class="dv">150</span></span>
<span id="cb58-10"><a href="#cb58-10"></a>win_step <span class="ot">&lt;-</span> <span class="dv">50</span></span>
<span id="cb58-11"><a href="#cb58-11"></a><span class="do">## Count reads for all windows</span></span>
<span id="cb58-12"><a href="#cb58-12"></a>wincounts <span class="ot">&lt;-</span> <span class="fu">windowCounts</span>(</span>
<span id="cb58-13"><a href="#cb58-13"></a>  <span class="at">bam.files =</span> h3k_bfl,</span>
<span id="cb58-14"><a href="#cb58-14"></a>  <span class="at">spacing =</span> win_step, <span class="at">width =</span> win_size, <span class="at">ext =</span> <span class="dv">200</span>,</span>
<span id="cb58-15"><a href="#cb58-15"></a>  <span class="at">filter =</span> <span class="fu">length</span>(h3k_bfl), <span class="at">param =</span> rp</span>
<span id="cb58-16"><a href="#cb58-16"></a>)</span>
<span id="cb58-17"><a href="#cb58-17"></a><span class="fu">dim</span>(wincounts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell fragment">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 467904      7</code></pre>
</div>
</div>
</section>
<section id="filter-sliding-windows" class="slide level2">
<h2>Filter Sliding Windows</h2>
<ul>
<li>Retain windows which are most likely to contain signal</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb60" data-code-line-numbers="1-3|4-9|10-14|1-15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1"></a><span class="do">## Load in the pre-defined ranges</span></span>
<span id="cb60-2"><a href="#cb60-2"></a>guide_ranges <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="st">"H3K27ac"</span>, <span class="st">"H3K27ac_chr10.bed"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb60-3"><a href="#cb60-3"></a>  <span class="fu">import.bed</span>(<span class="at">seqinfo =</span> sq)</span>
<span id="cb60-4"><a href="#cb60-4"></a><span class="do">## Filter using the guide ranges</span></span>
<span id="cb60-5"><a href="#cb60-5"></a>filtcounts <span class="ot">&lt;-</span> <span class="fu">dualFilter</span>(</span>
<span id="cb60-6"><a href="#cb60-6"></a>  <span class="at">x =</span> wincounts[, samples<span class="sc">$</span>H3K27ac<span class="sc">$</span>accession],</span>
<span id="cb60-7"><a href="#cb60-7"></a>  <span class="at">bg  =</span>  wincounts[, <span class="fu">unique</span>(samples<span class="sc">$</span>H3K27ac<span class="sc">$</span>input)],</span>
<span id="cb60-8"><a href="#cb60-8"></a>  <span class="at">ref =</span> guide_ranges, <span class="at">q =</span> <span class="fl">0.6</span></span>
<span id="cb60-9"><a href="#cb60-9"></a>)</span>
<span id="cb60-10"><a href="#cb60-10"></a><span class="do">## Update the colData</span></span>
<span id="cb60-11"><a href="#cb60-11"></a><span class="fu">colData</span>(filtcounts) <span class="ot">&lt;-</span> <span class="fu">colData</span>(filtcounts) <span class="sc">%&gt;%</span> </span>
<span id="cb60-12"><a href="#cb60-12"></a>  <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="st">"accession"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb60-13"><a href="#cb60-13"></a>  <span class="fu">left_join</span>(samples<span class="sc">$</span>H3K27ac) <span class="sc">%&gt;%</span> </span>
<span id="cb60-14"><a href="#cb60-14"></a>  <span class="fu">as</span>(<span class="st">"DataFrame"</span>)</span>
<span id="cb60-15"><a href="#cb60-15"></a><span class="fu">dim</span>(filtcounts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell fragment">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 19017     6</code></pre>
</div>
</div>
<div class="fragment">
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(filtcounts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of length 2
names(2): counts logCPM</code></pre>
</div>
</div>
<aside class="notes">
<ul>
<li><code>q</code> represents the proportion of windows which overlap the guide ranges that should be returned</li>
<li>Thresholds are set based on this value</li>
<li>By aplying this strategy we’ve gone from &gt;400,000 to 19,000 windows</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="fitassaydiff" class="slide level2">
<h2><code>fitAssayDiff()</code></h2>
<div class="sourceCode" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fitAssayDiff</span>(</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>  x, <span class="at">assay =</span> <span class="st">"counts"</span>,</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">design =</span> <span class="cn">NULL</span>, <span class="at">coef =</span> <span class="cn">NULL</span>,</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">lib.size =</span> <span class="st">"totals"</span>,</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="fu">c</span>(<span class="st">"qlf"</span>, <span class="st">"lt"</span>),</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">norm =</span> <span class="fu">c</span>(<span class="st">"none"</span>, <span class="st">"TMM"</span>, <span class="st">"RLE"</span>, <span class="st">"TMMwsp"</span>, <span class="st">"upperquartile"</span>),</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">groups =</span> <span class="cn">NULL</span>,</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">fc =</span> <span class="dv">1</span>, <span class="at">lfc =</span> <span class="fu">log2</span>(fc),</span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">asRanges =</span> <span class="cn">FALSE</span>,</span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">offset =</span> <span class="cn">NULL</span>,</span>
<span id="cb64-11"><a href="#cb64-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">null =</span> <span class="fu">c</span>(<span class="st">"interval"</span>, <span class="st">"worst.case"</span>),</span>
<span id="cb64-12"><a href="#cb64-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">weighted =</span> <span class="cn">FALSE</span>,</span>
<span id="cb64-13"><a href="#cb64-13" aria-hidden="true" tabindex="-1"></a>  ...</span>
<span id="cb64-14"><a href="#cb64-14" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li><code>edgeR::glmQLFit</code> is default method <span class="citation" data-cites="edgeRQLF2016">(<a href="#/references" role="doc-biblioref" onclick="">Chen, Lun, and Smyth 2016</a>)</span></li>
<li><code>limma-trend</code> for logCPM values <span class="citation" data-cites="Law2014-xq">(<a href="#/references" role="doc-biblioref" onclick="">Law et al. 2014</a>)</span></li>
<li>Can apply range-based H<sub>0</sub> <span class="citation" data-cites="McCarthy2009-qf">(<a href="#/references" role="doc-biblioref" onclick="">McCarthy and Smyth 2009</a>)</span></li>
<li>Optionally normalise <em>within treatment groups</em> (e.g.&nbsp;<code>groups = "treat"</code>)</li>
<li>Library sizes can be set as colSums (<code>lib.size = NULL</code>)</li>
</ul>
<aside class="notes">
<p>The default settings would be:</p>
<ol type="1">
<li>Library-Size Normalisation</li>
<li>GLM-QL models</li>
</ol>
<p>Very similar to DiffBind defaults, but calls <code>edgeR</code></p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="fixed-width-windows-1" class="slide level2">
<h2>Fixed-Width Windows</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb65" data-code-line-numbers="1-2|3-7|1-16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1"></a><span class="do">## Setup the design matrix</span></span>
<span id="cb65-2"><a href="#cb65-2"></a>X <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>treatment, <span class="at">data =</span> <span class="fu">colData</span>(er_se))</span>
<span id="cb65-3"><a href="#cb65-3"></a><span class="do">## Fit the model</span></span>
<span id="cb65-4"><a href="#cb65-4"></a>er_results <span class="ot">&lt;-</span> <span class="fu">fitAssayDiff</span>(</span>
<span id="cb65-5"><a href="#cb65-5"></a>  er_se, <span class="at">design =</span> X, <span class="at">norm =</span> <span class="st">"TMM"</span>, <span class="at">asRanges =</span> <span class="cn">TRUE</span>, <span class="at">fc =</span> <span class="fl">1.2</span></span>
<span id="cb65-6"><a href="#cb65-6"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb65-7"><a href="#cb65-7"></a>  <span class="fu">addDiffStatus</span>()</span>
<span id="cb65-8"><a href="#cb65-8"></a><span class="do">## Print the significant ranges</span></span>
<span id="cb65-9"><a href="#cb65-9"></a>er_results <span class="sc">%&gt;%</span> </span>
<span id="cb65-10"><a href="#cb65-10"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb65-11"><a href="#cb65-11"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb65-12"><a href="#cb65-12"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>gene_id) <span class="sc">%&gt;%</span> </span>
<span id="cb65-13"><a href="#cb65-13"></a>  <span class="fu">mutate</span>(</span>
<span id="cb65-14"><a href="#cb65-14"></a>    <span class="at">gene_name =</span> <span class="fu">vapply</span>(gene_name, paste, <span class="fu">character</span>(<span class="dv">1</span>), <span class="at">collapse =</span> <span class="st">"; "</span>)</span>
<span id="cb65-15"><a href="#cb65-15"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb65-16"><a href="#cb65-16"></a>  <span class="fu">arrange</span>(PValue) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell fragment">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 7 × 8
  range                   region  gene_name logFC logCPM   PValue     FDR status
  &lt;chr&gt;                   &lt;fct&gt;   &lt;chr&gt;     &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt; &lt;fct&gt; 
1 chr10:81101906-81102928 Upstre… PPIF      1.91    8.03 4.43e-11 8.34e-9 Incre…
2 chr10:79629641-79630271 Promot… DLG5      0.967   8.23 4.20e- 6 3.95e-4 Incre…
3 chr10:89407752-89408138 Intron  ENSG0000… 1.71    6.20 1.08e- 5 6.79e-4 Incre…
4 chr10:52233596-52233998 Exon    SGMS1; E… 1.13    6.70 2.37e- 4 1.11e-2 Incre…
5 chr10:51547205-51547782 Promot… MSMB      0.665   7.44 1.12e- 3 4.22e-2 Incre…
6 chr10:79266991-79267392 Intron  KCNMA1    1.24    5.56 1.58e- 3 4.52e-2 Incre…
7 chr10:71879941-71880280 Promot… AIFM2     1.20    5.68 1.68e- 3 4.52e-2 Incre…</code></pre>
</div>
</div>
<aside class="notes">
<p>Note that we’re applying a range-based H<sub>0</sub> such we’re only interested in sites with a &gt; 20% change in signal</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>

<section id="plotsplitdonut-1" class="slide level2">
<h2>plotSplitDonut</h2>

<img data-src="Bioc2023_presentation_files/figure-revealjs/plot-donut-custom-1.png" class="r-stretch"></section>
<section id="plotsplitdonut-2" class="slide level2">
<h2>plotSplitDonut</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb68" data-code-line-numbers="1-7,9-10|11-12|14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1"></a><span class="do">## Define inner &amp; outer palettes</span></span>
<span id="cb68-2"><a href="#cb68-2"></a>region_colours <span class="ot">&lt;-</span> <span class="fu">hcl.colors</span>(<span class="fu">length</span>(region_levels), <span class="st">"viridis"</span>, <span class="at">rev =</span> <span class="cn">TRUE</span>)</span>
<span id="cb68-3"><a href="#cb68-3"></a><span class="fu">names</span>(region_colours) <span class="ot">&lt;-</span> region_levels</span>
<span id="cb68-4"><a href="#cb68-4"></a>status_colours <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb68-5"><a href="#cb68-5"></a>  <span class="at">Increased =</span> <span class="st">"red3"</span>, <span class="at">Decreased =</span> <span class="st">"royalblue3"</span>, <span class="at">Unchanged =</span> <span class="st">"grey40"</span>,</span>
<span id="cb68-6"><a href="#cb68-6"></a>  <span class="at">Undetected =</span> <span class="st">"grey70"</span></span>
<span id="cb68-7"><a href="#cb68-7"></a>)</span>
<span id="cb68-8"><a href="#cb68-8"></a><span class="fu">plotSplitDonut</span>(</span>
<span id="cb68-9"><a href="#cb68-9"></a>  er_results, <span class="at">inner =</span> <span class="st">"region"</span>, <span class="at">outer =</span> <span class="st">"status"</span>,</span>
<span id="cb68-10"><a href="#cb68-10"></a>  <span class="at">inner_palette =</span> region_colours, <span class="at">outer_palette =</span> status_colours,</span>
<span id="cb68-11"><a href="#cb68-11"></a>  <span class="at">inner_glue =</span> <span class="st">"{str_wrap(.data[[inner]],15)}</span><span class="sc">\n</span><span class="st">n = {n}"</span>, <span class="at">inner_label_alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb68-12"><a href="#cb68-12"></a>  <span class="at">outer_glue =</span> <span class="st">"{str_wrap(.data[[inner]],15)}</span><span class="sc">\n</span><span class="st">ER {.data[[outer]]}</span><span class="sc">\n</span><span class="st">n = {n}"</span>,</span>
<span id="cb68-13"><a href="#cb68-13"></a>  <span class="at">outer_max_p =</span> <span class="fl">0.03</span>, <span class="at">outer_min_p =</span> <span class="dv">0</span>,</span>
<span id="cb68-14"><a href="#cb68-14"></a>  <span class="at">explode_outer =</span> <span class="st">"Increased"</span>, <span class="at">explode_r =</span> <span class="fl">0.2</span></span>
<span id="cb68-15"><a href="#cb68-15"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Can set inner and outer palettes separately</li>
<li>Labels are customisable using <code>glue</code></li>
<li>Explode segments based <code>regex</code> queries</li>
</ul>
</section>
<section id="sliding-windows-2" class="slide level2">
<h2>Sliding Windows</h2>
<ul>
<li>Involves the additional step of merging sliding windows</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb69" data-code-line-numbers="1|2|3-6|7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1"></a>X <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>treatment, <span class="at">data =</span> <span class="fu">colData</span>(filtcounts))</span>
<span id="cb69-2"><a href="#cb69-2"></a>h3k_all <span class="ot">&lt;-</span> <span class="fu">fitAssayDiff</span>(filtcounts, <span class="at">norm =</span> <span class="st">"TMM"</span>, <span class="at">design =</span> X, <span class="at">asRanges =</span> <span class="cn">TRUE</span>)</span>
<span id="cb69-3"><a href="#cb69-3"></a>h3k_results <span class="ot">&lt;-</span> <span class="fu">mergeByHMP</span>(</span>
<span id="cb69-4"><a href="#cb69-4"></a>  h3k_all, <span class="at">min_win =</span> <span class="dv">2</span>, <span class="at">merge_within =</span> win_size <span class="sc">+</span> <span class="dv">1</span>, <span class="at">keyval =</span> <span class="st">"merged"</span></span>
<span id="cb69-5"><a href="#cb69-5"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-6"><a href="#cb69-6"></a>  <span class="fu">addDiffStatus</span>()</span>
<span id="cb69-7"><a href="#cb69-7"></a><span class="fu">arrange</span>(h3k_results, hmp)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell fragment">
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 5 ranges and 9 metadata columns:
      seqnames            ranges strand | n_windows      n_up    n_down
         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
  [1]    chr10 79256601-79258700      * |        35         4         0
  [2]    chr10 43689201-43690150      * |        17         1         0
  [3]    chr10 74008151-74008750      * |        10         2         0
  [4]    chr10 63859401-63859750      * |         5         1         0
  [5]    chr10 79623301-79635950      * |       251         7         0
                 keyval_range    logCPM     logFC         hmp     hmp_fdr
                    &lt;GRanges&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
  [1] chr10:79257751-79258050   7.12950  1.693320 5.67542e-15 3.64930e-12
  [2] chr10:43689651-43689800   6.69278  1.866274 2.74260e-12 8.81747e-10
  [3] chr10:74008301-74008500   6.54608  1.735779 2.67001e-10 5.72273e-08
  [4] chr10:63859601-63859750   6.03721  1.848069 5.65428e-08 7.76644e-06
  [5] chr10:79630151-79630650   7.87510  0.905969 6.03922e-08 7.76644e-06
         status
       &lt;factor&gt;
  [1] Increased
  [2] Increased
  [3] Increased
  [4] Increased
  [5] Increased
  -------
  seqinfo: 1 sequence from hg19 genome</code></pre>
</div>
</div>
<aside class="notes">
<p>By default, returns the window with the lowest p-value Can return the entire range containing individual p-values below the final HMP</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="map-results" class="slide level2">
<h2>Map Results</h2>
<ul>
<li>Map ranges <strong>after</strong> analysis</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>h3k_results <span class="ot">&lt;-</span> h3k_results <span class="sc">%&gt;%</span> </span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">region =</span> <span class="fu">bestOverlap</span>(.<span class="sc">$</span>keyval_range, <span class="fu">unlist</span>(regions), <span class="at">var =</span> <span class="st">"region"</span>)</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mapByFeature</span>(<span class="at">genes =</span> gtf<span class="sc">$</span>gene, <span class="at">prom =</span> regions<span class="sc">$</span>promoter)</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a><span class="fu">arrange</span>(h3k_results, hmp)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 5 ranges and 12 metadata columns:
      seqnames            ranges strand | n_windows      n_up    n_down
         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
  [1]    chr10 79256601-79258700      * |        35         4         0
  [2]    chr10 43689201-43690150      * |        17         1         0
  [3]    chr10 74008151-74008750      * |        10         2         0
  [4]    chr10 63859401-63859750      * |         5         1         0
  [5]    chr10 79623301-79635950      * |       251         7         0
                 keyval_range    logCPM     logFC         hmp     hmp_fdr
                    &lt;GRanges&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
  [1] chr10:79257751-79258050   7.12950  1.693320 5.67542e-15 3.64930e-12
  [2] chr10:43689651-43689800   6.69278  1.866274 2.74260e-12 8.81747e-10
  [3] chr10:74008301-74008500   6.54608  1.735779 2.67001e-10 5.72273e-08
  [4] chr10:63859601-63859750   6.03721  1.848069 5.65428e-08 7.76644e-06
  [5] chr10:79630151-79630650   7.87510  0.905969 6.03922e-08 7.76644e-06
         status                region                                   gene_id
       &lt;factor&gt;           &lt;character&gt;                           &lt;CharacterList&gt;
  [1] Increased                Intron                     ENSG00000156113.25_17
  [2] Increased    Intergenic (&lt;10kb)                     ENSG00000198915.12_14
  [3] Increased    Intergenic (&gt;10kb)                       ENSG00000166295.9_6
  [4] Increased    Intergenic (&lt;10kb)                     ENSG00000150347.17_12
  [5] Increased Promoter (-2500/+500) ENSG00000204049.1_8,ENSG00000151208.17_11
                 gene_name
           &lt;CharacterList&gt;
  [1]               KCNMA1
  [2]             RASGEF1A
  [3]              ANAPC16
  [4]               ARID5B
  [5] ENSG00000204049,DLG5
  -------
  seqinfo: 1 sequence from hg19 genome</code></pre>
</div>
</div>
</section></section>
<section>
<section id="comparing-results" class="title-slide slide level1 center" data-background-color="#3c3c44" data-background-image="https://raw.githubusercontent.com/Bioconductor/BiocStickers/devel/extraChIPs/extraChIPs.png" data-background-size="35%" data-background-opacity="0.2">
<h1>Comparing Results</h1>

</section>
<section id="comparing-two-targets" class="slide level2">
<h2>Comparing Two Targets</h2>
<ul>
<li>Check where change is found in both/either/neither
<ul>
<li><code>mapGrlCols()</code></li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Plot combined status
<ul>
<li><code>plotSplitDonut()</code></li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Compare binding patterns:
<ul>
<li>Both signal (logCPM) and change (logFC)</li>
<li><code>plotPairwise()</code></li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Inspect Genomic Regions of interest
<ul>
<li><code>plotHFGC()</code></li>
</ul></li>
</ul>
</div>
</section>
<section id="comparing-two-targets-1" class="slide level2">
<h2>Comparing Two Targets</h2>
<ul>
<li>Comparison functions utilise <code>GRangesList</code> objects</li>
<li>Require shared column names</li>
</ul>
<div class="fragment">
<div class="cell">
<div class="sourceCode cell-code" id="cb73" data-code-line-numbers="4-7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1"></a>all_results <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>(</span>
<span id="cb73-2"><a href="#cb73-2"></a>  <span class="at">ER =</span> er_results,</span>
<span id="cb73-3"><a href="#cb73-3"></a>  <span class="at">H3K27ac =</span> h3k_results <span class="sc">%&gt;%</span> </span>
<span id="cb73-4"><a href="#cb73-4"></a>    <span class="fu">select</span>(</span>
<span id="cb73-5"><a href="#cb73-5"></a>      region, <span class="fu">starts_with</span>(<span class="st">"gene"</span>), <span class="fu">starts_with</span>(<span class="st">"log"</span>), </span>
<span id="cb73-6"><a href="#cb73-6"></a>      <span class="at">PValue =</span> hmp, <span class="at">FDR =</span> hmp_fdr, status</span>
<span id="cb73-7"><a href="#cb73-7"></a>    )</span>
<span id="cb73-8"><a href="#cb73-8"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<aside class="notes">
<ul>
<li>First, we need to setup each element to have the same column names</li>
<li>The we can compare the distributions of signal from both targets</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="mapgrlcols" class="slide level2">
<h2>mapGrlCols</h2>
<ul>
<li>Columns from each element can be mapped
<ul>
<li>Columns can be combined and collapsed into a single column</li>
</ul></li>
<li>Viable for a <u>named</u> <code>GRangesList</code> of any size</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>all_results <span class="sc">%&gt;%</span> </span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mapGrlCols</span>(<span class="at">var =</span> <span class="fu">c</span>(<span class="st">"logFC"</span>, <span class="st">"status"</span>), <span class="at">collapse =</span> <span class="st">"gene_name"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 741 ranges and 5 metadata columns:
        seqnames            ranges strand |  ER_logFC H3K27ac_logFC ER_status
           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;     &lt;numeric&gt;  &lt;factor&gt;
    [1]    chr10 42862951-42863350      * |        NA     0.3345653 NA       
    [2]    chr10 43047801-43048800      * | -0.301091    -0.0780895 Unchanged
    [3]    chr10 43047801-43048800      * | -0.301091    -0.2619553 Unchanged
    [4]    chr10 43132901-43134800      * |        NA     0.1350775 NA       
    [5]    chr10 43135701-43137250      * |        NA     0.0791231 NA       
    ...      ...               ...    ... .       ...           ...       ...
  [737]    chr10 99496401-99497850      * |        NA     0.0551099 NA       
  [738]    chr10 99608301-99610050      * |        NA    -0.5008959 NA       
  [739]    chr10 99621632-99621988      * |  0.115139            NA Unchanged
  [740]    chr10 99628751-99629400      * |        NA    -0.2298606 NA       
  [741]    chr10 99894101-99895400      * |        NA    -0.1143540 NA       
        H3K27ac_status              gene_name
              &lt;factor&gt;            &lt;character&gt;
    [1]      Unchanged        ENSG00000291065
    [2]      Unchanged       FXYD6P1; ZNF37BP
    [3]      Unchanged       FXYD6P1; ZNF37BP
    [4]      Unchanged ENSG00000285884; ZNF..
    [5]      Unchanged ENSG00000285884; ZNF..
    ...            ...                    ...
  [737]      Unchanged                ZFYVE27
  [738]      Unchanged    GOLGA7B; GOLGA7B-DT
  [739]      NA                       GOLGA7B
  [740]      Unchanged        CRTAC1; GOLGA7B
  [741]      Unchanged                R3HCC1L
  -------
  seqinfo: 1 sequence from hg19 genome</code></pre>
</div>
</div>
<aside class="notes">
<p>Missing values (i.e.&nbsp;undetected in one dataset) are simply left as <code>NA</code></p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="mapgrlcols-1" class="slide level2">
<h2>mapGrlCols</h2>
<ul>
<li>Columns from each element can be mapped
<ul>
<li>Columns can be combined and collapsed into a single column</li>
</ul></li>
<li>Makes finding sites changed for both comparisons simple</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>all_results <span class="sc">%&gt;%</span> </span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mapGrlCols</span>(<span class="at">var =</span> <span class="fu">c</span>(<span class="st">"logFC"</span>, <span class="st">"status"</span>), <span class="at">collapse =</span> <span class="st">"gene_name"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_detect</span>(ER_status, <span class="st">"(In|De)creased"</span>), <span class="fu">str_detect</span>(H3K27ac_status, <span class="st">"(In|De)creased"</span>)</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>  ) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 3 ranges and 5 metadata columns:
      seqnames            ranges strand |  ER_logFC H3K27ac_logFC ER_status
         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;     &lt;numeric&gt;  &lt;factor&gt;
  [1]    chr10 71879101-71881150      * |   1.20340      0.810052 Increased
  [2]    chr10 79623301-79635950      * |   0.96745      0.905969 Increased
  [3]    chr10 81101651-81103150      * |   1.90628      1.577851 Increased
      H3K27ac_status             gene_name
            &lt;factor&gt;           &lt;character&gt;
  [1]      Increased                 AIFM2
  [2]      Increased DLG5; ENSG00000204049
  [3]      Increased                  PPIF
  -------
  seqinfo: 1 sequence from hg19 genome</code></pre>
</div>
</div>
</section>

<section id="plotpairwise-logcpm" class="slide level2">
<h2>plotPairwise: logCPM</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPairwise</span>(all_results, <span class="at">var =</span> <span class="st">"logCPM"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Bioc2023_presentation_files/figure-revealjs/plot-pairwise-logcpm-1.png" width="960" class="r-stretch"><aside class="notes">
<p>Here we can see that there tends to be higher signal where both are detected</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="plotpairwise-logfc" class="slide level2">
<h2>plotPairwise: logFC</h2>

<img data-src="Bioc2023_presentation_files/figure-revealjs/plot-pairwise-logfc-1.png" class="r-stretch"></section>
<section id="plotpairwise-logfc-1" class="slide level2">
<h2>plotPairwise: logFC</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb80" data-code-line-numbers="2|5,9,10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1"></a><span class="fu">plotPairwise</span>(</span>
<span id="cb80-2"><a href="#cb80-2"></a>  all_results, <span class="at">var =</span> <span class="st">"logFC"</span>, <span class="at">colour =</span> <span class="st">"status"</span>, <span class="at">label =</span> <span class="st">"gene_name"</span></span>
<span id="cb80-3"><a href="#cb80-3"></a>) <span class="sc">+</span></span>
<span id="cb80-4"><a href="#cb80-4"></a>  <span class="fu">scale_colour_manual</span>(</span>
<span id="cb80-5"><a href="#cb80-5"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"lightskyblue"</span>, <span class="st">"palevioletred"</span>, <span class="st">"grey"</span>, <span class="st">"darkred"</span>)</span>
<span id="cb80-6"><a href="#cb80-6"></a>  ) <span class="sc">+</span></span>
<span id="cb80-7"><a href="#cb80-7"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb80-8"><a href="#cb80-8"></a>    <span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb80-9"><a href="#cb80-9"></a>      <span class="fu">setNames</span>(status_colours, <span class="fu">paste</span>(<span class="st">"ER"</span>, <span class="fu">names</span>(status_colours))),</span>
<span id="cb80-10"><a href="#cb80-10"></a>      <span class="fu">setNames</span>(status_colours, <span class="fu">paste</span>(<span class="st">"H3K27ac"</span>, <span class="fu">names</span>(status_colours)))</span>
<span id="cb80-11"><a href="#cb80-11"></a>    )</span>
<span id="cb80-12"><a href="#cb80-12"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Colouring by status can be informative</li>
<li>The furthest point from zero (per group) can be labelled</li>
<li>Point colours and boxplot fills can be set using <code>ggplot2</code> methods</li>
</ul>
<aside class="notes">
<p>Boxplots show us the behaviour in one based on the other, included sites where the alternative target is undetected</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="plothfgc" class="slide level2">
<h2>plotHFGC</h2>
<ul>
<li>Plots multiple tracks using Gviz <span class="citation" data-cites="gviz">(<a href="#/references" role="doc-biblioref" onclick="">Hahne and Ivanek 2016</a>)</span>
<ul>
<li><strong>H</strong>iC (or other interactions)</li>
<li><strong>F</strong>eatures of any type</li>
<li><strong>G</strong>ene models</li>
<li><strong>C</strong>overage</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>All tracks are optional</li>
<li>Cytogenetic bands for GRCh37/38 are included in the package</li>
</ul>
</div>
<div class="fragment">
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">"grch37.cytobands"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</section>
<section id="plothfgc-1" class="slide level2">
<h2>plotHFGC</h2>
<h3 id="hic-tracks">HiC tracks</h3>
<ul>
<li><code>GInteractions</code> object</li>
</ul>
<div class="fragment">
<!-- -->
<h3 id="feature-tracks">Feature tracks</h3>
<ul>
<li>A named <code>GRangesList</code> <span class="math inline">\(\implies\)</span> single track coloured by element
<ul>
<li>Output from <code>defineRegions()</code></li>
</ul></li>
<li>A named list of (named) <code>GRangesList</code> objects <span class="math inline">\(\implies\)</span> multiple tracks</li>
</ul>
</div>
<div class="fragment">
<!-- -->
<h3 id="gene-tracks">Gene Tracks</h3>
<ul>
<li><code>GRanges</code> object for a single track
<ul>
<li>Easily formed from exon-level annotations</li>
</ul></li>
<li>Named <code>GRangesList</code> will draw multiple tracks (Up, Down etc.)</li>
</ul>
</div>
<aside class="notes">
<p>Feature tracks may be the regions as defined, but can also add anything we like, such as chromHMM tracks, or known bindings sites, enahncers etc</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="plothfgc-2" class="slide level2">
<h2>plotHFGC</h2>
<h3 id="coverage-tracks">Coverage tracks</h3>
<ul>
<li>A <code>BigWigFileList</code> <span class="math inline">\(\implies\)</span> each file as a separate track</li>
<li>A list of <code>BigwigFileList</code> objects
<ul>
<li>Will overlay files within each <code>BigWigFileList</code></li>
<li>Each list element on a separate track</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Can also annotate coverage tracks (e.g.&nbsp;Increased, Decreased etc)</li>
</ul>
</div>
</section>


<section id="plothfgc-5" class="slide level2">
<h2>plotHFGC</h2>

<img data-src="Bioc2023_presentation_files/figure-revealjs/plotHFGC-1.png" class="r-stretch"></section>
<section id="plothfgc-6" class="slide level2">
<h2>plotHFGC</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>gr <span class="ot">&lt;-</span> all_results <span class="sc">%&gt;%</span> </span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mapGrlCols</span>(<span class="at">var =</span> <span class="st">"status"</span>, <span class="at">collapse =</span> <span class="st">"gene_name"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>    ER_status <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Increased"</span>, <span class="st">"Decreased"</span>), </span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>    H3K27ac_status <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Increased"</span>, <span class="st">"Decreased"</span>), </span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>    gene_name <span class="sc">==</span> <span class="st">"PPIF"</span></span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHFGC</span>(</span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a>  gr, <span class="at">cytobands =</span> grch37.cytobands,</span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> regions, <span class="at">featcol =</span> region_colours,</span>
<span id="cb86-11"><a href="#cb86-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes =</span> gene_models,</span>
<span id="cb86-12"><a href="#cb86-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">coverage =</span> cov_bwfl, <span class="at">linecol =</span> line_col, </span>
<span id="cb86-13"><a href="#cb86-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation =</span> cov_annot, <span class="at">annotcol =</span> status_colours,</span>
<span id="cb86-14"><a href="#cb86-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">zoom =</span> <span class="dv">20</span></span>
<span id="cb86-15"><a href="#cb86-15" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Once all objects defined:
<ul>
<li>Inspect multiple ranges quickly &amp; easily</li>
<li>Zoom in/out as needed</li>
</ul></li>
</ul>
</section></section>
<section id="references" class="title-slide slide level1 smaller scrollable" style="font-size: 80%">
<h1>References</h1>

<img src="assets/bodl_logo_white_background.jpg" class="slide-logo r-stretch"><div class="footer footer-default">

</div>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-edgeRQLF2016" class="csl-entry" role="listitem">
Chen, Yunshun, Aaron A T Lun, and Gordon K Smyth. 2016. <span>“From Reads to Genes to Pathways: Differential Expression Analysis of RNA-Seq Experiments Using Rsubread and the edgeR Quasi-Likelihood Pipeline.”</span> <em>F1000Research</em> 5: 1438. <a href="https://doi.org/10.12688/f1000research.8987.2">https://doi.org/10.12688/f1000research.8987.2</a>.
</div>
<div id="ref-gviz" class="csl-entry" role="listitem">
Hahne, Florian, and Robert Ivanek. 2016. <span>“Statistical Genomics: Methods and Protocols.”</span> In, edited by Ewy Mathé and Sean Davis, 335–51. New York, NY: Springer New York. <a href="https://doi.org/10.1007/978-1-4939-3578-9_16">https://doi.org/10.1007/978-1-4939-3578-9_16</a>.
</div>
<div id="ref-Hickey2021-mz" class="csl-entry" role="listitem">
Hickey, Theresa E, Luke A Selth, Kee Ming Chia, Geraldine Laven-Law, Heloisa H Milioli, Daniel Roden, Shalini Jindal, et al. 2021. <span>“The Androgen Receptor Is a Tumor Suppressor in Estrogen Receptor-Positive Breast Cancer.”</span> <em>Nat. Med.</em> 27 (2): 310–20.
</div>
<div id="ref-Law2014-xq" class="csl-entry" role="listitem">
Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. <span>“Voom: Precision Weights Unlock Linear Model Analysis Tools for <span class="nocase">RNA-seq</span> Read Counts.”</span> <em>Genome Biol.</em> 15 (2): R29.
</div>
<div id="ref-plyranges" class="csl-entry" role="listitem">
Lee, Stuart, Cook, Dianne, Lawrence, and Michael. 2019. <span>“Plyranges: A Grammar of Genomic Data Transformation.”</span> <em>Genome Biol.</em> 20 (1): 4. <a href="http://dx.doi.org/10.1186/s13059-018-1597-8">http://dx.doi.org/10.1186/s13059-018-1597-8</a>.
</div>
<div id="ref-csaw2016" class="csl-entry" role="listitem">
Lun, Aaron T L, and Gordon K Smyth. 2016. <span>“Csaw: A <span>B</span>ioconductor Package for Differential Binding Analysis of <span>C</span>h<span>I</span><span>P</span>-Seq Data Using Sliding Windows.”</span> <em>Nucleic Acids Res.</em> 44 (5): e45.
</div>
<div id="ref-McCarthy2009-qf" class="csl-entry" role="listitem">
McCarthy, Davis J, and Gordon K Smyth. 2009. <span>“Testing Significance Relative to a Fold-Change Threshold Is a <span>TREAT</span>.”</span> <em>Bioinformatics</em> 25 (6): 765–71.
</div>
<div id="ref-DiffBind2012" class="csl-entry" role="listitem">
Ross-Innes, Caryn S., Rory Stark, Andrew E. Teschendorff, Kelly A. Holmes, H. Raza Ali, Mark J. Dunning, Gordon D. Brown, et al. 2012. <span>“Differential Oestrogen Receptor Binding Is Associated with Clinical Outcome in Breast Cancer.”</span> <em>Nature</em> 481: –4. <a href="http://www.nature.com/nature/journal/v481/n7381/full/nature10730.html">http://www.nature.com/nature/journal/v481/n7381/full/nature10730.html</a>.
</div>
<div id="ref-tidyverse2019" class="csl-entry" role="listitem">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. <span>“Welcome to the <span class="nocase">tidyverse</span>.”</span> <em>Journal of Open Source Software</em> 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.
</div>
<div id="ref-Wilson2019-ln" class="csl-entry" role="listitem">
Wilson, Daniel J. 2019. <span>“The Harmonic Mean <em>p</em>-Value for Combining Dependent Tests.”</span> <em>Proc. Natl. Acad. Sci. U. S. A.</em> 116 (4): 1195–1200.
</div>
</div>
</section>


    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="Bioc2023_presentation_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="Bioc2023_presentation_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="Bioc2023_presentation_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="Bioc2023_presentation_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="Bioc2023_presentation_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="Bioc2023_presentation_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="Bioc2023_presentation_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="Bioc2023_presentation_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="Bioc2023_presentation_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="Bioc2023_presentation_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1024,

        height: 768,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>