---
title: "BiocAsia 2024"
subtitle: "transmogR: A Bioconductor package to enable the easy<br>incorporation of variants into a reference transcriptome"
author: "Dr Stevie Pederson (They/Them)<br>Black Ochre Data Labs<br>The Kids Research Institute Australia / ANU"
date: "2024-11-08"
date-format: long
title-slide-attributes:
    data-background-color: "#3D3D40"
    data-background-opacity: "70%"
    data-background-size: "20%"
    data-background-position: "95% 10%"
    data-background-image: "https://smped.github.io/transmogR/reference/figures/transmogR.png"
    data-notes: "Incredibly grateful to the BODL team, our Indigenous Governance committee and all sutdy participants for allowing me to work on this project"
editor: source
format: 
  revealjs:
    bibliography: references.bib
    theme: [bodl.scss]
    code-line-numbers: false
    width: 1280
    height: 720
    sansfont: Times New Roman
    logo: assets/bodl_logo_white_background.jpg
    slide-number: c/t
include-after: |
  <script type="text/javascript">
    Reveal.on('ready', event => {
      if (event.indexh === 0) {
        document.querySelector("div.has-logo > img.slide-logo").style.display = "none";
      }
    });
    Reveal.addEventListener('slidechanged', (event) => {
      if (event.indexh === 0) {
        Reveal.configure({ slideNumber: null });
        document.querySelector("div.has-logo > img.slide-logo").style.display = "none";
      }
      if (event.indexh === 1) { 
        Reveal.configure({ slideNumber: 'c' });
        document.querySelector("div.has-logo > img.slide-logo").style.display = null;
      }
    });
  </script>    
knitr: 
  opts_chunk: 
    echo: true
    include: true
    warning: false
    message: false
    fig.align: left
    fig.width: 8
    fig.heght: 6
---

## The PROPHECY Study

- Currently leading the transcriptomic analysis for the PROPHECY study
    + **P**reventing **R**enal, **OP**thalmic and **H**eart **E**vents in **C**ommunit**Y**
- Study with ~1350 SA-based Aboriginal participants
    + Genomic Variants, DNA Methylation, Transcriptomics, Proteomics, Lipidomics, Metabolomics
    + Inflammatory Marker Panel, Retinal Scans, ECG, Urine Samples
    + $>$ 1000 metadata categories
- Will be the largest existing indigenous dataset when collected

::: {.notes}
- All omics layers are taken from Whole Blood Samples
:::

## Type II Diabetes and Complications

:::: {style="font-size:90%;"}

:::: {.columns}

::: {.column width='45%'}

For Aboriginal Australians, Type II Diabetes rates are extremely high with younger onset[^1]

[^1]: https://www.indigenoushpf.gov.au

![](assets/t2d_prevalence.png){.absolute bottom=100 left=0 width="600"}

:::

::: {.column width='5%'}

:::

::: {.column width='50%'}

::: {.fragment fragment-index=1}

Similar for Cardiovascular & Chronic Kidney Disease

![](assets/cvd_prevalence.png){.absolute top=110 right=100 width="450"}


![](assets/ckd_prevalence.png){.absolute bottom=0 right=100 width="450"}
:::

:::

::::

:::

::: {.notes}
- Both CKD and CVD can be complications from Type II Diabetes
- This increase in disease burden translates to increased mortality, reduced life expectancy, an increased reliance on the health system and lower quality of life
:::

## Motivation

- Source of this risk remains unknown
    + After controlling for known external factors $\implies$ still highly increased risk

. . .    
    
- Is there a genetic component?
    + Regulatory regions, Coding Changes, Structural Variation, etc?
- Is there an epigenetic component?
    + DNA Methylation

. . .

**If genetic risk $\implies$ will we be able to see it using GRCh38?**

. . .

<br>
Pangenome approaches for transcriptomics are still too immature


::: {.notes}
- Alex, Kim etc setup in consultation with community
:::

## Variation In Aboriginal Australians

::: {.columns}

::: {.column width="37%"}

:::: {style="font-size:90%"}

- About 25% of genetic variation in Aboriginal Australians is unique [@Easteal2020-fo]<br><br>

::: {.incremental}

- Can we incorporate into transcriptomic analysis?<br><br>
- STARconsensus [@Kaminow2022-dz] for genomic alignments
    + gene-level counts
    + transcript-level analysis?
    + Does it matter?

:::

::::

:::

::: {.column width="63%"}

![*Consensus variants* found in >50% of unrelated individuals within a population, using the 1000GP AFR population members and the complete 1000GP dataset (panhuman) in comparison to PROPHECY participants](assets/ol-all-var-1.png)

:::
    
:::

## transmogR

![](https://smped.github.io/transmogR/reference/figures/transmogR.png){.absolute top=40 right=0 width="200"}

- Able to produce a variant-modified reference transcriptome
    + Using a transcriptomic reference $\implies$ don't need co-ordinates
    + STARconsensus returns co-ordinates relative to unmodified reference
    
. . .    
    
- Can also create a variant-modified reference genome
    + Used by `salmon` as decoy sequences [@Srivastava2020-tm]
    + *Not used* for any coordinate-based information
    
. . .
    
- This allows for quantification using `salmon` (or `kallisto`)
    + Overdispersion estimates returned for transcript-level analysis [@Baldoni2024-zf]
    
::: {.notes}
Enables creation of raw fasta files $\implies$ still need to index as per normal
:::

## Main Functions

![](assets/calvin.png){.absolute top=0 right=30 width="240"}

- `transmogrify()`: Create variant-modified transcriptome sequences
    + Add optional tags to modified transcript names
    
. . .
    
- `genomogrify()`: Create variant-modified genome sequences
    + Add optional tags
    
. . .
    
- `digestSalmon()`:
    + Returns a `SummarizedExperiment` with assays `counts`, `scaledCounts`, `TPM`, `effectiveLength` 
    + Overdispersions & transcript lengths returned in `rowData`
    + Can handle divergent references, e.g. parY-masked + chrY-masked 
    

## Creating a New Reference

- Requires a VCF with variants
    + Or a GRanges object with `ALT/REF`
    + Only tested with SNPs + InDels <50bp
- Transcripts from GTF or `exonsBy()`
- Genome from fasta or `BSgenome`
- Takes about 25-30min
    + Need to improve parallelisation
    + Can be RAM intensive $\implies$ generally run on a server/HPC
- Output Fasta files using `Biostrings::writeXStringSet()`

## Does It Make Any Difference?

- Only tested using a haploid transcriptome
- Pilot Analysis using 6 participants
    + Compared against unmodified GRCh38 (as worst case)
    + Compared against personalised GRCh38 as 'best-case' reference
- Ran STARconsensus + transmogR/salmon
    + Observations from STARconsensus [@Kaminow2022-dz] replicated
    + PROPHECY closest to personalised $\implies$ GRCh38 least similar
    
. . .    

- Also tested using cohort-level data subset (93 participants)
    + Sequencing focussed on CKD, Prof Sam El Osta group (Baker Heart & Diabetes Institute)
    + Using the PROPHECY consensus variants vs GRCh38 (chrY/parY-masked)

## Technical Assessment: 6x Pilot Samples

::: {.columns}

::: {.column width="40%"}
- Change is shown relative to personalised references
- Only a small number of fragments at the summary level
- PROPHECY consistently closest to personalised in
    1. Assigned Fragments
    2. Concordant Fragments
    3. Correct Orientation (ISF)
:::

::: {.column width="60%"}

![](assets/boxplot-lib-format-1.png)
:::

:::

## Technical Assessment: 6x Pilot Samples

::: {.columns}

::: {.column width="35%"}
- Looking at reads moving between genes
- 1 Changed alignment $\neq$ 1 changed count
- Far more divergent than summary statistics indicate
- Lots of change in HLA transcripts <br>$\implies$ stand-alone analysis
:::

::: {.column width="65%"}

![](assets/salmon-moved-across-1.png)
:::

:::

::: {.notes}
- Alignments were taken from bam files output by salmon
- 1 changed alignment $\neq$ 1 changed count due to multi-mapping and model fitting
:::

## Technical Assessment: 6x Pilot Samples {visibility='hidden'}

::: {.columns}

::: {.column width="35%"}
- Comparing transcript-level expression estimates against personalised references
    + Using scaled counts [@Baldoni2024-zf]
- Baseline expression estimates will vary
    + Impacted by raw counts and overdispersions

:::

::: {.column width="65%"}

![](assets/compare-trans-lfc-1.png)
:::

:::

## Cohort Level Analysis

:::: {style="font-size:85%;"}

::: {.columns}

::: {.column width="35%"}

- log2-ratio of scaled counts within each participant
    + GRCh38 vs PROPHECY-modified
    + Showing average change (Mean logFC) vs variation (SD logFC)
    + Also baseline logCPM (from GRCh38)
- Some transcripts become far more variable (x $\approx$ 0;  y $\uparrow$)
- Some become consistently higher/lower in signal estimates

:::

::: {.column width="65%"}

![](assets/lfc-sd-scaled-1.png)

:::

:::

::::


## Change in Gene-Ranks

::: {.columns}

::: {.column}
- DGE Ranks: T2D vs No T2D
- Most genes are consistent
- Some do change ranks quite significantly
:::

::: {.column}
![](assets/compare-dge-pval-full-font14.png)
:::

:::

## Change in Transcript-Ranks

::: {.columns}

::: {.column}
- DTE Ranks: T2D vs No T2D
- Most transcripts are also consistent
- Some do change ranks quite significantly
:::

::: {.column}
![](assets/compare-dte-pval-full-font14.png)
:::

:::



## Acknowledgements

<br>

:::: {style="font-size:55%;"}

::: {.columns}

::: {.column width="25%"}
### Black Ochre Data Labs
- Alex Brown
- **Jimmy Breen**
- Alastair Ludington
- **Yassine Souilmi**
- **Sam Godwin**
- Liza Kretzschmar
- Sam Buckberry
- Holly Massacci
- Katharine Brown
- Adam Heterick
- Justine Clark
- Amanda Richards-Satour
- Bastien Llamas
- Mary Brushe
- Sarah Munns
- Rose Senesi
- Kaashifah Bruce
:::

::: {.column width="28%"}
### Baker Heart & Diabetes Institute
- Sam El-Osta
- Ishant Khurana
- Scott Maxwell
- Moshe Olshansky

<br>

### National Centre for Indigenous Genomics
- Hardip Patel
- Azure Hermes

<br>

### SAHMRI / Wardliparingga
- Natasha Howard
- Marlie Frank
- Odette Pearson
- Kim Morey

:::

::: {.column width="25%"}
### ALIGN
- Johanna Barclay
- Annalee Stearne
- Louise Lyons<br>

### University of Sydney
- Jean Yang

### SAGC
- Paul Wang
- John Salamon
- Sen Wang
- Renee Smith

### Victor Chang
- Jason Kovacic

:::

:::

::::

![](assets/bodl_logo_white_background.jpg){.absolute top=100 right=0 width="300"}

![](assets/TheKids-RGB_Logo_Colour-Pattern.png){.absolute top=250 right=0 width="300"}

![](assets/ANU_Primary_Horizontal_GoldBlack.jpg){.absolute top=500 right=20 width="280"}


## References
