<!DOCTYPE html>
<html lang="en"><head>
<script src="20221201_extraChIPs_files/libs/clipboard/clipboard.min.js"></script>
<script src="20221201_extraChIPs_files/libs/quarto-html/tabby.min.js"></script>
<script src="20221201_extraChIPs_files/libs/quarto-html/popper.min.js"></script>
<script src="20221201_extraChIPs_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="20221201_extraChIPs_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="20221201_extraChIPs_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="20221201_extraChIPs_files/libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.1.189">

  <meta name="author" content="Stephen (Stevie) Pederson">
  <meta name="dcterms.date" content="2022-12-01">
  <title>Using extraChIPs with GRAVI</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="20221201_extraChIPs_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="20221201_extraChIPs_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
      }
    pre.numberSource { margin-left: 3em;  padding-left: 4px; }
    div.sourceCode
      { color: #f8f8f2;  }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #f8f8f2; } /* Normal */
    code span.al { color: #f07178; background-color: #2a0f15; font-weight: bold; } /* Alert */
    code span.an { color: #d4d0ab; } /* Annotation */
    code span.at { color: #00e0e0; } /* Attribute */
    code span.bn { color: #d4d0ab; } /* BaseN */
    code span.bu { color: #abe338; } /* BuiltIn */
    code span.cf { color: #ffa07a; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #abe338; } /* Char */
    code span.cn { color: #ffd700; } /* Constant */
    code span.co { color: #f8f8f2; font-style: italic; } /* Comment */
    code span.cv { color: #ffd700; } /* CommentVar */
    code span.do { color: #f8f8f2; } /* Documentation */
    code span.dt { color: #ffa07a; } /* DataType */
    code span.dv { color: #d4d0ab; } /* DecVal */
    code span.er { color: #f07178; text-decoration: underline; } /* Error */
    code span.ex { color: #00e0e0; font-weight: bold; } /* Extension */
    code span.fl { color: #d4d0ab; } /* Float */
    code span.fu { color: #ffa07a; } /* Function */
    code span.im { color: #abe338; } /* Import */
    code span.in { color: #d4d0ab; } /* Information */
    code span.kw { color: #ffa07a; font-weight: bold; } /* Keyword */
    code span.op { color: #ffa07a; } /* Operator */
    code span.ot { color: #00e0e0; } /* Other */
    code span.pp { color: #dcc6e0; } /* Preprocessor */
    code span.re { color: #00e0e0; background-color: #f8f8f2; } /* RegionMarker */
    code span.sc { color: #abe338; } /* SpecialChar */
    code span.ss { color: #abe338; } /* SpecialString */
    code span.st { color: #abe338; } /* String */
    code span.va { color: #00e0e0; } /* Variable */
    code span.vs { color: #abe338; } /* VerbatimString */
    code span.wa { color: #dcc6e0; } /* Warning */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <link rel="stylesheet" href="20221201_extraChIPs_files/libs/revealjs/dist/theme/quarto.css" id="theme">
  <link href="20221201_extraChIPs_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="20221201_extraChIPs_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="20221201_extraChIPs_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="20221201_extraChIPs_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-captioned.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-captioned) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-captioned.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-captioned .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-captioned .callout-caption  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-captioned.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-captioned.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-caption {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-caption {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-caption {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-captioned .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-captioned .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-captioned) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-caption {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-caption {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-caption {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-caption {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-caption {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-dark">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="center">
  <h1 class="title">Using extraChIPs with GRAVI</h1>
  <p class="subtitle">Hot mess or slick analysis?</p>
  <p class="author">Stephen (Stevie) Pederson</p>
  <p class="institute">BiocAsia</p>
  <p class="date">2022-12-01</p>
</section>

<section>
<section id="introduction" class="title-slide slide level1 center">
<h1>Introduction</h1>
<div class="cell">

</div>
</section>
<section id="who-am-i" class="slide level2">
<h2>Who Am I?</h2>
<ul>
<li>Post-doctoral Bioinformatician (ECR)
<ul>
<li>Black Ochre Data Laboratories &nbsp; <img data-src="tki.jpg" width="80"></li>
<li>Based in Adelaide, South Australia</li>
<li>Traditional lands of the Kaurna people</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Author/Maintainer of <code>ngsReports</code> &nbsp; <img data-src="https://github.com/steveped/stickers/raw/main/alt_ngsReports_sticker.png" width="70">
<ul>
<li>Parse &amp; plot <code>FastQC</code>, <code>cutadapt</code>, <code>STAR</code>, <code>macs2</code> etc.</li>
</ul></li>
</ul>
</div>
</section>
<section id="gravi" class="slide level2">
<h2>GRAVI</h2>
<ul>
<li>2020 - 2022: Dame Roma Mitchell Cancer Research Laboratories (DRMCRL)</li>
<li>Activation of the Androgen Receptor (AR) as a therapeutic strategy for Breast Cancer <span class="citation" data-cites="Hickey2021">(<a href="#/references" role="doc-biblioref" onclick="">Hickey et al. 2021</a>)</span></li>
<li>Combined genomic changes to AR, ER &amp; GATA3 binding after DHT-treatment (ChIP-Seq)
<ul>
<li>4 cell lines + 2 PDX model</li>
<li>Additional Histone Marks</li>
<li>Additional HiC / HiChIP Data (Sometimes)</li>
<li>RNA Seq data</li>
</ul></li>
</ul>
<!-- ## GRAVI: Datasets -->
<!-- ![](DRMCRL_datasets.jpg) -->
</section>
<section id="gravi-1" class="slide level2">
<h2>GRAVI</h2>
<ul>
<li>Gene Regulatory Analysis using Variable Inputs (https://github.com/steveped/GRAVI)</li>
<li>snakemake workflow
<ul>
<li>HTML Output inspired by <code>workflowr</code> <span class="citation" data-cites="workflowr">(<a href="#/references" role="doc-biblioref" onclick="">Blischak, Carbonetto, and Stephens 2019</a>)</span></li>
<li>All code visible <span class="math inline">\(\implies\)</span> Reproducible as complete or in sections</li>
<li>Standardised all output formats/structures <!-- - Performs 1) Custom Annotations, 2) Peak Calling, 3) Differential Binding --> <!--   + Enrichment Analysis performed during multiple steps --> <!--   + All `R` Environments saved (marked as `temp()`) --></li>
</ul></li>
</ul>
</section>
<section id="gravi-2" class="slide level2">
<h2>GRAVI</h2>
<ul>
<li>Minimal Data:
<ul>
<li>1 ChIP Target across two conditions</li>
</ul></li>
<li>Best for 2-3 ChIP targets with 2-3 conditions
<ul>
<li>Enables Pair-wise Comparisons</li>
<li>Optional Data: RNA-Seq Results, HiC Interactions, Genomic Features (H3K27ac)</li>
</ul></li>
</ul>
</section></section>
<section>
<section id="extrachips" class="title-slide slide level1 center">
<h1>extraChIPs</h1>

</section>
<section id="gravi-3" class="slide level2">
<h2>GRAVI</h2>
<ul>
<li><code>extraChIPs</code> was developed as key infrastructure for GRAVI</li>
<li>3 Primary sets of Functions
<ol type="1">
<li>Working with <code>GRanges</code> focussed on <code>mcols()</code></li>
<li>ChIP-Seq Helper Functions (i.e.&nbsp;peaks and differential binding)</li>
<li>Data Visualisation</li>
</ol></li>
</ul>
</section>
<section id="working-with-ranges-tibble-coercion" class="slide level2">
<h2>Working With Ranges: <code>tibble</code> Coercion</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="do">## S3 method for class 'DataFrame'</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">as_tibble</span>(x, <span class="at">rangeAsChar =</span> <span class="cn">TRUE</span>, ...)</span>
<span id="cb1-3"><a href="#cb1-3"></a></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="do">## S3 method for class 'GenomicRanges'</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">as_tibble</span>(x, <span class="at">rangeAsChar =</span> <span class="cn">TRUE</span>, <span class="at">name =</span> <span class="st">"range"</span>, ...)</span>
<span id="cb1-6"><a href="#cb1-6"></a></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="do">## S3 method for class 'Seqinfo'</span></span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">as_tibble</span>(x, ...)</span>
<span id="cb1-9"><a href="#cb1-9"></a></span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="do">## S3 method for class 'GInteractions'</span></span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="fu">as_tibble</span>(x, <span class="at">rangeAsChar =</span> <span class="cn">TRUE</span>, <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">".x"</span>, <span class="st">".y"</span>), ...)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br></p>
<ul>
<li>Defined for <code>GRanges</code>, <code>DataFrame</code>, <code>Seqinfo</code> and <code>GInteractions</code> classes</li>
<li>Handles <code>S4</code> Compressed list columns well (so far)
<ul>
<li>Uses <code>vctrs::vec_proxy()</code> to coerce to S3 lists</li>
</ul></li>
</ul>
</section>
<section id="working-with-ranges-tibble-coercion-1" class="slide level2">
<h2>Working With Ranges: <code>tibble</code> Coercion</h2>
<div class="cell">

</div>
<p>Starting with the protein-coding transcripts for <em>CTLA4</em></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>gr</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 4 ranges and 2 metadata columns:
      seqnames              ranges strand |   gene_name transcript_name
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;     &lt;character&gt;
  [1]     chr2 204732494-204738688      + |       CTLA4       CTLA4-205
  [2]     chr2 204732666-204737498      + |       CTLA4       CTLA4-201
  [3]     chr2 204732666-204737535      + |       CTLA4       CTLA4-204
  [4]     chr2 204732666-204737535      + |       CTLA4       CTLA4-203
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</section>
<section id="working-with-ranges-tibble-coercion-2" class="slide level2">
<h2>Working With Ranges: <code>tibble</code> Coercion</h2>
<p>Now perform the coercion using <code>as_tibble()</code></p>
<div class="cell columns column-output-location">
<div class="column">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="fu">library</span>(extraChIPs)</span>
<span id="cb4-2"><a href="#cb4-2"></a>tbl <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(gr)</span>
<span id="cb4-3"><a href="#cb4-3"></a>tbl</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div><div class="column">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4 × 3
  range                      gene_name transcript_name
  &lt;chr&gt;                      &lt;chr&gt;     &lt;chr&gt;          
1 chr2:204732494-204738688:+ CTLA4     CTLA4-205      
2 chr2:204732666-204737498:+ CTLA4     CTLA4-201      
3 chr2:204732666-204737535:+ CTLA4     CTLA4-204      
4 chr2:204732666-204737535:+ CTLA4     CTLA4-203      </code></pre>
</div>
</div>
</div>
<div class="fragment" data-fragment-index="1">
<p>Coerce back to a <code>GRanges</code> uses <code>colToRanges()</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>sq <span class="ot">&lt;-</span> <span class="fu">seqinfo</span>(gr)</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="fu">colToRanges</span>(tbl, <span class="at">var =</span> <span class="st">"range"</span>, <span class="at">seqinfo =</span> sq)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 4 ranges and 2 metadata columns:
      seqnames              ranges strand |   gene_name transcript_name
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;     &lt;character&gt;
  [1]     chr2 204732494-204738688      + |       CTLA4       CTLA4-205
  [2]     chr2 204732666-204737498      + |       CTLA4       CTLA4-201
  [3]     chr2 204732666-204737535      + |       CTLA4       CTLA4-204
  [4]     chr2 204732666-204737535      + |       CTLA4       CTLA4-203
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</div>
</section>
<section id="operations-retaining-mcols-reducemc" class="slide level2">
<h2>Operations Retaining <code>mcols</code>: <code>reduceMC()</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>gr</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 4 ranges and 2 metadata columns:
      seqnames              ranges strand |   gene_name transcript_name
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;     &lt;character&gt;
  [1]     chr2 204732494-204738688      + |       CTLA4       CTLA4-205
  [2]     chr2 204732666-204737498      + |       CTLA4       CTLA4-201
  [3]     chr2 204732666-204737535      + |       CTLA4       CTLA4-204
  [4]     chr2 204732666-204737535      + |       CTLA4       CTLA4-203
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
<p>Common functions like <code>reduce</code> lose <code>mcols</code> information</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="fu">reduce</span>(gr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 1 range and 0 metadata columns:
      seqnames              ranges strand
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;
  [1]     chr2 204732494-204738688      +
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</section>
<section id="operations-retaining-mcols-reducemc-1" class="slide level2">
<h2>Operations Retaining <code>mcols</code>: <code>reduceMC()</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>gr</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 4 ranges and 2 metadata columns:
      seqnames              ranges strand |   gene_name transcript_name
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;     &lt;character&gt;
  [1]     chr2 204732494-204738688      + |       CTLA4       CTLA4-205
  [2]     chr2 204732666-204737498      + |       CTLA4       CTLA4-201
  [3]     chr2 204732666-204737535      + |       CTLA4       CTLA4-204
  [4]     chr2 204732666-204737535      + |       CTLA4       CTLA4-203
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
<p><code>reduceMC()</code> collapses the <code>mcols</code> information</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="fu">reduceMC</span>(gr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 1 range and 2 metadata columns:
      seqnames              ranges strand |   gene_name
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
  [1]     chr2 204732494-204738688      + |       CTLA4
                        transcript_name
                        &lt;CharacterList&gt;
  [1] CTLA4-205,CTLA4-201,CTLA4-204,...
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</section>
<section id="operations-retaining-mcols-reducemc-2" class="slide level2">
<h2>Operations Retaining <code>mcols</code>: <code>reduceMC()</code></h2>
<p>More realistically we might want to find unique TSS within a gene</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb16-2"><a href="#cb16-2"></a>gr <span class="sc">%&gt;%</span> </span>
<span id="cb16-3"><a href="#cb16-3"></a>  <span class="fu">resize</span>(<span class="at">width =</span> <span class="dv">1</span>, <span class="at">fix =</span> <span class="st">'start'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-4"><a href="#cb16-4"></a>  <span class="fu">reduceMC</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 2 ranges and 2 metadata columns:
      seqnames    ranges strand |   gene_name               transcript_name
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;               &lt;CharacterList&gt;
  [1]     chr2 204732494      + |       CTLA4                     CTLA4-205
  [2]     chr2 204732666      + |       CTLA4 CTLA4-201,CTLA4-204,CTLA4-203
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</section>
<section id="set-operations-retaining-mcols" class="slide level2">
<h2>Set Operations Retaining <code>mcols</code></h2>
<ul>
<li>Set operations have <code>*MC()</code> versions</li>
<li><code>setdiff()</code> Vs <code>setdiffMC()</code></li>
<li><code>union</code> Vs <code>unionMC()</code></li>
<li><code>intersect</code> Vs <code>intersectMC()</code></li>
</ul>
<div class="fragment">
<p><code>mcols</code> are only retained from the query range (i.e.&nbsp;<code>x</code> in <code>setdiffMC(x, y)</code>)</p>
</div>
</section>
<section id="tidyverse-inspired-functions-chopmc" class="slide level2">
<h2>Tidyverse-inspired Functions: <code>chopMC()</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>gr</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 4 ranges and 2 metadata columns:
      seqnames              ranges strand |   gene_name transcript_name
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;     &lt;character&gt;
  [1]     chr2 204732494-204738688      + |       CTLA4       CTLA4-205
  [2]     chr2 204732666-204737498      + |       CTLA4       CTLA4-201
  [3]     chr2 204732666-204737535      + |       CTLA4       CTLA4-204
  [4]     chr2 204732666-204737535      + |       CTLA4       CTLA4-203
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
<div class="fragment">
<ul>
<li><code>chopMC()</code> acts like <code>chop()</code> using the range as the backbone</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="fu">chopMC</span>(gr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 3 ranges and 2 metadata columns:
      seqnames              ranges strand |   gene_name     transcript_name
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;     &lt;CharacterList&gt;
  [1]     chr2 204732494-204738688      + |       CTLA4           CTLA4-205
  [2]     chr2 204732666-204737498      + |       CTLA4           CTLA4-201
  [3]     chr2 204732666-204737535      + |       CTLA4 CTLA4-204,CTLA4-203
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
</div>
</section>
<section id="tidyverse-inspired-functions-distinctmc" class="slide level2">
<h2>Tidyverse-inspired Functions: <code>distinctMC()</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>gr</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 4 ranges and 2 metadata columns:
      seqnames              ranges strand |   gene_name transcript_name
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;     &lt;character&gt;
  [1]     chr2 204732494-204738688      + |       CTLA4       CTLA4-205
  [2]     chr2 204732666-204737498      + |       CTLA4       CTLA4-201
  [3]     chr2 204732666-204737535      + |       CTLA4       CTLA4-204
  [4]     chr2 204732666-204737535      + |       CTLA4       CTLA4-203
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
<ul>
<li><code>distinctMC()</code> behaves like <code>distinct()</code> using the ranges + any requested columns</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="fu">distinctMC</span>(gr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 3 ranges and 0 metadata columns:
      seqnames              ranges strand
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;
  [1]     chr2 204732494-204738688      +
  [2]     chr2 204732666-204737498      +
  [3]     chr2 204732666-204737535      +
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</section>
<section id="tidyverse-inspired-functions-distinctmc-1" class="slide level2">
<h2>Tidyverse-inspired Functions: <code>distinctMC()</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>gr</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 4 ranges and 2 metadata columns:
      seqnames              ranges strand |   gene_name transcript_name
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;     &lt;character&gt;
  [1]     chr2 204732494-204738688      + |       CTLA4       CTLA4-205
  [2]     chr2 204732666-204737498      + |       CTLA4       CTLA4-201
  [3]     chr2 204732666-204737535      + |       CTLA4       CTLA4-204
  [4]     chr2 204732666-204737535      + |       CTLA4       CTLA4-203
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
<ul>
<li><code>distinctMC()</code> behaves like <code>distinct()</code> using the ranges + any requested columns</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a><span class="fu">distinctMC</span>(gr, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 3 ranges and 2 metadata columns:
      seqnames              ranges strand |   gene_name transcript_name
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;     &lt;character&gt;
  [1]     chr2 204732494-204738688      + |       CTLA4       CTLA4-205
  [2]     chr2 204732666-204737498      + |       CTLA4       CTLA4-201
  [3]     chr2 204732666-204737535      + |       CTLA4       CTLA4-204
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</section>
<section id="additional-helpers" class="slide level2">
<h2>Additional Helpers</h2>
<p><code>stitchRanges()</code> joins ranges setting barriers</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a>x <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="fu">c</span>(<span class="st">"chr1:1-10"</span>, <span class="st">"chr1:101-110"</span>, <span class="st">"chr1:201-210"</span>))</span>
<span id="cb30-2"><a href="#cb30-2"></a>y <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="st">"chr1:150-200:+"</span>)</span>
<span id="cb30-3"><a href="#cb30-3"></a>x</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 3 ranges and 0 metadata columns:
      seqnames    ranges strand
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
  [1]     chr1      1-10      *
  [2]     chr1   101-110      *
  [3]     chr1   201-210      *
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<p><br></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a><span class="fu">stitchRanges</span>(x, <span class="at">exclude =</span> y, <span class="at">maxgap =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 2 ranges and 0 metadata columns:
      seqnames    ranges strand
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
  [1]     chr1     1-110      *
  [2]     chr1   201-210      *
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
</section></section>
<section>
<section id="working-with-peaks" class="title-slide slide level1 center">
<h1>Working with Peaks</h1>

</section>
<section id="importpeaks" class="slide level2">
<h2><code>importPeaks()</code></h2>
<p>Most often, we’ll load in replicates with identical structure <span class="math inline">\(\implies\)</span> <code>GRangesList</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb34-2"><a href="#cb34-2"></a>fl <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"GATA3_E2_rep{1:2}.narrowPeak"</span>)</span>
<span id="cb34-3"><a href="#cb34-3"></a>peaks <span class="ot">&lt;-</span> <span class="fu">importPeaks</span>(fl, <span class="at">seqinfo =</span> sq)</span>
<span id="cb34-4"><a href="#cb34-4"></a>peaks</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRangesList object of length 2:
$GATA3_E2_rep1.narrowPeak
GRanges object with 10 ranges and 5 metadata columns:
                             seqnames          ranges strand |     score
                                &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
   ZR-75-1_GATA3_E2_1_peak_1     chr1   523658-523929      * |       174
   ZR-75-1_GATA3_E2_1_peak_2     chr1   780928-781330      * |       373
   ZR-75-1_GATA3_E2_1_peak_3     chr1   821067-821391      * |        71
   ZR-75-1_GATA3_E2_1_peak_4     chr1   849166-849528      * |       232
   ZR-75-1_GATA3_E2_1_peak_5     chr1   852773-852955      * |       112
   ZR-75-1_GATA3_E2_1_peak_6     chr1 1707118-1707445      * |       170
   ZR-75-1_GATA3_E2_1_peak_7     chr1 1790623-1790995      * |       175
   ZR-75-1_GATA3_E2_1_peak_8     chr1 1800979-1801188      * |        70
   ZR-75-1_GATA3_E2_1_peak_9     chr1 1814836-1815115      * |        80
  ZR-75-1_GATA3_E2_1_peak_10     chr1 1933329-1933586      * |       117
                             signalValue    pValue    qValue      peak
                               &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
   ZR-75-1_GATA3_E2_1_peak_1     7.90069  20.38453  17.44608       134
   ZR-75-1_GATA3_E2_1_peak_2    12.49110  40.68121  37.35409       209
   ZR-75-1_GATA3_E2_1_peak_3     4.24703   9.79599   7.18429       157
   ZR-75-1_GATA3_E2_1_peak_4    12.30190  26.35153  23.27520       189
   ZR-75-1_GATA3_E2_1_peak_5     7.84787  13.98412  11.22576       118
   ZR-75-1_GATA3_E2_1_peak_6    10.07489  19.93628  17.00446       168
   ZR-75-1_GATA3_E2_1_peak_7     9.65966  20.51598  17.57464       200
   ZR-75-1_GATA3_E2_1_peak_8     6.36320   9.63799   7.02867        97
   ZR-75-1_GATA3_E2_1_peak_9     6.73437  10.68456   8.03742       151
  ZR-75-1_GATA3_E2_1_peak_10     6.37055  14.51252  11.73312       133
  -------
  seqinfo: 24 sequences from GRCh37 genome

$GATA3_E2_rep2.narrowPeak
GRanges object with 10 ranges and 5 metadata columns:
                             seqnames          ranges strand |     score
                                &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
   ZR-75-1_GATA3_E2_2_peak_1     chr1   523672-523939      * |        73
   ZR-75-1_GATA3_E2_2_peak_2     chr1   780837-781294      * |       310
   ZR-75-1_GATA3_E2_2_peak_3     chr1   821066-821346      * |        17
   ZR-75-1_GATA3_E2_2_peak_4     chr1   849167-849522      * |       352
   ZR-75-1_GATA3_E2_2_peak_5     chr1 1707074-1707325      * |        32
   ZR-75-1_GATA3_E2_2_peak_6     chr1 1790623-1791019      * |       173
   ZR-75-1_GATA3_E2_2_peak_7     chr1 1800936-1801227      * |       121
   ZR-75-1_GATA3_E2_2_peak_8     chr1 1814913-1815157      * |       121
   ZR-75-1_GATA3_E2_2_peak_9     chr1 1933372-1933563      * |        42
  ZR-75-1_GATA3_E2_2_peak_10     chr1 2462419-2462660      * |        65
                             signalValue    pValue    qValue      peak
                               &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
   ZR-75-1_GATA3_E2_2_peak_1     5.12315  10.08017   7.30926       136
   ZR-75-1_GATA3_E2_2_peak_2    14.67066  34.47041  31.01986       308
   ZR-75-1_GATA3_E2_2_peak_3     2.82475   4.16110   1.76187       148
   ZR-75-1_GATA3_E2_2_peak_4    15.79546  38.73673  35.20665       176
   ZR-75-1_GATA3_E2_2_peak_5     4.54744   5.74956   3.22346       143
   ZR-75-1_GATA3_E2_2_peak_6     9.75019  20.46024  17.32872       232
   ZR-75-1_GATA3_E2_2_peak_7     8.29678  15.09020  12.11843       168
   ZR-75-1_GATA3_E2_2_peak_8     8.29678  15.09020  12.11843       146
   ZR-75-1_GATA3_E2_2_peak_9     4.23712   6.90630   4.29843        90
  ZR-75-1_GATA3_E2_2_peak_10     4.90164   9.25272   6.52910       112
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</section>
<section id="plotoverlaps" class="slide level2">
<h2><code>plotOverlaps()</code></h2>
<div class="cell columns column-output-location">
<div class="column">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb36-2"><a href="#cb36-2"></a><span class="fu">names</span>(peaks) <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(</span>
<span id="cb36-3"><a href="#cb36-3"></a>  <span class="fu">names</span>(peaks), <span class="st">".narrowPeak"</span></span>
<span id="cb36-4"><a href="#cb36-4"></a>)</span>
<span id="cb36-5"><a href="#cb36-5"></a><span class="fu">plotOverlaps</span>(</span>
<span id="cb36-6"><a href="#cb36-6"></a>  peaks, </span>
<span id="cb36-7"><a href="#cb36-7"></a>  <span class="at">set_col =</span> <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"yellow"</span>),</span>
<span id="cb36-8"><a href="#cb36-8"></a>  <span class="at">alpha =</span> <span class="fl">0.2</span></span>
<span id="cb36-9"><a href="#cb36-9"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div><div class="column">
<div class="cell-output-display">
<p><img data-src="20221201_extraChIPs_files/figure-revealjs/unnamed-chunk-20-1.png" width="576"></p>
</div>
</div>
</div>
</section>
<section id="plotoverlaps-1" class="slide level2">
<h2><code>plotOverlaps()</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a><span class="fu">plotOverlaps</span>(peaks, <span class="at">type =</span> <span class="st">"upset"</span>, <span class="at">var =</span> <span class="st">"score"</span>, <span class="at">set_col =</span> <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"yellow"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="20221201_extraChIPs_files/figure-revealjs/unnamed-chunk-21-1.png" width="960" class="r-stretch"></section>
<section id="makeconsensus" class="slide level2 smaller">
<h2><code>makeConsensus()</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a><span class="fu">makeConsensus</span>(peaks)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 11 ranges and 3 metadata columns:
       seqnames          ranges strand | GATA3_E2_rep1 GATA3_E2_rep2         n
          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;logical&gt;     &lt;logical&gt; &lt;numeric&gt;
   [1]     chr1   523658-523939      * |          TRUE          TRUE         2
   [2]     chr1   780837-781330      * |          TRUE          TRUE         2
   [3]     chr1   821066-821391      * |          TRUE          TRUE         2
   [4]     chr1   849166-849528      * |          TRUE          TRUE         2
   [5]     chr1   852773-852955      * |          TRUE         FALSE         1
   [6]     chr1 1707074-1707445      * |          TRUE          TRUE         2
   [7]     chr1 1790623-1791019      * |          TRUE          TRUE         2
   [8]     chr1 1800936-1801227      * |          TRUE          TRUE         2
   [9]     chr1 1814836-1815157      * |          TRUE          TRUE         2
  [10]     chr1 1933329-1933586      * |          TRUE          TRUE         2
  [11]     chr1 2462419-2462660      * |         FALSE          TRUE         1
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</section>
<section id="makeconsensus-1" class="slide level2 smaller">
<h2><code>makeConsensus()</code></h2>
<p>Retain one or more columns from each replicate</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a><span class="fu">makeConsensus</span>(peaks, <span class="at">var =</span> <span class="st">"score"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 11 ranges and 4 metadata columns:
       seqnames          ranges strand |         score GATA3_E2_rep1
          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;NumericList&gt;     &lt;logical&gt;
   [1]     chr1   523658-523939      * |       174, 73          TRUE
   [2]     chr1   780837-781330      * |       373,310          TRUE
   [3]     chr1   821066-821391      * |         71,17          TRUE
   [4]     chr1   849166-849528      * |       232,352          TRUE
   [5]     chr1   852773-852955      * |           112          TRUE
   [6]     chr1 1707074-1707445      * |       170, 32          TRUE
   [7]     chr1 1790623-1791019      * |       175,173          TRUE
   [8]     chr1 1800936-1801227      * |        70,121          TRUE
   [9]     chr1 1814836-1815157      * |        80,121          TRUE
  [10]     chr1 1933329-1933586      * |       117, 42          TRUE
  [11]     chr1 2462419-2462660      * |            65         FALSE
       GATA3_E2_rep2         n
           &lt;logical&gt; &lt;numeric&gt;
   [1]          TRUE         2
   [2]          TRUE         2
   [3]          TRUE         2
   [4]          TRUE         2
   [5]         FALSE         1
   [6]          TRUE         2
   [7]          TRUE         2
   [8]          TRUE         2
   [9]          TRUE         2
  [10]          TRUE         2
  [11]          TRUE         1
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</section>
<section id="makeconsensus-2" class="slide level2 smaller">
<h2><code>makeConsensus()</code></h2>
<p>Example: Keep peak centres from each replicate</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a><span class="fu">library</span>(plyranges)</span>
<span id="cb42-2"><a href="#cb42-2"></a>peaks <span class="sc">%&gt;%</span> </span>
<span id="cb42-3"><a href="#cb42-3"></a>  <span class="fu">endoapply</span>(mutate, <span class="at">centre =</span> start <span class="sc">+</span> peak) <span class="sc">%&gt;%</span> </span>
<span id="cb42-4"><a href="#cb42-4"></a>  <span class="fu">makeConsensus</span>(<span class="at">var =</span> <span class="st">"centre"</span>, <span class="at">p =</span> <span class="dv">1</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 9 ranges and 4 metadata columns:
      seqnames          ranges strand |          centre GATA3_E2_rep1
         &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;NumericList&gt;     &lt;logical&gt;
  [1]     chr1   523658-523939      * |   523792,523808          TRUE
  [2]     chr1   780837-781330      * |   781137,781145          TRUE
  [3]     chr1   821066-821391      * |   821224,821214          TRUE
  [4]     chr1   849166-849528      * |   849355,849343          TRUE
  [5]     chr1 1707074-1707445      * | 1707286,1707217          TRUE
  [6]     chr1 1790623-1791019      * | 1790823,1790855          TRUE
  [7]     chr1 1800936-1801227      * | 1801076,1801104          TRUE
  [8]     chr1 1814836-1815157      * | 1814987,1815059          TRUE
  [9]     chr1 1933329-1933586      * | 1933462,1933462          TRUE
      GATA3_E2_rep2         n
          &lt;logical&gt; &lt;numeric&gt;
  [1]          TRUE         2
  [2]          TRUE         2
  [3]          TRUE         2
  [4]          TRUE         2
  [5]          TRUE         2
  [6]          TRUE         2
  [7]          TRUE         2
  [8]          TRUE         2
  [9]          TRUE         2
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</section>
<section id="makeconsensus-3" class="slide level2 smaller">
<h2><code>makeConsensus()</code></h2>
<p>Example: Keep peak centres from each replicate</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a>consensus_peaks <span class="ot">&lt;-</span> peaks <span class="sc">%&gt;%</span> </span>
<span id="cb44-2"><a href="#cb44-2"></a>  <span class="fu">endoapply</span>(mutate, <span class="at">centre =</span> start <span class="sc">+</span> peak) <span class="sc">%&gt;%</span> </span>
<span id="cb44-3"><a href="#cb44-3"></a>  <span class="fu">makeConsensus</span>(<span class="at">var =</span> <span class="st">"centre"</span>, <span class="at">p =</span> <span class="dv">1</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-4"><a href="#cb44-4"></a>  <span class="fu">mutate</span>(<span class="at">centre =</span> <span class="fu">vapply</span>(centre, median, <span class="fu">numeric</span>(<span class="dv">1</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb44-5"><a href="#cb44-5"></a>  <span class="fu">select</span>(centre, n)</span>
<span id="cb44-6"><a href="#cb44-6"></a>consensus_peaks</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 9 ranges and 2 metadata columns:
      seqnames          ranges strand |    centre         n
         &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;
  [1]     chr1   523658-523939      * |    523800         2
  [2]     chr1   780837-781330      * |    781141         2
  [3]     chr1   821066-821391      * |    821219         2
  [4]     chr1   849166-849528      * |    849349         2
  [5]     chr1 1707074-1707445      * |   1707252         2
  [6]     chr1 1790623-1791019      * |   1790839         2
  [7]     chr1 1800936-1801227      * |   1801090         2
  [8]     chr1 1814836-1815157      * |   1815023         2
  [9]     chr1 1933329-1933586      * |   1933462         2
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</section>
<section id="map-to-features" class="slide level2 smaller">
<h2>Map To Features</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb46-2"><a href="#cb46-2"></a>features <span class="ot">&lt;-</span> <span class="fu">import.gff</span>(<span class="st">"ZR-75_h3k27ac_features.gtf"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(feature)</span>
<span id="cb46-3"><a href="#cb46-3"></a>features[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 5 ranges and 1 metadata column:
      seqnames        ranges strand |     feature
         &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
  [1]     chr1 762015-763309      * |    promoter
  [2]     chr1 845401-846041      * |    enhancer
  [3]     chr1 855730-856863      * |    promoter
  [4]     chr1 858449-862203      * |    promoter
  [5]     chr1 866935-869327      * |    enhancer
  -------
  seqinfo: 1 sequence from GRCh37 genome; no seqlengths</code></pre>
</div>
</div>
<div class="fragment" data-fragment-index="1">
<p><code>bestOverlap()</code> makes this simple</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1"></a>consensus_peaks<span class="sc">$</span>feature <span class="ot">&lt;-</span> <span class="fu">bestOverlap</span>(consensus_peaks, features, <span class="at">var =</span> <span class="st">"feature"</span>, <span class="at">missing =</span> <span class="st">"none"</span>)</span>
<span id="cb48-2"><a href="#cb48-2"></a>consensus_peaks </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 9 ranges and 3 metadata columns:
      seqnames          ranges strand |    centre         n     feature
         &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;
  [1]     chr1   523658-523939      * |    523800         2        none
  [2]     chr1   780837-781330      * |    781141         2        none
  [3]     chr1   821066-821391      * |    821219         2        none
  [4]     chr1   849166-849528      * |    849349         2        none
  [5]     chr1 1707074-1707445      * |   1707252         2        none
  [6]     chr1 1790623-1791019      * |   1790839         2    enhancer
  [7]     chr1 1800936-1801227      * |   1801090         2        none
  [8]     chr1 1814836-1815157      * |   1815023         2        none
  [9]     chr1 1933329-1933586      * |   1933462         2    enhancer
  -------
  seqinfo: 24 sequences from GRCh37 genome</code></pre>
</div>
</div>
</div>
</section>
<section id="summarise-binding-patterns" class="slide level2">
<h2>Summarise Binding Patterns</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1"></a><span class="fu">plotPie</span>(consensus_peaks, <span class="at">fill =</span> <span class="st">"feature"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="20221201_extraChIPs_files/figure-revealjs/unnamed-chunk-28-1.png" width="960" class="r-stretch"></section>
<section id="summarise-binding-patterns-1" class="slide level2">
<h2>Summarise Binding Patterns</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a><span class="fu">plotPie</span>(consensus_peaks, <span class="at">fill =</span> <span class="st">"feature"</span>, <span class="at">scale_by =</span> <span class="st">"width"</span>) <span class="sc">+</span></span>
<span id="cb51-2"><a href="#cb51-2"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb51-3"><a href="#cb51-3"></a>  <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"Feature"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="20221201_extraChIPs_files/figure-revealjs/unnamed-chunk-29-1.png" width="960" class="r-stretch"></section></section>
<section>
<section id="differential-binding" class="title-slide slide level1 center">
<h1>Differential Binding</h1>

</section>
<section id="dualfilter" class="slide level2">
<h2><code>dualFilter()</code></h2>
<p>Sliding Windows (<code>csaw::windowCounts()</code>) <span class="citation" data-cites="csaw-db">(<a href="#/references" role="doc-biblioref" onclick="">Lun and Smyth 2014</a>)</span></p>
<ul>
<li>Discard non-signal windows using <code>dualFilter()</code></li>
<li>Combines 1) <em>signal-to-input</em> and 2) <em>overall signal</em> as filtering strategy</li>
<li>Provide regions of expected signal as a guide</li>
</ul>
</section>
<section id="gravi-4" class="slide level2">
<h2>GRAVI</h2>
<ul>
<li>Smooth Quantile Normalisation on logCPM <span class="citation" data-cites="sqn">(<a href="#/references" role="doc-biblioref" onclick="">Hicks et al. 2017</a>)</span></li>
<li><code>limma-trend</code> <span class="citation" data-cites="Law2014-xq">(<a href="#/references" role="doc-biblioref" onclick="">Law et al. 2014</a>)</span></li>
<li>Range-based <span class="math inline">\(H_0\)</span> (<code>treat()</code>) <span class="citation" data-cites="McCarthy2009-qf">(<a href="#/references" role="doc-biblioref" onclick="">McCarthy and Smyth 2009</a>)</span></li>
</ul>
</section>
<section id="plotassaydensities" class="slide level2">
<h2><code>plotAssayDensities()</code></h2>

<img data-src="plot-logcpm-densities-1.png" class="r-stretch quarto-figure-left"><p class="caption">A) Raw logCPM and b) Smooth Quantile Normalised logCPM for the Androgen Receptor (AR)</p></section>
<section id="plotassayrle" class="slide level2">
<h2><code>plotAssayRle()</code></h2>

<img data-src="plot-rle-normalised-1.png" class="r-stretch quarto-figure-left"><p class="caption">RLE plot calculating RLE within treatment groups</p></section>
<section id="plotassaypca" class="slide level2">
<h2><code>plotAssayPCA()</code></h2>

<img data-src="plot-pca-1.png" class="r-stretch quarto-figure-left"><p class="caption">PCA plots A) before and B) after SQN</p></section>
<section id="mergebycol" class="slide level2">
<h2><code>mergeByCol()</code></h2>
<ul>
<li>Differential binding tested using sliding windows</li>
<li>Merge adjacent sliding windows taking representative window</li>
<li>GRAVI defaults to window with maximum signal</li>
<li>wraps multiple <code>csaw</code> steps</li>
<li>returns representative range</li>
</ul>
</section>
<section id="mergebycol-1" class="slide level2">
<h2><code>mergeByCol()</code></h2>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1"></a>win</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 3 ranges and 3 metadata columns:
      seqnames    ranges strand |   AveExpr     logFC    PValue
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
  [1]     chr1     1-100      * |   4.49781   1.88678 0.0181456
  [2]     chr1    51-150      * |   7.13153   1.61697 0.0128710
  [3]     chr1   101-200      * |   5.92108   1.41863 0.0452510
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<p><br></p>
<div class="fragment" data-fragment-index="1">
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a><span class="fu">mergeByCol</span>(win, <span class="at">col =</span> <span class="st">"AveExpr"</span>, <span class="at">pval =</span> <span class="st">"PValue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 1 range and 8 metadata columns:
      seqnames    ranges strand | n_windows      n_up    n_down keyval_range
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;    &lt;GRanges&gt;
  [1]     chr1     1-200      * |         3         1         0  chr1:51-150
        AveExpr     logFC    PValue PValue_fdr
      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;
  [1]   7.13153   1.61697  0.012871   0.012871
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
</div>
</section>
<section id="mapbyfeature" class="slide level2">
<h2><code>mapByFeature()</code></h2>
<pre><code>mapByFeature(
  gr,
  genes,
  prom,
  enh,
  gi,
  cols = c("gene_id", "gene_name", "symbol"),
  gr2prom = 0,
  gr2enh = 0,
  gr2gi = 0,
  gr2gene = 1e+05,
  prom2gene = 0,
  enh2gene = 1e+05,
  gi2gene = 0,
  ...
)</code></pre>
</section>
<section id="plotprofileheatmap" class="slide level2">
<h2><code>plotProfileHeatmap()</code></h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="AR_Up_GATA3_Up_profile_heatmap.png" width="650"></p>
</div><div class="column" style="width:50    %;">
<p>Uses <code>ggplot2</code> plotting with facets for samples or peak groupings</p>
</div>
</div>
</section>
<section id="plothfgc" class="slide level2">
<h2><code>plotHFGC()</code></h2>
<div class="columns">
<div class="column" style="width:55%;">
<ul>
<li>Wraps <code>GViz</code> plotting functions</li>
<li>plot <em>optional</em> 1) HiC, 2) Features, 3) Genes or 4) Coverage</li>
<li>Helpful for visualising multiple ranges quickly</li>
</ul>
</div><div class="column" style="width:45%;">
<p><img data-src="AR_Up_GATA3_Up_AveExpr.png"></p>
</div>
</div>
</section>
<section id="references" class="slide level2 smaller scrollable">
<h2>References</h2>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-workflowr" class="csl-entry" role="doc-biblioentry">
Blischak, John D, Peter Carbonetto, and Matthew Stephens. 2019. <span>“Creating and Sharing Reproducible Research Code the Workflowr Way [Version 1; Peer Review: 3 Approved].”</span> <em>F1000Research</em> 8 (1749). <a href="https://doi.org/10.12688/f1000research.20843.1">https://doi.org/10.12688/f1000research.20843.1</a>.
</div>
<div id="ref-Hickey2021" class="csl-entry" role="doc-biblioentry">
Hickey, Theresa E., Luke A. Selth, Kee Ming Chia, Geraldine Laven-Law, Heloisa H. Milioli, Daniel Roden, Shalini Jindal, et al. 2021. <span>“<span class="nocase">The androgen receptor is a tumor suppressor in estrogen receptor–positive breast cancer</span>.”</span> <em>Nature Medicine 2021 27:2</em> 27 (2): 310–20. <a href="https://doi.org/10.1038/s41591-020-01168-7">https://doi.org/10.1038/s41591-020-01168-7</a>.
</div>
<div id="ref-sqn" class="csl-entry" role="doc-biblioentry">
Hicks, Stephanie C, Kwame Okrah, Joseph N Paulson, John Quackenbush, Rafael A Irizarry, and Héctor Corrada Bravo. 2017. <span>“<span class="nocase">Smooth quantile normalization</span>.”</span> <em>Biostatistics</em> 19 (2): 185–98. <a href="https://doi.org/10.1093/biostatistics/kxx028">https://doi.org/10.1093/biostatistics/kxx028</a>.
</div>
<div id="ref-Law2014-xq" class="csl-entry" role="doc-biblioentry">
Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. <span>“Voom: Precision Weights Unlock Linear Model Analysis Tools for <span class="nocase">RNA-seq</span> Read Counts.”</span> <em>Genome Biol.</em> 15 (2): R29.
</div>
<div id="ref-csaw-db" class="csl-entry" role="doc-biblioentry">
Lun, Aaron T L, and Gordon K Smyth. 2014. <span>“<span>D</span>e Novo Detection of Differentially Bound Regions for <span>C</span>h<span>I</span><span>P</span>-Seq Data Using Peaks and Windows: Controlling Error Rates Correctly.”</span> <em>Nucleic Acids Res.</em> 42 (11): e95.
</div>
<div id="ref-McCarthy2009-qf" class="csl-entry" role="doc-biblioentry">
McCarthy, Davis J, and Gordon K Smyth. 2009. <span>“Testing Significance Relative to a Fold-Change Threshold Is a <span>TREAT</span>.”</span> <em>Bioinformatics</em> 25 (6): 765–71.
</div>
</div>

<img src="https://github.com/steveped/stickers/raw/main/extraChIPs_sticker.png" class="slide-logo r-stretch"><div class="footer footer-default">

</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="20221201_extraChIPs_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="20221201_extraChIPs_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="20221201_extraChIPs_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="20221201_extraChIPs_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="20221201_extraChIPs_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="20221201_extraChIPs_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="20221201_extraChIPs_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="20221201_extraChIPs_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="20221201_extraChIPs_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="20221201_extraChIPs_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1600,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        function fireSlideChanged(previousSlide, currentSlide) {

          // dispatch for htmlwidgets
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for reveal
        if (window.Reveal) {
          window.Reveal.addEventListener("slidechanged", function(event) {
            fireSlideChanged(event.previousSlide, event.currentSlide);
          });
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        target: function(trigger) {
          return trigger.previousElementSibling;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        setTimeout(function() {
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'quarto-reveal',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const cites = ref.parentNode.getAttribute('data-cites').split(' ');
        tippyHover(ref, function() {
          var popup = window.document.createElement('div');
          cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    });
    </script>
    

</body></html>