---
title: "Alternate References For Transcriptomics"
subtitle: "Adelaide Bioinformatics Seminars"
author: "Dr Stevie Pederson (They/Them)"
affiliation: "Black Ochre Data Labs"
date: "2024-05-03"
date-format: long
title-slide-attributes:
    data-background-color: "#3D3D40"
    data-background-opacity: "20%"
    data-background-size: "35%"
    data-background-position: "50% 50%"
editor: source
format: 
  revealjs:
    bibliography: references.bib
    theme: [bodl.scss]
    code-line-numbers: false
    width: 1280
    height: 720
    sansfont: Times New Roman
    logo: assets/bodl_logo_white_background.jpg
    slide-number: c/t
include-after: |
  <script type="text/javascript">
    Reveal.on('ready', event => {
      if (event.indexh === 0) {
        document.querySelector("div.has-logo > img.slide-logo").style.display = "none";
      }
    });
    Reveal.addEventListener('slidechanged', (event) => {
      if (event.indexh === 0) {
        Reveal.configure({ slideNumber: null });
        document.querySelector("div.has-logo > img.slide-logo").style.display = "none";
      }
      if (event.indexh === 1) { 
        Reveal.configure({ slideNumber: 'c' });
        document.querySelector("div.has-logo > img.slide-logo").style.display = null;
      }
    });
  </script>    
knitr: 
  opts_chunk: 
    echo: true
    include: true
    warning: false
    message: false
    fig.align: left
    fig.width: 8
    fig.heght: 6
---

##

::: {.columns}

::: {.column width="35%"}

### Acknowledgement of Country


:::

::: {.column width="65%"}

I would like to acknowledge that many of us are meeting today on Kaurna Country.<br><br>

I acknowledge the deep feelings of attachment and relationship of the Kaurna people to their Place.<br><br>

I also pay my respects to the cultural authority of Aboriginal and Torres Strait Islander peoples from other areas of Australia online today, and pay my respects to Elders past, present and emerging. 


:::

:::

::: {.notes}
I also wish to express how grateful I am for the opportunity to be involved in this space.
I'm acutely aware of the trust that has been placed in me and am very pleased to be surrounded by so many wise & experienced researchers
:::

## About Me

::: {.columns}

::: {.column width='60%'}

- Postdoctoral Fellow, Black Ochre Data Labs, Indigenous Genomics, Telethon Kids Institute

::: {.fragment fragment-index=1}
- 2020-2022: Dame Roma Mitchell Cancer Research Labs, University of Adelaide
:::

::: {.fragment fragment-index=2}
- 2014-2020: Bioinformatics Hub, University of Adelaide
:::

::: {.fragment fragment-index=3}
- 2008-2018: Slowest PhD Student in the world...
:::

::: {.fragment fragment-index=4}
- Dec 2002: First used R & `limma`
:::

::: {.fragment fragment-index=5}
- 1^st^ Year B.Sc. in 1986 $\implies$ Dropped out in 1987
:::

:::

::: {.column width='40%'}

::: {.r-stack}

::: {.fragment .fade-out fragment-index=1}

![](assets/TKI.jpg){.absolute top=50 right=50}

:::

::: {.fragment .fade-out fragment-index=4}
::: {.fragment .fade-in fragment-index=1}

![](assets/UoAlogo.png){.absolute top=50 right=50 width='400px'}
:::
:::

::: {.fragment .fade-out fragment-index=5}

![](assets/R_logo.png){.fragment .fade-in fragment-index=4}

:::

::: {.fragment .fade-in fragment-index=5}

![](assets/waterfall.jpg){.absolute top=50 right=50}

:::

:::

:::

:::

::: {.notes}
- Have primarily worked in bulk transcriptomics (not much scRNA)
- Lots of ChIP-Seq as well
- I dropped out to become an influencer & scored a lucrative deal with an umbrella company. Just a bit ahead of my time really
:::

## Bioconductor Package Developer


![](https://github.com/smped/ngsReports/blob/devel/man/figures/ngsReports.png?raw=true){.absolute top=100 left=50 width='250px'}

![](https://github.com/smped/extraChIPs/blob/devel/man/figures/extraChIPs.png?raw=true){.absolute top=400 left=180 width='250px'}

::: {.fragment}

![](https://raw.githubusercontent.com/Bioconductor/BiocStickers/devel/sSNAPPY/sSNAPPY.png){.absolute top=100 left=400 width='250px'}

:::

::: {.fragment}

![](https://smped.github.io/motifTestR/reference/figures/favicon.png){.absolute top=100 left=750 width='250px'}

![](https://smped.github.io/transmogR/reference/figures/transmogR.png){.absolute top=400 left=580 width='250px'}

:::

::: {.notes}
- ngsReports is setup for parsing FastQC & Fastp files across large experiments, as well as lots of other log files (AdapterRemoval, cutadapt, STAR, salmon, bowtie etc)
    + I still use this most weeks for my own work so it's actively maintained
- extraChIPs is for ChIP-Seq analysis & integration across ChIP targets
    + I found DiffBind super-frustrating so wrote this to make my life easier
    + Also use many of the utility functions very regularly
- Also very pleased for Dr Nora Liu to have sSNAPPY available on Bioc, which is an enrichment package for challenging clinical samples where you have pre/post treatment
- Lachlan Baer's package tadar for dealing with differential allelic representation when using CRISPR etc to generate mutants

- Two new packages in Bioc 3.19 (released on Wednesday)
    + transmogR relates to the main topic for today
    + motifTesR was a side project, but might be useful for many

:::

## motifTestR

- Was adding motif enrichment testing to the GRAVI workflow
- Decided to avoid the MEME-Suite &#129327;
    + Plays badly with `conda`
    + I struggle to interpret the results
    + Not R native (although `memes` does wrap some of it)
    
. . .

- Particularly interested in motif positions within a set of sequence
    + Analogous to `centrimo`
- Added enrichment testing because I could

::: {.notes}
- The MEME-Suite plays really badly with `conda` and take a huge effort to install correctly
:::

## motifTestR: Positional Bias

- If no positional bias $\implies$ motifs will match uniformly across the sequence width

::: {.incremental}

1. Break sequences into bins
2. Assign equal probabilites ($\pi_i$) of matches within bins $i = 1, ... I$
    + except near boundaries
3. Test matches in bin $i$ are as expected using $\pi_i$
    + `binomial.test()`
4. Use harmonic mean p-value to obtain a single p-value for each motif
    + Returns 'enriched sequence regions' where $p_i$ < $HMP(p)$

:::

::: {.notes}
- Does require all sequences to be the same width
- Seems to turn up the things I expect to see knowing the data
- Results are easily interpretable
- Probs the best shit idea I've ever had
:::

## motifTestR: Positional Bias

- Can be performed using:
    a. absolute distance from centre or
    b. the entire sequence width
    
. . .    

- All sequences must be the same length
- Relies on a single 'best match' per sequence (but also applies weights)
    + Also finds motif matches for you
    + Heavy reliance on `Biostrings`
    
. . .
    
- 7,000 seq $\times$ 400bp $\times$ 400 motifs $\implies$ < 30sec (phoenix: 8 cores)
- 50,000 seq $\times$ 800bp $\times$ 400 motifs $\implies$ ~2min (phoenix: 8 cores)

## motifTestR: Positional Bias Plots

::: {.columns}

::: {.column width='40%'}

<br>

``` r
A <- top_matches %>% 
  plotMatchPos(
    se = FALSE, abs = TRUE, linewidth = 1/2,
    binwidth = motif_params$binwidth / 2
  ) +
  labs(
    x = "Distance From Centre", 
    y = "Proportion", 
    colour = "Motifs In Cluster"
  ) +
  theme(legend.position = "none")
B <- top_matches %>% 
  plotMatchPos(
    type = "cdf", geom = "line", abs = TRUE,
    linewidth = 1 / 2, binwidth = 1
  ) +
  labs(
    x = "Distance From Centre", 
    y = "Proportion", 
    colour = "Representative Motif"
  )
A + B
```

:::

::: {.column width='60%'}

![Sequences were taken from AR ChIP-Seq binding sites](assets/plot-smoothed-pos-1.png)

:::


:::

::: {.notes}
- This is set of sequences from AR ChIP-Seq
- Glucocorticoid Receptor Motif is very concentrated zero the centre (which AR bind)
- Also picks up ZN770 as being ~150bp from centre
:::

## motifTestR: Positional Bias Plots

::: {.columns}

::: {.column width='40%'}

<br>

``` r
top_matches %>% 
  plotMatchPos(
    geom = "point", abs =TRUE, use_totals = TRUE,
    binwidth = 1
  ) +
  geom_smooth(
    se = FALSE, colour = "black", 
    linewidth = 1/2, method = 'loess'
  ) +
  facet_wrap(~name, labeller = as_labeller(lb)) +
  theme(legend.position = "none") +
  labs(
    x = "Distance From Sequence Centre", 
    y = "Total Matches"
  )
```

<br>

Show the matches at each position

:::

::: {.column width='60%'}

![](assets/plot-total-pos-matches-1.png)

:::

:::

::: {.notes}
- The same function can show actual matches (binwidth = 1)
- Showing counts reveals that ZN770 isn't really common
:::


## motifTestR: Positional Bias Plots

::: {.columns}

::: {.column width='40%'}

<br>

``` r
motif_list %>% 
    dplyr::filter(altname %in% top_pos) %>% 
    to_list() %>% 
    getPwmMatches(
      fw_seq, abs = TRUE, best_only = TRUE, 
      mc.cores = threads
    ) %>% 
    plotMatchPos(
      abs = TRUE, type = "heatmap", 
      cluster = TRUE, use_totals = TRUE,
      binwidth = 10
    ) + 
    labs(
    x = "Bin", fill = "Total\n Matches"
    )
```

::: {style="font-size:80%;"}

- Clustering & heatmaps can help identify redundancy
- Useful for larger sets of motifs / matches

:::

:::

::: {.column width='60%'}

![](assets/heatmap-pos-1.png)

:::

:::

## motifTestR: Motif Enrichment

::: {style="font-size: 90%"}

::: {.columns}

::: {.column width='47%'}

1. *Hypergeometric*: Subset A vs Subset B
    + Specifically for sequences where one subset changes & the other doesn't
    + Realistically, results are specific to the each BG set
    

::: {.fragment}
2. *Poisson*: SetA Vs BG
    + BG Set >>> Set A
    + Estimates match rate per sequence in BG $\implies$ `poisson.test()`
    + Quick(ish) & dirty
    + Explicitly assumes matches are Poisson

:::

:::

::: {.column width='5%'}

:::

::: {.column width='47%'}

::: {.fragment}
3. *QuasiPoisson*: SetA Vs BG (In Blocks)
    + Tests match rate per **block** (i.e. iteration) in BG
    + Allows for overdispersion

<br>
:::

::: {.fragment}
4. *Iterations*: SetA Vs BG (In Blocks)
    + No distributional assumptions (except CLT)
    + Derives mean & sd per iteration in BG<br>$\implies$ $Z$-score + p-value

:::

:::

:::

:::

## motifTestR: `makeRMRanges()`

- Added function for obtaining *R*andom *M*atched *R*anges for a test set
- Relies on a discrete set of features
    + e.g. promoters, enhancers, etc
    
. . .

- Samples RMRanges to exactly match the feature distribution of the test set
    + Can be sampled in blocks for iterative methods (quasipoisson / iteration)

## motifTestR: Motif Enrichment

::: {.columns}

::: {.column width='70%'}

- Is much, much slower... &#128542;

::: {.fragment fragment-index=1}
- 7000 $\times$ 400bp $\times$ 400 motifs (quasipoisson):
    + Using 7000 $\times$ 1000 = 7x10^6^ RMRanges
    + 7 minutes (phoenix: 8 cores)
    
:::

::: {.fragment fragment-index=2}
- 50,000 $\times$ 800bp $\times$ 400 motifs (quasipoisson):
    + Using 50,000 $\times$ 1000 = 5x10^7^ RMRanges
    + 1hr 35min minutes (phoenix: 8 cores)

:::

::: {.fragment fragment-index=3}
- 400 Vs 6500 $\times$ 400bp $\times$ 400 motifs (hypergeometric):
    + 35 sec (phoenix: 8 cores)
:::


:::

::: {.column width='30%'}

::: {.fragment fragment-index=4}

- This dog wants you to try it...

<br>

<iframe src="https://giphy.com/embed/6MWahPArixa6I" width="480" height="480" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/cute-dog-6MWahPArixa6I">via GIPHY</a></p>
:::

:::

:::

# And Now My Real Talk... {background-color="#3D3D40"}

## Current Research

- PROPHECY (Preventing Renal OPthalmic and Heart Events in CommunitY)
- Multi-Omics study led by Prof Alex Brown, with Chief Data Scientist A/Prof Jimmy Breen

. . .

- Investigating the very high rates of Diabetes within Indigenous Australians
    + Also very high rates of complications: CVD & CKD
    
. . .

- Is strongly led by Community in partnership with BODL
    + Hoping to bring benefits of Precision Medicine
    + New approach to Indigenous Health [$\implies$ listen first]{.fragment}

## Life Expectancy Gaps

![Courtesy of A/Prof Jimmy Breen, Black Ochre Data Labs, TKI](assets/life_expectancy.png)

## Mortality From Diabetes

![](assets/mortality.png){width='800px'}

## Why?

- Taking all socio-economic indicators $\implies$ disease risk is still hugely increased for Indigenous Australiams
- Not only disease risk, but higher rates of profound complications

. . .

- Are there risk alleles?
    + If so, what are they? How do they contribute to risk?
- Are there protective alleles?

. . .

- Is there an epigenetic contribution?
- Additional environmental factors?

## Multi-Omic Approaches

![](assets/prophecy.png){width='1000px'}

## The Transcriptomics Layer

- Whole blood samples
- Can we capture dynamics of disease / signatures / biomarkers etc?
- Approach will be classical statistics + network approaches

. . .

- Tried to be excruciatingly careful with design:
    + plate layout, pooled replicates, globin-depletion, ERCC spike-ins, RIN scores etc
- Library preps are finally underway
- Data lands around September

# Why Think About Reference Genomes? {background-color="#3D3D40"}

## Genetic Diversity

- Indigenous Australians are almost totally absent from repositories of genetic variation 
- $\sim$ 25% of variation within Indigenous Australian populations is unique & poorly characterised [@Easteal2020-fo]

. . .

- If contributing to disease risk and we use a standard reference [$\implies$ will we see it?]{.fragment}

. . .

- Can we bring the reference closer to our participants' genomes?

. . .

- NCIG are very close to a pangenome graph assembled from Indigneous Australians
    + A little too far off of to be viable yet
    + Monica Guilhaus doing an amazing job figuring out spliced pan-transcriptome graphs [@Sibbesen2023-md]
    
::: {.notes}
At least one member of our Indigenous Governance groups has said "Why are we even comparing ourselves to any other group. We should compare ourselves to us?
:::

## Gene-Level Approaches

- *STARconsensus* [@Kaminow2022-dz] enables inclusion of a population-specific consensus variant set
    + Also possible for personalised genomes
    + Now has a `diploid` mode for phased, personalised variant sets
    
. . .

- Incorporates variants during indexing
    + Indexes twice (Standard + Modified)
    + Returns alignment co-ordinates relative to the unmodified reference
    
. . .

- Simple to obtain counts using standard methods (e.g. featureCounts)

## The Downside

- Locked into using STAR
    + In my hands can't align full-length, reference-derived transcripts
- Counts are gene-level

. . .

- We understand the distribution of counts at the gene level
    + `edgeR` & `DESeq2`: well established Negative Binomial methods
    
. . .

- Transcript-level counts have additional uncertainty
    + Need to be scaled by an over-dispersion estimate<br>$\implies$ can then be treated as Negative Binomial [@Baldoni2024-zf]
    + Cannot obtain these from `STAR` + `featureCounts`
    
## Transcript-Level Analysis

- `salmon` provides transcript-level counts with overdispersion estimate!
- Can we modify a reference transcriptome using a set of variants?

::: {.fragment}

<br><br>

<iframe src="https://giphy.com/embed/3ohs2xhB3jP6XfJ4ME" width="480" height="265" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/colbertlateshow-late-show-the-with-stephen-colbert-3ohs2xhB3jP6XfJ4ME">via GIPHY</a></p>

![](https://smped.github.io/transmogR/reference/figures/transmogR.png){.absolute bottom=50 right=100 width='400px'}

:::

## transmogR + salmon

- Takes about 15-20 mins, even on a laptop
- Modified transcripts can be optionally tagged:
    + Indicating variant types (`s/i/d`)
    + Naming the variant set (e.g. `panhuman`)
- Generally using SNVs + InDels < 50bp

. . .

- I've been using a modified transcriptome & modified genome as the *gentrome*
    + The genome is used by `salmon` as decoy sequences [@Srivastava2020-tm]
    
. . . 

- Indexing for `salmon` still takes a while...

::: {.notes}
Structural variants really need a new approach like the graph-based ones
:::

# DBPAInT: Developing Best Practices for the Analysis of Indigenous Transcriptomes  {background-color="#3D3D40"}

::: {.notes}
Ths is very much work in progress with some results coming in last night
:::

## The Approach

- STARconsensus: using variants from the wrong population still improves performance [@Kaminow2022-dz]

. . .

1. 1000 Genomes Project Panhuman
    + alleles found in >50% of unrelated across entire project

. . .

2. 1000 GP AFR Subpopulation
    + alleles found in >50% of unrelated within the AFR population
    + most distinct population from the Panhuman
    
. . .

3. PROPHECY consensus set
    + alleles found in > 50% of unrelated members of the study
    + Used 3 familial steps to define 'unrelated'
    
## The Analysis

- 6 female pilot samples
    + All genomes are y-masked

- Replicating elements of STARconsensus
- Taking GRCh38 as the worst case $\implies$ changes are improvements?
    + Four sets of genomic alignments / sample
    + Personalised variants arriving in the next few days (weeks?)
    
. . .

- Also running a `transmogR` + `salmon` analysis
    + Four sets of transcripomic alignments / sample
    + Haven't figured out how to do a diploid transcriptome yet...
    
::: {.notes}
Personalised references give the closest to ground truth we can realistically get
:::

## The Variants

![](assets/ol-all-var-1.png)

::: {.notes}
- High levels of common variation $\implies$ even the wrong variant set shold be an improvement
- AFR & PROPHECY have the most uniqueness, but PROPHECY appears closer to the panhuman set
:::

## PROPHECY Variants Overlapping Exons

![](assets/prophecy-by-trans-1.png)

::: {.notes}
Taking the variants which overlap exons, how many transcripts are impacted
:::

## Key Questions

1. Are technical metrics improved?
2. Does it really make a difference?
3. Does any difference actually capture the underlying biology?

. . .

- STAR is deterministic for given reference
    + Change the reference even slightly $\implies$ no guarantees
    
. . .

- `salmon` is not deterministic

. . .

- Some of the changes will just be noise, but *how much is noise*?

## Technical Analyses

- No real change to number of reads aligning (`STAR` or `salmon`)
- Very noticeable changes within alignments
- (Nearly) all technical metrics appear to be supportive of a variant modified reference

. . .

- First step is to check uniquely aligned reads from STARconsensus...


## Uniquely Aligning Reads (STARconsensus)

::: {.columns}

::: {.column}

![All variant sets show an increase in the number of uniquely-aligned reads (10-30,000)](assets/plot-nett-unique-1.png)
:::

::: {.column}

::: {.fragment}

![PROPHECY shows a very slight increase in the number of CIGAR M-bases per uniquely-aligned read](assets/compare-matching-bases-1.png)
:::

:::

:::

## However... {.smaller}

::: {.columns}

::: {.column width='65%'}

![](assets/barplot-all-changes-1.png)

:::

::: {.column width='35%'}

- About 1 in 300 reads moves
- About 30-40,000 reads remain uniquely aligned but in a different location

::: {.fragment}
- How much is noise?
- Are the changes shared between samples?
- Do the changed reads overlap a variant?
- Where exactly are they moving *from* and *to*?
:::

:::

:::

## Reads Overlapping Variants

::: {.columns}

::: {.column width='64%'}

![](assets/var-alluvial-1.png){fig-align='left'}

:::

::: {.column width='36%'}

Taking reads where gapped alignments overlap a variant

- chr12 to chr1 shared by all
- Also chr9 to chr7
- chrX to chr19 shared by 1, 2, 6
- chr12 to chr16 shared by 2 $\rightarrow$ 5
:::

:::

::: {.notes}
These strike me as patterns which support the underlying biology & diversity within particiapnts
:::

## Reads Overlapping Variants

::: {.columns}

::: {.column width='60%'}
![](assets/alluvial-gene-type-1.png){width='600px'}
:::

::: {.column width='40%'}

Taking all samples combined

- Increase in alignments to protein coding genes
- Decrease in pseudogenes

:::

:::

## Common Genes Being Impacted

::: {.columns}

::: {.column width='65%'}

![](assets/individual-genes-1.png)
:::

::: {.column width='35%'}

Taking the 20 most common shifts

- HLA-A
- HLA-H
- Ribosomal proteins
- Histone Proteins

:::

:::

## Quick Summary

- These are only the uniquely-aligned reads from STARconsensus
    + Don't overlap alignment positions in either reference
    + Haven't looked at unmapped $\rightarrow$ mapped, or multi-mappers
- Looks like it may be consistent & true biology?
- I got these results Wed night so I'm still processing them mentally

## Differential Gene Expression Analysis

::: {.columns}

::: {.column width='60%'}

![](assets/plot-ma-1.png){fig-align="left"}

:::

::: {.column width='40%'}


- GRCh38 vs PROPHECY-modified
- Identical input reads
- All changes are alignment induced

:::

:::

## Genes Impacted Variably {.smaller}

::: {.columns}

::: {.column width='60%'}

![](assets/plot-variable-1.png){fig-align="left"}

:::

::: {.column width='40%'}



Taking expression estimates within each reference:

1. Find the difference in counts per gene & sample
2. Take mean & sd of differences

::: {.fragment}

- Consistent changes will be low-variability, far from zero
- Inconsistent changes high-variability but near zero
- Like looking through an MA plot with varibility as 3^rd^ dimension

:::



:::

:::

# What about `salmon` + `transmogR` {background-color="#3D3D40"}

## Key Differences to STAR

- Only ~30% of reads align uniquely
    + Largest NH:i tag is ~24,000
    + 90% of reads align with NH:i < 10
    
. . .

- BAM files directly connect read ID to transcript
    + Makes comparison much simpler
    
. . .

- How do the *transcript-level* counts change?
- How do the *overdispersion* estimates change?


## Salmon Metrics

- Increase in Properly-Paired alignments (+3000)
- PROPHECY has strongest increase in correctly oriented fragments (+10,000 ISF)
- Decrease in umapped reads (-2000)

. . .

- Minor change in library sizes
- Similar numbers of detected transcripts (>1 count)

## Overdispersion Estimates

![Looking at how the number of transcripts / gene also impacts these](assets/compare-overdispersions-1.png){fig-align="left"}

## Raw Transcript-Level Counts {visibility="hidden"}

![Comparing raw transcript-level counts. Changes > 100-fold are labelled](assets/compare-trans-counts-raw-1.png)

## Scaled Transcript-Level Counts

![This seems quite acceptable](assets/compare-trans-counts-1.png){fig-align="left"}

## DTE {visibility="hidden"}

![](assets/trans-ma-scaled-1.png){fig-align="left"}


## Variable Transcripts

![](assets/trans-variable-1.png){fig-align="left"}

## Tracking Individual Reads

- Can easily track a read & all it's transcripts 
- Subset data to reads with NH:i $\leq 10$
- Subset reads to those aligning to a modified transcript either before or after

. . .

- Unique-mappers are far less informative for transcript-level alignments
- Haven't checked for moves between chromosomes
    + Transcript & Gene-Level moves are more fine-grained


## Tracking Reads Which Change Transcripts

![Reads changing transcript **within** a gene](assets/salmon-moved-within-1.png){fig-align="left"}

::: {.notes}
The consistency between samples is very encouraging
:::

## Tracking Reads Which Change Transcripts

![Reads changing transcript to different genes](assets/salmon-moved-across-1.png){fig-align="left"}

::: {.notes}
- Changes are again very consistent between samples
- Some similarity to STAR changes, but not the pseudogene story!!!
:::

## Changing Transcripts: Gene-Level Categories

![Shows a nice decrease in reads mapping nowhere](assets/salmon-moved-gene-alluvial-1.png){fig-align="left"}

::: {.notes}
A small increase in protein coding genes for some samples
:::

## Changing Transcripts: Transcript-Level Categories

![Lots of change between but no categories really sing out](assets/salmon-moved-trans-alluvial-1.png){fig-align="left"}

## Where To From Here

- Personalised Variant sets will give us the best estimate of ground-truth
    + Will run STARconsensus in Diploid mode
    + Haven't figured out how to do this for `transmogR` + `salmon
    + Will just choose randomly from Het sites (???)

. . .
  
- Running a mini 90-sample cohort to see how it really plays at scale
    + Previously analysed by Prof Sam El-Osta's team (Monash)
    + Have a specific subset of complications
    
- How will WGCNA modules be impacted?
- How will true DGE / DTE results be impacted?

## References
