<!DOCTYPE html>
<html lang="en"><head>
<script src="20240503_transmogR_files/libs/clipboard/clipboard.min.js"></script>
<script src="20240503_transmogR_files/libs/quarto-html/tabby.min.js"></script>
<script src="20240503_transmogR_files/libs/quarto-html/popper.min.js"></script>
<script src="20240503_transmogR_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="20240503_transmogR_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="20240503_transmogR_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="20240503_transmogR_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="20240503_transmogR_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.353">

  <meta name="author" content="Dr Stevie Pederson (They/Them) Black Ochre Data Labs / ANU">
  <meta name="dcterms.date" content="2024-05-03">
  <title>Alternate References For Transcriptomics</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="20240503_transmogR_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="20240503_transmogR_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="20240503_transmogR_files/libs/revealjs/dist/theme/quarto.css">
  <link href="20240503_transmogR_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="20240503_transmogR_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="20240503_transmogR_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="20240503_transmogR_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" data-background-color="#3D3D40" data-background-opacity="20%" data-background-position="50% 50%" data-background-size="35%" class="quarto-title-block center">
  <h1 class="title">Alternate References For Transcriptomics</h1>
  <p class="subtitle">Adelaide Bioinformatics Seminars</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Dr Stevie Pederson (They/Them)<br>Black Ochre Data Labs / ANU 
</div>
</div>
</div>

  <p class="date">May 3, 2024</p>
</section>
<section id="section" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:35%;">
<h3 id="acknowledgement-of-country">Acknowledgement of Country</h3>
</div><div class="column" style="width:65%;">
<p>I would like to acknowledge that many of us are meeting today on Kaurna Country.<br><br></p>
<p>I acknowledge the deep feelings of attachment and relationship of the Kaurna people to their Place.<br><br></p>
<p>I also pay my respects to the cultural authority of Aboriginal and Torres Strait Islander peoples from other areas of Australia online today, and pay my respects to Elders past, present and emerging.</p>
</div>
</div>
<aside class="notes">
<p>I also wish to express how grateful I am for the opportunity to be involved in this space. I’m acutely aware of the trust that has been placed in me and am very pleased to be surrounded by so many wise &amp; experienced researchers</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="about-me" class="slide level2">
<h2>About Me</h2>
<div class="columns">
<div class="column" style="width:60%;">
<ul>
<li>Postdoctoral Fellow, Black Ochre Data Labs, Indigenous Genomics, Telethon Kids Institute</li>
</ul>
<div class="fragment" data-fragment-index="1">
<ul>
<li>2020-2022: Dame Roma Mitchell Cancer Research Labs, University of Adelaide</li>
</ul>
</div>
<div class="fragment" data-fragment-index="2">
<ul>
<li>2014-2020: Bioinformatics Hub, University of Adelaide</li>
</ul>
</div>
<div class="fragment" data-fragment-index="3">
<ul>
<li>2008-2018: Slowest PhD Student in the world…</li>
</ul>
</div>
<div class="fragment" data-fragment-index="4">
<ul>
<li>Dec 2002: First used R &amp; <code>limma</code></li>
</ul>
</div>
<div class="fragment" data-fragment-index="5">
<ul>
<li>1<sup>st</sup> Year B.Sc. in 1986 <span class="math inline">\(\implies\)</span> Dropped out in 1987</li>
</ul>
</div>
</div><div class="column" style="width:40%;">
<div class="r-stack">
<div class="fragment fade-out" data-fragment-index="1">
<p><img data-src="assets/TKI.jpg" class="absolute" style="top: 50px; right: 50px; "></p>
</div>
<div class="fragment fade-out" data-fragment-index="4">
<div class="fragment fade-in" data-fragment-index="1">
<p><img data-src="assets/UoAlogo.png" class="absolute" style="top: 50px; right: 50px; width: 400px; "></p>
</div>
</div>
<div class="fragment fade-out" data-fragment-index="5">
<p><img data-src="assets/R_logo.png" class="fragment fade-in" data-fragment-index="4"></p>
</div>
<div class="fragment fade-in" data-fragment-index="5">
<p><img data-src="assets/waterfall.jpg" class="absolute" style="top: 50px; right: 50px; "></p>
</div>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>Have primarily worked in bulk transcriptomics (not much scRNA)</li>
<li>Lots of ChIP-Seq as well</li>
<li>I dropped out to become an influencer &amp; scored a lucrative deal with an umbrella company. Just a bit ahead of my time really</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="bioconductor-package-developer" class="slide level2">
<h2>Bioconductor Package Developer</h2>
<p><img data-src="https://github.com/smped/ngsReports/blob/devel/man/figures/ngsReports.png?raw=true" class="absolute" style="top: 100px; left: 50px; width: 250px; "></p>
<p><img data-src="https://github.com/smped/extraChIPs/blob/devel/man/figures/extraChIPs.png?raw=true" class="absolute" style="top: 400px; left: 180px; width: 250px; "></p>
<div class="fragment">
<p><img data-src="https://raw.githubusercontent.com/Bioconductor/BiocStickers/devel/sSNAPPY/sSNAPPY.png" class="absolute" style="top: 100px; left: 400px; width: 250px; "></p>
</div>
<div class="fragment">
<p><img data-src="https://smped.github.io/motifTestR/reference/figures/favicon.png" class="absolute" style="top: 100px; left: 750px; width: 250px; "></p>
<p><img data-src="https://smped.github.io/transmogR/reference/figures/transmogR.png" class="absolute" style="top: 400px; left: 580px; width: 250px; "></p>
</div>
<aside class="notes">
<ul>
<li><p>ngsReports is setup for parsing FastQC &amp; Fastp files across large experiments, as well as lots of other log files (AdapterRemoval, cutadapt, STAR, salmon, bowtie etc)</p>
<ul>
<li>I still use this most weeks for my own work so it’s actively maintained</li>
</ul></li>
<li><p>extraChIPs is for ChIP-Seq analysis &amp; integration across ChIP targets</p>
<ul>
<li>I found DiffBind super-frustrating so wrote this to make my life easier</li>
<li>Also use many of the utility functions very regularly</li>
</ul></li>
<li><p>Also very pleased for Dr Nora Liu to have sSNAPPY available on Bioc, which is an enrichment package for challenging clinical samples where you have pre/post treatment</p></li>
<li><p>Lachlan Baer’s package tadar for dealing with differential allelic representation when using CRISPR etc to generate mutants</p></li>
<li><p>Two new packages in Bioc 3.19 (released on Wednesday)</p>
<ul>
<li>transmogR relates to the main topic for today</li>
<li>motifTesR was a side project, but might be useful for many</li>
</ul></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="motiftestr" class="slide level2">
<h2>motifTestR</h2>
<ul>
<li>Was adding motif enrichment testing to the GRAVI workflow</li>
<li>Decided to avoid the MEME-Suite 🤯
<ul>
<li>Plays badly with <code>conda</code></li>
<li>I struggle to interpret the results</li>
<li>Not R native (although <code>memes</code> does wrap some of it)</li>
</ul></li>
</ul>
<p><img data-src="https://smped.github.io/motifTestR/reference/figures/favicon.png" class="absolute" style="top: 250px; right: 50px; width: 250px; "></p>
<div class="fragment">
<ul>
<li>Particularly interested in motif positions within a set of sequence
<ul>
<li>Analogous to <code>centrimo</code></li>
</ul></li>
<li>Added enrichment testing because I could</li>
</ul>
<aside class="notes">
<ul>
<li>The MEME-Suite plays really badly with <code>conda</code> and take a huge effort to install correctly</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="motiftestr-positional-bias" class="slide level2">
<h2>motifTestR: Positional Bias</h2>
<ul>
<li>If no positional bias <span class="math inline">\(\implies\)</span> motifs will match uniformly across the sequence width</li>
</ul>
<div>
<ol type="1">
<li class="fragment">Break sequences into bins</li>
<li class="fragment">Assign probabilities (<span class="math inline">\(\pi_i\)</span>) of matches within bins <span class="math inline">\(i = 1, ... I\)</span>
<ul>
<li class="fragment">Proportion of the total viable positions for a match within each bin</li>
</ul></li>
<li class="fragment">Test matches in bin <span class="math inline">\(i\)</span> are as expected using <span class="math inline">\(\pi_i\)</span> <span class="math inline">\(\implies p_i\)</span>
<ul>
<li class="fragment"><code>binomial.test()</code></li>
</ul></li>
<li class="fragment">Use harmonic mean p-value to obtain a single p-value for each motif
<ul>
<li class="fragment">Returns ‘enriched sequence regions’ where <span class="math inline">\(p_i\)</span> &lt; <span class="math inline">\(HMP(p.)\)</span></li>
</ul></li>
</ol>
</div>
<aside class="notes">
<ul>
<li>Does require all sequences to be the same width</li>
<li>Seems to turn up the things I expect to see knowing the data</li>
<li>Results are easily interpretable</li>
<li>Probs the best shit idea I’ve ever had</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="motiftestr-positional-bias-1" class="slide level2">
<h2>motifTestR: Positional Bias</h2>
<ul>
<li>Can be performed using:
<ol type="a">
<li>absolute distance from centre or</li>
<li>the entire sequence width</li>
</ol></li>
</ul>
<div class="fragment">
<ul>
<li>All sequences must be the same length</li>
<li>Relies on a single ‘best match’ per sequence (but also applies weights)
<ul>
<li>Also finds motif matches for you</li>
<li>Heavy reliance on <code>Biostrings</code></li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>7,000 seq <span class="math inline">\(\times\)</span> 400bp <span class="math inline">\(\times\)</span> 400 motifs <span class="math inline">\(\implies\)</span> &lt; 30sec (phoenix: 8 cores)</li>
<li>50,000 seq <span class="math inline">\(\times\)</span> 800bp <span class="math inline">\(\times\)</span> 400 motifs <span class="math inline">\(\implies\)</span> ~2min (phoenix: 8 cores)</li>
</ul>
</div>
</section>
<section id="motiftestr-positional-bias-plots" class="slide level2">
<h2>motifTestR: Positional Bias Plots</h2>
<div class="columns">
<div class="column" style="width:40%;">
<p><br></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>A <span class="ot">&lt;-</span> top_matches <span class="sc">%&gt;%</span> </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMatchPos</span>(</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">abs =</span> <span class="cn">TRUE</span>, <span class="at">linewidth =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">binwidth =</span> <span class="dv">5</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Distance From Centre"</span>, </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Proportion"</span>, </span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="st">"Motifs In Cluster"</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>B <span class="ot">&lt;-</span> top_matches <span class="sc">%&gt;%</span> </span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMatchPos</span>(</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">"cdf"</span>, <span class="at">geom =</span> <span class="st">"line"</span>, <span class="at">abs =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">linewidth =</span> <span class="dv">1</span> <span class="sc">/</span> <span class="dv">2</span>, <span class="at">binwidth =</span> <span class="dv">1</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Distance From Centre"</span>, </span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Proportion"</span>, </span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="st">"Representative Motif"</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>A <span class="sc">+</span> B</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div><div class="column" style="width:60%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="assets/plot-smoothed-pos-1.png"></p>
<figcaption>Sequences were taken from AR ChIP-Seq binding sites</figcaption>
</figure>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>This is set of sequences from AR ChIP-Seq</li>
<li>Glucocorticoid Receptor Motif is very concentrated zero the centre (which AR bind)</li>
<li>Also picks up ZN770 as being ~150bp from centre</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="motiftestr-positional-bias-plots-1" class="slide level2">
<h2>motifTestR: Positional Bias Plots</h2>
<div class="columns">
<div class="column" style="width:40%;">
<p><br></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>top_matches <span class="sc">%&gt;%</span> </span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMatchPos</span>(</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"point"</span>, <span class="at">abs =</span><span class="cn">TRUE</span>, <span class="at">use_totals =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">binwidth =</span> <span class="dv">1</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">colour =</span> <span class="st">"black"</span>, </span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">linewidth =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="at">method =</span> <span class="st">'loess'</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>name, <span class="at">labeller =</span> <span class="fu">as_labeller</span>(lb)) <span class="sc">+</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Distance From Sequence Centre"</span>, </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Total Matches"</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><br></p>
<p>Show the matches at each position</p>
</div><div class="column" style="width:60%;">
<p><img data-src="assets/plot-total-pos-matches-1.png"></p>
</div>
</div>
<aside class="notes">
<ul>
<li>The same function can show actual matches (binwidth = 1)</li>
<li>Showing counts reveals that ZN770 isn’t really common</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="motiftestr-positional-bias-plots-2" class="slide level2">
<h2>motifTestR: Positional Bias Plots</h2>
<div class="columns">
<div class="column" style="width:40%;">
<p><br></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>motif_list <span class="sc">%&gt;%</span> </span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(altname <span class="sc">%in%</span> top_pos) <span class="sc">%&gt;%</span> </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">to_list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">getPwmMatches</span>(</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>      fw_seq, <span class="at">abs =</span> <span class="cn">TRUE</span>, <span class="at">best_only =</span> <span class="cn">TRUE</span>, </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">mc.cores =</span> threads</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plotMatchPos</span>(</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">abs =</span> <span class="cn">TRUE</span>, <span class="at">type =</span> <span class="st">"heatmap"</span>, </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">cluster =</span> <span class="cn">TRUE</span>, <span class="at">use_totals =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">binwidth =</span> <span class="dv">10</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Bin"</span>, <span class="at">fill =</span> <span class="st">"Total</span><span class="sc">\n</span><span class="st"> Matches"</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div style="font-size:80%;">
<ul>
<li>Clustering &amp; heatmaps can help identify redundancy</li>
<li>Useful for larger sets of motifs / matches</li>
</ul>
</div>
</div><div class="column" style="width:60%;">
<p><img data-src="assets/heatmap-pos-1.png"></p>
</div>
</div>
</section>
<section id="motiftestr-motif-enrichment" class="slide level2">
<h2>motifTestR: Motif Enrichment</h2>
<div style="font-size: 90%">
<div class="columns">
<div class="column" style="width:47%;">
<ol type="1">
<li><em>Hypergeometric</em>: Subset A vs Subset B
<ul>
<li>Specifically for sequences where one subset changes &amp; the other doesn’t</li>
<li>Realistically, results are specific to the each BG set</li>
</ul></li>
</ol>
<div class="fragment">
<ol start="2" type="1">
<li><em>Poisson</em>: SetA Vs BG
<ul>
<li>BG Set &gt;&gt;&gt; Set A</li>
<li>Estimates match rate per sequence in BG <span class="math inline">\(\implies\)</span> <code>poisson.test()</code></li>
<li>Quick(ish) &amp; dirty</li>
<li>Explicitly assumes matches are Poisson</li>
</ul></li>
</ol>
</div>
</div><div class="column" style="width:5%;">

</div><div class="column" style="width:47%;">
<div class="fragment">
<ol start="3" type="1">
<li><em>QuasiPoisson</em>: SetA Vs BG (In Blocks)
<ul>
<li>Tests match rate per <strong>block</strong> (i.e.&nbsp;iteration) in BG</li>
<li>Allows for overdispersion</li>
</ul></li>
</ol>
<p><br></p>
</div>
<div class="fragment">
<ol start="4" type="1">
<li><em>Iterations</em>: SetA Vs BG (In Blocks)
<ul>
<li>No distributional assumptions (except CLT)</li>
<li>Derives mean &amp; sd per iteration in BG<br><span class="math inline">\(\implies\)</span> <span class="math inline">\(Z\)</span>-score + p-value</li>
</ul></li>
</ol>
</div>
</div>
</div>
</div>
</section>
<section id="motiftestr-makermranges" class="slide level2">
<h2>motifTestR: <code>makeRMRanges()</code></h2>
<ul>
<li>Added function for obtaining <em>R</em>andom <em>M</em>atched <em>R</em>anges for a test set</li>
<li>Relies on a discrete set of features
<ul>
<li>e.g.&nbsp;promoters, enhancers, etc</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Samples RMRanges to exactly match the feature distribution of the test set
<ul>
<li>Can be sampled in blocks for iterative methods (quasipoisson / iteration)</li>
</ul></li>
</ul>
</div>
</section>
<section id="motiftestr-motif-enrichment-1" class="slide level2">
<h2>motifTestR: Motif Enrichment</h2>
<div class="columns">
<div class="column" style="width:70%;">
<ul>
<li>Is much, much slower… 😞</li>
</ul>
<div class="fragment" data-fragment-index="1">
<ul>
<li>7000 <span class="math inline">\(\times\)</span> 400bp <span class="math inline">\(\times\)</span> 400 motifs (quasipoisson):
<ul>
<li>Using 7000 <span class="math inline">\(\times\)</span> 1000 = 7x10<sup>6</sup> RMRanges</li>
<li>7 minutes (phoenix: 8 cores)</li>
</ul></li>
</ul>
</div>
<div class="fragment" data-fragment-index="2">
<ul>
<li>50,000 <span class="math inline">\(\times\)</span> 800bp <span class="math inline">\(\times\)</span> 400 motifs (quasipoisson):
<ul>
<li>Using 50,000 <span class="math inline">\(\times\)</span> 1000 = 5x10<sup>7</sup> RMRanges</li>
<li>1hr 35min minutes (phoenix: 8 cores)</li>
</ul></li>
</ul>
</div>
<div class="fragment" data-fragment-index="3">
<ul>
<li>400 Vs 6500 <span class="math inline">\(\times\)</span> 400bp <span class="math inline">\(\times\)</span> 400 motifs (hypergeometric):
<ul>
<li>35 sec (phoenix: 8 cores)</li>
</ul></li>
</ul>
</div>
</div><div class="column" style="width:30%;">
<div class="fragment" data-fragment-index="4">
<ul>
<li>This dog wants you to try it…</li>
</ul>
<p><br></p>
<iframe src="https://giphy.com/embed/6MWahPArixa6I" width="480" height="480" frameborder="0" class="giphy-embed" allowfullscreen="">
</iframe>
<p>
<a href="https://giphy.com/gifs/cute-dog-6MWahPArixa6I">via GIPHY</a>
</p>
</div>
</div>
</div>
</section>
<section>
<section id="and-now-my-real-talk" class="title-slide slide level1 center" data-background-color="#3D3D40">
<h1>And Now My Real Talk…</h1>

</section>
<section id="current-research" class="slide level2">
<h2>Current Research</h2>
<ul>
<li>PROPHECY (Preventing Renal OPthalmic and Heart Events in CommunitY)</li>
<li>Multi-Omics study led by Prof Alex Brown, with Chief Data Scientist A/Prof Jimmy Breen</li>
</ul>
<div class="fragment">
<ul>
<li>Investigating the very high rates of Diabetes within Indigenous Australians
<ul>
<li>Also very high rates of complications: CVD &amp; CKD</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Is strongly led by Community in partnership with BODL
<ul>
<li>Hoping to bring benefits of Precision Medicine</li>
<li>New approach to Indigenous Health <span class="fragment"><span class="math inline">\(\implies\)</span> listen first</span></li>
</ul></li>
</ul>
</div>
</section>
<section id="life-expectancy-gaps" class="slide level2">
<h2>Life Expectancy Gaps</h2>

<img data-src="assets/life_expectancy.png" class="r-stretch quarto-figure-center"><p class="caption">Courtesy of A/Prof Jimmy Breen, Black Ochre Data Labs, TKI</p></section>
<section id="mortality-from-diabetes" class="slide level2">
<h2>Mortality From Diabetes</h2>

<img data-src="assets/mortality.png" width="800" class="r-stretch"></section>
<section id="why" class="slide level2">
<h2>Why?</h2>
<ul>
<li>Taking all socio-economic indicators <span class="math inline">\(\implies\)</span> disease risk is still hugely increased for Indigenous Australians</li>
<li>Not only disease risk, but higher rates of profound complications</li>
</ul>
<div class="fragment">
<ul>
<li>Are there risk alleles?
<ul>
<li>If so, what are they? How do they contribute to risk?</li>
</ul></li>
<li>Are there protective alleles?</li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Is there an epigenetic contribution?</li>
<li>Additional environmental factors?</li>
</ul>
</div>
</section>
<section id="multi-omic-approaches" class="slide level2">
<h2>Multi-Omic Approaches</h2>

<img data-src="assets/prophecy.png" width="1000" class="r-stretch"></section>
<section id="the-transcriptomics-layer" class="slide level2">
<h2>The Transcriptomics Layer</h2>
<ul>
<li>Whole blood samples</li>
<li>Can we capture dynamics of disease / signatures / biomarkers etc?</li>
<li>Approach will be classical statistics + network approaches</li>
</ul>
<div class="fragment">
<ul>
<li>Tried to be excruciatingly careful with design:
<ul>
<li>plate layout, pooled replicates, globin-depletion, ERCC spike-ins, RIN scores etc</li>
</ul></li>
<li>Library preps are finally underway</li>
<li>Data lands around September</li>
</ul>
</div>
</section></section>
<section>
<section id="why-think-about-reference-genomes" class="title-slide slide level1 center" data-background-color="#3D3D40">
<h1>Why Think About Reference Genomes?</h1>

</section>
<section id="genetic-diversity" class="slide level2">
<h2>Genetic Diversity</h2>
<ul>
<li>Indigenous Australians are almost totally absent from repositories of genetic variation</li>
<li><span class="math inline">\(\sim\)</span> 25% of variation within Indigenous Australian populations is unique &amp; poorly characterised <span class="citation" data-cites="Easteal2020-fo">(<a href="#/references" role="doc-biblioref" onclick="">Easteal et al. 2020</a>)</span></li>
</ul>
<div class="fragment">
<ul>
<li>If contributing to disease risk and we use a standard reference <span class="fragment"><span class="math inline">\(\implies\)</span> will we see it?</span></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Can we bring the reference closer to our participants’ genomes?</li>
</ul>
</div>
<div class="fragment">
<ul>
<li>NCIG are very close to a pangenome graph assembled from Indigenous Australians
<ul>
<li>A little too far off of to be viable yet</li>
<li>Monica Guilhaus doing an amazing job figuring out spliced pan-transcriptome graphs <span class="citation" data-cites="Sibbesen2023-md">(<a href="#/references" role="doc-biblioref" onclick="">Sibbesen et al. 2023</a>)</span></li>
</ul></li>
</ul>
<aside class="notes">
<p>At least one member of our Indigenous Governance groups has said “Why are we even comparing ourselves to any other group. We should compare ourselves to us?</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="gene-level-approaches" class="slide level2">
<h2>Gene-Level Approaches</h2>
<ul>
<li><em>STARconsensus</em> <span class="citation" data-cites="Kaminow2022-dz">(<a href="#/references" role="doc-biblioref" onclick="">Kaminow et al. 2022</a>)</span> enables inclusion of a population-specific consensus variant set
<ul>
<li>Also possible for personalised genomes</li>
<li>Now has a <code>diploid</code> mode for phased, personalised variant sets</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Incorporates variants during indexing
<ul>
<li>Indexes twice (Standard + Modified)</li>
<li>Returns alignment co-ordinates relative to the unmodified reference</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Simple to obtain counts using standard methods (e.g.&nbsp;featureCounts)</li>
</ul>
</div>
</section>
<section id="the-downside" class="slide level2">
<h2>The Downside</h2>
<ul>
<li>Locked into using STAR
<ul>
<li>In my hands can’t align full-length, reference-derived transcripts</li>
</ul></li>
<li>Counts are gene-level</li>
</ul>
<div class="fragment">
<ul>
<li>We understand the distribution of counts at the gene level
<ul>
<li><code>edgeR</code> &amp; <code>DESeq2</code>: well established Negative Binomial methods</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Transcript-level counts have additional uncertainty
<ul>
<li>Need to be scaled by an over-dispersion estimate<br><span class="math inline">\(\implies\)</span> can then be treated as Negative Binomial <span class="citation" data-cites="Baldoni2024-zf">(<a href="#/references" role="doc-biblioref" onclick="">Baldoni et al. 2024</a>)</span></li>
<li>Cannot obtain these from <code>STAR</code> + <code>featureCounts</code></li>
</ul></li>
</ul>
</div>
</section>
<section id="transcript-level-analysis" class="slide level2">
<h2>Transcript-Level Analysis</h2>
<ul>
<li><code>salmon</code> provides transcript-level counts with overdispersion estimates!</li>
<li>Can we modify a reference transcriptome using a set of variants?</li>
</ul>
<div class="fragment">
<p><br><br></p>
<iframe src="https://giphy.com/embed/3ohs2xhB3jP6XfJ4ME" width="480" height="265" frameborder="0" class="giphy-embed" allowfullscreen="">
</iframe>
<p>
<a href="https://giphy.com/gifs/colbertlateshow-late-show-the-with-stephen-colbert-3ohs2xhB3jP6XfJ4ME">via GIPHY</a>
</p>
<p><img data-src="https://smped.github.io/transmogR/reference/figures/transmogR.png" class="absolute" style="bottom: 50px; right: 100px; width: 400px; "></p>
</div>
</section>
<section id="transmogr-salmon" class="slide level2">
<h2>transmogR + salmon</h2>
<ul>
<li>Takes about 15-20 mins, even on a laptop</li>
<li>Modified transcripts can be optionally tagged:
<ul>
<li>Indicating variant types (<code>s/i/d</code>)</li>
<li>Naming the variant set (e.g.&nbsp;<code>panhuman</code>)</li>
</ul></li>
<li>Generally using SNVs + InDels &lt; 50bp</li>
</ul>
<div class="fragment">
<ul>
<li>I’ve been using a modified transcriptome &amp; modified genome as the <em>gentrome</em>
<ul>
<li>The genome is used by <code>salmon</code> as decoy sequences <span class="citation" data-cites="Srivastava2020-tm">(<a href="#/references" role="doc-biblioref" onclick="">Srivastava et al. 2020</a>)</span></li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Indexing for <code>salmon</code> still takes a while…</li>
</ul>
<aside class="notes">
<p>Structural variants really need a new approach like the graph-based ones</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section></section>
<section>
<section id="dbpaint-developing-best-practices-for-the-analysis-of-indigenous-transcriptomes" class="title-slide slide level1 center" data-background-color="#3D3D40">
<h1>DBPAInT: Developing Best Practices for the Analysis of Indigenous Transcriptomes</h1>
<aside class="notes">
<p>This is very much work in progress with some results coming in last night</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="the-approach" class="slide level2">
<h2>The Approach</h2>
<ul>
<li>STARconsensus: using variants from the wrong population still improves performance <span class="citation" data-cites="Kaminow2022-dz">(<a href="#/references" role="doc-biblioref" onclick="">Kaminow et al. 2022</a>)</span></li>
</ul>
<div class="fragment">
<ol type="1">
<li>1000 Genomes Project Panhuman
<ul>
<li>alleles found in &gt;50% of unrelated across entire project</li>
</ul></li>
</ol>
</div>
<div class="fragment">
<ol start="2" type="1">
<li>1000 GP AFR Subpopulation
<ul>
<li>alleles found in &gt;50% of unrelated within the AFR population</li>
<li>most distinct population from the Panhuman</li>
</ul></li>
</ol>
</div>
<div class="fragment">
<ol start="3" type="1">
<li>PROPHECY consensus set
<ul>
<li>alleles found in &gt; 50% of unrelated members of the study</li>
<li>Used 3 familial steps to define ‘unrelated’</li>
</ul></li>
</ol>
</div>
</section>
<section id="the-analysis" class="slide level2">
<h2>The Analysis</h2>
<ul>
<li>6 female pilot samples
<ul>
<li>All genomes are y-masked</li>
</ul></li>
<li>Replicating elements of STARconsensus</li>
<li>Taking GRCh38 as the worst case <span class="math inline">\(\implies\)</span> changes are improvements?
<ul>
<li>Four sets of genomic alignments / sample</li>
<li>Personalised variants arriving in the next few days (weeks?)</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Also running a <code>transmogR</code> + <code>salmon</code> analysis
<ul>
<li>Four sets of transcriptomic alignments / sample</li>
<li>Haven’t figured out how to do a diploid transcriptome yet…</li>
</ul></li>
</ul>
<aside class="notes">
<p>Personalised references give the closest to ground truth we can realistically get</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="the-variants" class="slide level2">
<h2>The Variants</h2>

<img data-src="assets/ol-all-var-1.png" class="r-stretch"><aside class="notes">
<ul>
<li>High levels of common variation <span class="math inline">\(\implies\)</span> even the wrong variant set should be an improvement</li>
<li>AFR &amp; PROPHECY have the most uniqueness, but PROPHECY appears closer to the panhuman set</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="prophecy-variants-overlapping-exons" class="slide level2">
<h2>PROPHECY Variants Overlapping Exons</h2>

<img data-src="assets/prophecy-by-trans-1.png" class="r-stretch"><aside class="notes">
<p>Taking the variants which overlap exons, how many transcripts are impacted</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="key-questions" class="slide level2">
<h2>Key Questions</h2>
<ol type="1">
<li>Are technical metrics improved?</li>
<li>Does it really make a difference?</li>
<li>Does any difference actually capture the underlying biology?</li>
</ol>
<div class="fragment">
<ul>
<li>STAR is deterministic for given reference
<ul>
<li>Change the reference even slightly <span class="math inline">\(\implies\)</span> no guarantees</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li><code>salmon</code> is not deterministic</li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Some of the changes will just be noise, but <em>how much is noise</em>?</li>
</ul>
</div>
</section>
<section id="technical-analyses" class="slide level2">
<h2>Technical Analyses</h2>
<ul>
<li>No real change to number of reads aligning (<code>STAR</code> or <code>salmon</code>)</li>
<li>Very noticeable changes within alignments</li>
<li>(Nearly) all technical metrics appear to be supportive of a variant modified reference</li>
</ul>
<div class="fragment">
<ul>
<li>First step is to check uniquely aligned reads from STARconsensus…</li>
</ul>
</div>
</section>
<section id="uniquely-aligning-reads-starconsensus" class="slide level2">
<h2>Uniquely Aligning Reads (STARconsensus)</h2>
<div class="columns">
<div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="assets/plot-nett-unique-1.png"></p>
<figcaption>All variant sets show an increase in the number of uniquely-aligned reads (10-30,000)</figcaption>
</figure>
</div>
</div><div class="column">
<div class="fragment">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="assets/compare-matching-bases-1.png"></p>
<figcaption>PROPHECY shows a very slight increase in the number of CIGAR M-bases per uniquely-aligned read</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="however" class="slide level2 smaller">
<h2>However…</h2>
<div class="columns">
<div class="column" style="width:65%;">
<p><img data-src="assets/barplot-all-changes-1.png"></p>
</div><div class="column" style="width:35%;">
<ul>
<li>About 1 in 300 reads moves</li>
<li>About 30-40,000 reads remain uniquely aligned but in a different location</li>
</ul>
<div class="fragment">
<ul>
<li>How much is noise?</li>
<li>Are the changes shared between samples?</li>
<li>Do the changed reads overlap a variant?</li>
<li>Where exactly are they moving <em>from</em> and <em>to</em>?</li>
</ul>
</div>
</div>
</div>
</section>
<section id="reads-overlapping-variants" class="slide level2">
<h2>Reads Overlapping Variants</h2>
<div class="columns">
<div class="column" style="width:64%;">
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="assets/var-alluvial-1.png"></p>
</figure>
</div>
</div><div class="column" style="width:36%;">
<p>Taking reads where gapped alignments overlap a variant</p>
<ul>
<li>chr12 to chr1 shared by all</li>
<li>Also chr9 to chr7</li>
<li>chrX to chr19 shared by 1, 2, 6</li>
<li>chr12 to chr16 shared by 2 <span class="math inline">\(\rightarrow\)</span> 5</li>
</ul>
</div>
</div>
<aside class="notes">
<p>These strike me as patterns which support the underlying biology &amp; diversity within participants</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="reads-overlapping-variants-1" class="slide level2">
<h2>Reads Overlapping Variants</h2>
<div class="columns">
<div class="column" style="width:60%;">
<p><img data-src="assets/alluvial-gene-type-1.png" width="600"></p>
</div><div class="column" style="width:40%;">
<p>Taking all samples combined</p>
<ul>
<li>Increase in alignments to protein coding genes</li>
<li>Decrease in pseudogenes</li>
</ul>
</div>
</div>
</section>
<section id="common-genes-being-impacted" class="slide level2">
<h2>Common Genes Being Impacted</h2>
<div class="columns">
<div class="column" style="width:65%;">
<p><img data-src="assets/individual-genes-1.png"></p>
</div><div class="column" style="width:35%;">
<p>Taking the 20 most common shifts</p>
<ul>
<li>HLA-A</li>
<li>HLA-H</li>
<li>Ribosomal proteins</li>
<li>Histone Proteins</li>
</ul>
</div>
</div>
</section>
<section id="quick-summary" class="slide level2">
<h2>Quick Summary</h2>
<ul>
<li>These are only the uniquely-aligned reads from STARconsensus
<ul>
<li>Don’t overlap alignment positions in either reference</li>
<li>Haven’t looked at unmapped <span class="math inline">\(\rightarrow\)</span> mapped, or multi-mappers</li>
</ul></li>
<li>Looks like it may be consistent &amp; true biology?</li>
<li>I got these results Wed night so I’m still processing them mentally</li>
</ul>
</section>
<section id="differential-gene-expression-analysis" class="slide level2">
<h2>Differential Gene Expression Analysis</h2>
<div class="columns">
<div class="column" style="width:60%;">
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="assets/plot-ma-1.png"></p>
</figure>
</div>
</div><div class="column" style="width:40%;">
<ul>
<li>GRCh38 vs PROPHECY-modified</li>
<li>Identical input reads</li>
<li>All changes are alignment induced</li>
</ul>
</div>
</div>
</section>
<section id="genes-impacted-variably" class="slide level2 smaller">
<h2>Genes Impacted Variably</h2>
<div class="columns">
<div class="column" style="width:60%;">
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="assets/plot-variable-1.png"></p>
</figure>
</div>
</div><div class="column" style="width:40%;">
<p>Taking expression estimates within each reference:</p>
<ol type="1">
<li>Find the difference in counts per gene &amp; sample</li>
<li>Take mean &amp; sd of differences</li>
</ol>
<div class="fragment">
<ul>
<li>Consistent changes will be low-variability, far from zero</li>
<li>Inconsistent changes high-variability but near zero</li>
<li>Like looking through an MA plot with variability as 3<sup>rd</sup> dimension</li>
</ul>
</div>
</div>
</div>
</section></section>
<section>
<section id="what-about-salmon-transmogr" class="title-slide slide level1 center" data-background-color="#3D3D40">
<h1>What about <code>salmon</code> + <code>transmogR</code></h1>

</section>
<section id="key-differences-to-star" class="slide level2">
<h2>Key Differences to STAR</h2>
<ul>
<li>Only ~30% of reads align uniquely
<ul>
<li>Largest NH:i tag is ~24,000</li>
<li>90% of reads align with NH:i &lt; 10</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>BAM files directly connect read ID to transcript
<ul>
<li>Makes comparison much simpler</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>How do the <em>transcript-level</em> counts change?</li>
<li>How do the <em>overdispersion</em> estimates change?</li>
</ul>
</div>
</section>
<section id="salmon-metrics" class="slide level2">
<h2>Salmon Metrics</h2>
<ul>
<li>Increase in Properly-Paired alignments (+3000)</li>
<li>PROPHECY has strongest increase in correctly oriented fragments (+10,000 ISF)</li>
<li>Decrease in umapped reads (-2000)</li>
</ul>
<div class="fragment">
<ul>
<li>Minor change in library sizes</li>
<li>Similar numbers of detected transcripts (&gt;1 count)</li>
</ul>
</div>
</section>
<section id="overdispersion-estimates" class="slide level2">
<h2>Overdispersion Estimates</h2>

<img data-src="assets/compare-overdispersions-1.png" class="r-stretch quarto-figure-left"><p class="caption">Looking at how the number of transcripts / gene also impacts these</p></section>

<section id="scaled-transcript-level-counts" class="slide level2">
<h2>Scaled Transcript-Level Counts</h2>

<img data-src="assets/compare-trans-counts-1.png" class="r-stretch quarto-figure-left"><p class="caption">This seems quite acceptable</p></section>

<section id="variable-transcripts" class="slide level2">
<h2>Variable Transcripts</h2>

<img data-src="assets/trans-variable-1.png" class="r-stretch quarto-figure-left"></section>
<section id="tracking-individual-reads" class="slide level2">
<h2>Tracking Individual Reads</h2>
<ul>
<li>Can easily track a read &amp; all it’s transcripts</li>
<li>Subset data to reads with NH:i <span class="math inline">\(\leq 10\)</span></li>
<li>Subset reads to those aligning to a modified transcript either before or after</li>
</ul>
<div class="fragment">
<ul>
<li>Unique-mappers are far less informative for transcript-level alignments</li>
<li>Haven’t checked for moves between chromosomes
<ul>
<li>Transcript &amp; Gene-Level moves are more fine-grained</li>
</ul></li>
</ul>
</div>
</section>
<section id="tracking-reads-which-change-transcripts" class="slide level2">
<h2>Tracking Reads Which Change Transcripts</h2>

<img data-src="assets/salmon-moved-within-1.png" class="r-stretch quarto-figure-left"><p class="caption">Reads changing transcript <strong>within</strong> a gene</p><aside class="notes">
<p>The consistency between samples is very encouraging</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="tracking-reads-which-change-transcripts-1" class="slide level2">
<h2>Tracking Reads Which Change Transcripts</h2>

<img data-src="assets/salmon-moved-across-1.png" class="r-stretch quarto-figure-left"><p class="caption">Reads changing transcript to different genes</p><aside class="notes">
<ul>
<li>Changes are again very consistent between samples</li>
<li>Some similarity to STAR changes, but not the pseudogene story!!!</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="changing-transcripts-gene-level-categories" class="slide level2">
<h2>Changing Transcripts: Gene-Level Categories</h2>

<img data-src="assets/salmon-moved-gene-alluvial-1.png" class="r-stretch quarto-figure-left"><p class="caption">Shows a nice decrease in reads mapping nowhere</p><aside class="notes">
<p>A small increase in protein coding genes for some samples</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="changing-transcripts-transcript-level-categories" class="slide level2">
<h2>Changing Transcripts: Transcript-Level Categories</h2>

<img data-src="assets/salmon-moved-trans-alluvial-1.png" class="r-stretch quarto-figure-left"><p class="caption">Lots of change between but no categories really sing out</p></section>
<section id="where-to-from-here" class="slide level2">
<h2>Where To From Here</h2>
<ul>
<li>Personalised Variant sets will give us the best estimate of ground-truth
<ul>
<li>Will run STARconsensus in Diploid mode</li>
<li>Haven’t figured out how to do this for <code>transmogR</code> + <code>salmon</code></li>
<li>Will just choose randomly from Het sites (???)</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Running a mini 90-sample cohort to see how it really plays at scale
<ul>
<li>Previously analysed by Prof Sam El-Osta’s team (Monash)</li>
<li>Have a specific subset of complications</li>
</ul></li>
<li>How will WGCNA modules be impacted?</li>
<li>How will true DGE / DTE results be impacted?</li>
</ul>
<aside class="notes">
<p>Liza is also running de novo transcriptome assemblies on these samples</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="acknowledgements" class="slide level2 smaller">
<h2>Acknowledgements</h2>
<div class="columns">
<div class="column" style="width:30%;">
<h4 id="black-ochre-data-labs"><u>Black Ochre Data Labs</u></h4>
<p>Alex Brown<br>
<strong>Jimmy Breen</strong><br>
Sam Buckberry<br>
<strong>Yassine Souilmi</strong><br>
Bastien Llamas<br>
Katharine Browne<br>
<strong>Liza Kretzschmar</strong><br>
Alastair Ludington<br>
Holly Massacci<br>
<strong>Sam Godwin</strong><br>
Kaashifah Bruce<br>
Rebecca Simpson<br>
Sarah Munns<br>
Ashlee Thomson</p>
</div><div class="column" style="width:30%;">
<h4 id="tki-align"><u>TKI / ALIGN</u></h4>
<p>Johanna Barclay<br>
Amanda Richards-Satour<br>
Justine Clark<br>
Rose Senesci<br>
Analee Stearne<br>
Louise Lyons<br>
Dawn Lewis<br>
Mary Brushe<br>
Karrina DeMasi<br>
Phoebe McColl</p>
<h4 id="ncig"><u>NCIG</u></h4>
<p>Hardip Patel</p>
</div><div class="column" style="width:30%;">
<h4 id="sahmri"><u>SAHMRI</u></h4>
<p><strong>Tash Howard</strong><br>
Marlie Frank</p>
<h4 id="sagc"><u>SAGC</u></h4>
<p>Sen Wang<br>
Paul Wang<br>
Renee Smith<br>
<br></p>
<h4 id="university-of-adelaide"><u>University of Adelaide</u></h4>
<p>Lachlan Baer<br>
<strong>Monica Guilhaus</strong><br>
Wenjun (Nora) Liu<br>
Megan Monaghan</p>
</div>
</div>
<p><img data-src="https://d1blp3aor2iuhp.cloudfront.net/images/logos/logo-ANU-black.svg" class="absolute" style="left: 350px; bottom: 10px; width: 200px; "></p>
<p><img data-src="assets/TKI.jpg" class="absolute" style="left: 0px; bottom: 10px; width: 200px; "></p>
</section>
<section id="references" class="slide level2 smaller scrollable">
<h2>References</h2>

<img src="assets/bodl_logo_white_background.jpg" class="slide-logo r-stretch"><div class="footer footer-default">

</div>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-Baldoni2024-zf" class="csl-entry" role="listitem">
Baldoni, Pedro L, Yunshun Chen, Soroor Hediyeh-Zadeh, Yang Liao, Xueyi Dong, Matthew E Ritchie, Wei Shi, and Gordon K Smyth. 2024. <span>“Dividing Out Quantification Uncertainty Allows Efficient Assessment of Differential Transcript Expression with edgeR.”</span> <em>Nucleic Acids Res.</em> 52 (3): e13.
</div>
<div id="ref-Easteal2020-fo" class="csl-entry" role="listitem">
Easteal, Simon, Ruth M Arkell, Renzo F Balboa, Shayne A Bellingham, Alex D Brown, Tom Calma, Matthew C Cook, et al. 2020. <span>“Equitable Expanded Carrier Screening Needs Indigenous Clinical and Population Genomic Data.”</span> <em>Am. J. Hum. Genet.</em> 107 (2): 175–82.
</div>
<div id="ref-Kaminow2022-dz" class="csl-entry" role="listitem">
Kaminow, Benjamin, Sara Ballouz, Jesse Gillis, and Alexander Dobin. 2022. <span>“Pan-Human Consensus Genome Significantly Improves the Accuracy of <span class="nocase">RNA-seq</span> Analyses.”</span> <em>Genome Res.</em> 32 (4): 738–49.
</div>
<div id="ref-Sibbesen2023-md" class="csl-entry" role="listitem">
Sibbesen, Jonas A, Jordan M Eizenga, Adam M Novak, Jouni Sirén, Xian Chang, Erik Garrison, and Benedict Paten. 2023. <span>“Haplotype-Aware Pantranscriptome Analyses Using Spliced Pangenome Graphs.”</span> <em>Nat. Methods</em>, January.
</div>
<div id="ref-Srivastava2020-tm" class="csl-entry" role="listitem">
Srivastava, Avi, Laraib Malik, Hirak Sarkar, Mohsen Zakeri, Fatemeh Almodaresi, Charlotte Soneson, Michael I Love, Carl Kingsford, and Rob Patro. 2020. <span>“Alignment and Mapping Methodology Influence Transcript Abundance Estimation.”</span> <em>Genome Biol.</em> 21 (1): 239.
</div>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="20240503_transmogR_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="20240503_transmogR_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="20240503_transmogR_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="20240503_transmogR_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="20240503_transmogR_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="20240503_transmogR_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="20240503_transmogR_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="20240503_transmogR_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="20240503_transmogR_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="20240503_transmogR_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1280,

        height: 720,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    <script type="text/javascript">
      Reveal.on('ready', event => {
        if (event.indexh === 0) {
          document.querySelector("div.has-logo > img.slide-logo").style.display = "none";
        }
      });
      Reveal.addEventListener('slidechanged', (event) => {
        if (event.indexh === 0) {
          Reveal.configure({ slideNumber: null });
          document.querySelector("div.has-logo > img.slide-logo").style.display = "none";
        }
        if (event.indexh === 1) {
          Reveal.configure({ slideNumber: 'c' });
          document.querySelector("div.has-logo > img.slide-logo").style.display = null;
        }
      });
    </script>
    

</body></html>