---
title: "ALIGN Capacity Building"
subtitle: "The Why, What and How of Transcriptomics"
author: "Dr Stevie Pederson (They/Them)"
affiliation: "Black Ochre Data Labs"
date: "2023-11-07"
date-format: long
title-slide-attributes:
    data-background-color: "#3c3c44"
    data-background-opacity: "20%"
    data-background-size: "35%"
    data-background-position: "50% 50%"
editor: source
format: 
  revealjs:
    bibliography: references.bib
    theme: [custom.scss]
    code-line-numbers: false
    width: 1280
    height: 720
    sansfont: Times New Roman
    logo: assets/bodl_logo_white_background.jpg
    slide-number: c/t
knitr: 
  opts_chunk: 
    echo: true
    include: true
    warning: false
    message: false
    fig.align: left
---

##

::: {.columns}

::: {.column width="35%"}

### Acknowledgement of Country

:::

::: {.column width="65%"}

I would like to acknowledge that I'm presenting today from Kaurna Country.<br><br>

I acknowledge the deep feelings of attachment and relationship of the Kaurna people to their Place.<br><br>

I also pay my respects to the cultural authority of Aboriginal and Torres Strait Islander peoples from other areas of Australia online today, and pay my respects to Elders past, present and emerging. 


:::

:::

::: {.notes}
I also wish to express how grateful I am for the opportunity to be involved in this space.
I'm acutely aware of the trust that has been placed in me and am very pleased to be surrounded by so many wise & experienced researchers
:::

## Who Am I? 

- Oldest of the nerds in Black Ochre Data Labs but still an ECR

. . .

- Unintentionally took the "winding path" approach to my PhD (2008-2018)
    + Undertaken in a T Cell  / Autoimmunity Research Group
    + Developed a novel Bayesian Statistical model for alternate transcript detection (using microarrays)

. . . 

- Coordinator of the University of Adelaide Bioinformatics Hub (2014-2020)
    + Analysed dozens of RNA-Seq datasets (+ more)
    + Developed multiple training workshops and undergrad/postgraduate courses

. . .     
    
- Massive R nerd (5 Bioconductor packages)
    

::: {.notes}
- I had a career in music before retraining through my 30s and 40s
- PhD was working in a T cell group looking at autoimmune disease
- Developed a Bayesian model for detection of alternate transcript expression
- Am the lead developer for two Bioconductor packages and have contributed to 3 more
- Currently sit on the Community Advisory Boards for Bioconductor
- Also working on my 6th package
:::

## Who Am I? 

- First discovered microarrays and R in 2002
    + Most statistical understanding has been learned through this lens
    + Very different to epidemiology, study design etc
    + Primary training is in biology (e.g. genetics, biochemistry)

. . .

- Majority of experience in bulk tissue (i.e. not single-cell)

. . .

- Also spent two years researching ER-dependent Breast Cancers
    + Looking at Transcription Factors and DNA state (ChIP-Seq)
    + Data Integration (RNA-Seq, ChIP-Seq, HiC etc.)

::: {.notes}
- First exposure to bioinformatics was an undergraduate summer scholarship in summer of 2002-2003
- Statistical training has been very different to epidemiology & study design
:::

# Why Transcriptomics? {background-color="#3c3c44"}

## Why?

- Transcriptomics is essentially the analysis of RNA
- High-throughput viewpoint
- Provides insight into a **highly dynamic** component of biology
- Long history of identifying biomarkers and establishing mechanisms
    - Is essentially a hypothesis-generating step
    - Often need to test hypotheses using conventional experimental approaches
    
## Why?

PROPHECY $\implies$ Preventing Renal OPHtalmic Events in CommunitY

. . .

- Key focus is Type 2 Diabetes and complications (CVD and CKD)

. . .

- Multiple 'omics layers: DNA (Genomics), DNA-methylation (Epigenomics), protein (Proteomics), metabolites (Metabolomics) etc

. . .

- Whole blood $\implies$ large immune component ("white blood cells")
- Can we see markers of the different stages of Type 2 Diabetes <br>(i.e. complications developing)
    
. . . 

- By integrating with other -omics layers:
    + Can we refine our hypotheses into underlying mechanisms? <br>[$\implies$ clinical intervention]{.fragment .fade-in}
    
::: {.notes}
- I'm primarily responsible for the analysis of the transcriptomics layer of PROPHECY
    + Anelle, Tash Howard, Jimmy, Alex etc all play pivotal roles as well (sample management, sample collection, finalising protocols, data security etc)
- Also have lipidomics, metabolomics, retinal scans, ECGs, inflammatory markers
- Shall we call retinal scans retinomics?
:::
    
# How Does Biology Even Work? {background-color="#3c3c44"}

::: {.notes}
- If only any of us could answer this question
- As soon as we think we understand something, we find the exception
- So much redundancy built-in + so much complexity
- To understand transcriptomics, we do need to spend a little little looking at the things around transcripomics which can provide a strong context
:::

## DNA

::: {.columns}

::: {.column width='56%'}

![Image created using www.biorender.com](assets/cell_structure.png){width="1000px"}

:::

::: {.column width='44%'}

- Most are familiar with DNA and cells
- DNA stays in the nucleus
- Directly inherited from parents<br>$\implies$ unique people
- The same in every cell [(roughly)]{.fragment .fade-in}

::: {.fragment}

- Generally static [(until cell division)]{.fragment .fade-in}

:::

::: {.fragment}
- Lots of interaction with other molecules: DNA-methylation, Histones etc.
:::

:::


:::

::: {.notes}
- Most of us a familiar with DNA and the idea that our bodies are made of trillions of cells (28-36 trillion)
- We all have our own unique DNA, passed down from our parents
- DNA lives in the nucleus of most cells (not RBC)
- Chromosomes replicate and divide with cell division
- Is generally static, but a huge amount happens to it & around it
    + Static in terms of the sequence doesn't change and the molecule is heavily protected and preserved
:::

## DNA

::: {.columns}

::: {.column width="55%"}
    
![Made using www.biorender.com](assets/dna_breakdown.png){fig-align="left"}

:::

::: {.column width="45%"}

Each chromosome is a string of millions of nucleotides

- 4 nucleotides (or bases): A, C, G, T
- Double-stranded molecule, i.e. double helix
    + Adds incredible stability 
    + Provides error handling mechanism
- Amazing amount of 3D structure & packaging

::: {.fragment}
- Only forms the classic ![](assets/sister_chromatids.jpg){width="4%"} shape during cell division
:::

:::

:::

::: {.notes}
- DNA is made of a sugar-phosphate backbone with different baes (or nucleotides) attached
    + Full names for bases Adenine, Cytosine, Guanine, Thymine
- Heavily packaged &#8658; all chromosomes in a human cell would be about 2m long if stretched out
- The sequence is static, but a lot happens to the DNA and to package the DNA
- Essentially the same static molecule having things done to it
:::

## DNA

We often think of there being [**one unique, ideal reference sequence**]{.underline}

- This is very far from reality 
- Each copy of Chromosome 1 is different to the other copy within each of us
    + Both copies are (essentially) the same between all of our cells
    + No value judgements on similarity to reference $\implies$ just who we are
    
. . .    
    
- Same for all other chromosomes (1-22 + X)
- Scale this out to billions of people...
- Some regions are relatively invariant (needed for survival)
    
## DNA
    
- Unique combinations of bases are ours and ours alone, for each of us
    + Inherited from parents (+ a few spontaneous changes)
    
. . .    
    
- We often refer to these unique sequences as *variants*
    + Single nucleotides (SNVs)
    + Missing or extra nucleotides (InDels) 
    + Larger structural variants (SVs) and repetitive elements (REs)
    
. . .    
    
- Collections of variants within a population $\implies$ *genetic diversity*

::: {.notes}
- There are large runs of DNA which rarely change between individuals
- I have a mutation in one of my haemoglobin not shared by either parent
    + My HbA1c cannot be measured, whereas both of theirs can
    + Was communicated to me as a reportable discovery during a public health study (NWHS, 20 years ago)
    + My blood responds poorly to low-oxygen situations: like respiratory infections
    + Also no plans to climb Mt Everest
    
[Click]
    
- Having "variants" implicitly ties back to this idea of a single reference
    + Not mutations at all!!! Just diversity
:::



## The Central Dogma of Biology

::: {.columns}

::: {.column width="40%"}
<br>
<iframe src="https://giphy.com/embed/mRI1hW0ZBVUly" width="450px" height="450px" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/biology-primer-mRI1hW0ZBVUly">via GIPHY</a></p>

:::

::: {.column width="60%"}

DNA is essentially a giant book of instructions


- Some regions are known as genes
    + *Transcribed* into RNA
    + Many RNA types (mRNA, miRNA, ncRNA, rRNA etc.)
    

::: {.fragment}
    
- mRNA molecules are *translated into proteins*
    + Groups of three bases encode a single amino acid

:::

::: {.fragment}

- Proteins *do stuff* i.e. pretty much everything
   + Keratin $\implies$ Hair structure
   + Haemoglobin $\implies$ Oxygen transport
   + Antibodies $\implies$ Detect "alien" molecules
   
:::

:::

:::

::: {.notes}
- DNA can almost be considered as a giant set of personalised instructions
- Most regions have some kind of function $\implies$ some help determine structure
    + Others are transcribed
- Transcribed regions (i.e. RNA) is known as the **Transcriptome**
- Groups of 3 bases are known as codons and these code for amino acids
    + Amino acids are key structural element of proteins
    + Each amino acid can have multiple codons (or codes)
- Also enzymes which breakdown food are proteins
- The initial idea of the Central Dogma (Crick, 1956) relates to information flow, not that RNA must be converted into proteins
:::

## Proteins

Proteins are the molecular machines that run our bodies

. . .

- Incredibly dynamic system
    + Some are stable $\implies$ some degrade rapidly
    + Some exist only within cells $\implies$ some are exported/imported
    
. . .

- Each cell-type (e.g. neuron, skin cell) will make a different set of proteins
    + Some structural proteins may be common between unrelated cell-types

. . .

- Proteins can be heavily modified
    + Phosphorylation, Sumoylation, Ubiquitination etc.
    + Can be single molecules or bind to multiple partners
- Different functions or activities for every state


::: {.notes}
- Proteins DO THINGS
    + Stomach enzymes like pepsin digest other proteins, others like lactase cut sugars
    + Cell surface receptors (ACE2)
- Another key feature of proteins which sets them apart from DNA & RNA is their role outside cells
    + The relationship bertwen a protein and the cell-of-origin is less clear than nucleic acids
- A common structural protein is Actin
- By dogma, Nucleic Acids don't exist outside cells (apart from when they do)
    + dicistronic tRNA-mRNA molecules are extracellular signalling molecules in plants
:::

## Proteins

*Proteomics* is the study of proteins

. . . 

- Is very limited technique
    + Tens of Thousands exist 
    + Can usually only identify hundreds (maybe 1-2000) from a sample
    
. . .    
    
- We don't understand much of the complexity
    + Single modifications can have huge consequences

. . .

- Compared to DNA/RNA sequencing is a very immature field
    + Technology is extremely variable and rapidly developing

::: {.notes}
- Proteomics is still really immature but can still shed important light on underlying biology
- Most proteomic data is produced using Mass Spectrometry
    + Fragments the proteins and we detect spectra mass on mass & charge spectra
    + Still no "one" technology that dominates the field
- Some tissues are dominated by a small set of proteins (e.g. albumin in blood is >70% of all protein)
- Being difficult study makes it harder for us to study $\implies$ we often look elsewhere that's easier
- This is where transcriptomics comes in
:::

# What is Transcripomics? {background-color="#3c3c44"}

## Transcriptomics

*Transcriptomics is the study of RNA molecules within a cell or tissue*

. . .

- Not all genes are transcribed (i.e. expressed) in every cell
    + ~60,000 annotated genes in the genome
    + ~10-15,000 detectable genes in a given cell-type
    
. . .

- Transcriptomes are *cell-type specific*
    + Related cell-types will often have great similarity, e.g. T-cell sub-types
    
. . .

- Often take a cell-line (or cell-type) and expose to a treatment
    + Low variability $\implies$ treatment response is easy(ish) to spot
    
## Transcriptomics

- Transcriptomics is often an abundance-focussed analysis
    + Represents a *stable state* of a cell
    + Differential Gene Expression [$\implies$ *abundance changes in response to stimuli*]{.fragment .fade-in}

. . .    
  
- Traditionally focussed on mRNA
- Common early assumption $\implies$ mRNA abundance reflects protein levels
    + Is sometimes true ...
    
    
::: {.notes}
- RNA molecules which are translated into proteins are generally referred to as messenger RNA (mRNA)
- Despite many mRNA representing an intermediary state between DNA & protein, this reveals a lot of information about the cell
    + e.g. what it's responding to, how it's maintaining itself
- Some genes are transcribed without protein product
    + Enables rapid response if protein is needed quickly
- Great examples of rapid transcriptional response are in T cells which encounter their target antigen or see pro-inflammatory cytokines
    + Huge & rapid transcriptional consequences
:::
    
## Transcriptomics
    
Our classic viewpoint is DNA $\implies$ mRNA $\implies$ Protein

::: {.columns}
   
::: {.column width="55%"}

:::: {.fragment fragment-index=1}
    
Many other types of RNA play key roles

- miRNA bind to and degrade target mRNA

::::

:::: {.fragment fragment-index=2}

- lncRNA form highly complex structures <br>$\implies$can silence chromosomal regions (e.g. *XIST*)[^1]

::::

:::: {.fragment fragment-index=3}

- rRNA are key components of ribosomes

::::

:::: {.fragment fragment-index=4}

- tRNA interact with ribosomes <br>$\implies$ translate mRNA into proteins[^2]

::::

:::

::: {.column width="45%"}

:::: {.r-stack}

![](https://github.com/UofABioinformaticsHub/transcriptomics_applications/blob/master/lectures/figures/Xist.png?raw=true){.fragment width="85%" fragment-index=2 .fade-in-then-out}

![](https://cdn.pfizer.com/pfizercom/20191227_trna.gif){.fragment width="85%" fragment-index=5}

::::

:::

:::

[^1]: Taken from Fang et al., “Probing Xist RNA Structure in Cells Using Targeted Structure-Seq”
[^2]: Animation courtesy of www.pfizer.com

::: {.notes}
- Many of the functional RNA also provide vital insights, but these are often poorly understood
- We generally ignore Ribosomal RNA, and given they make up ~80% of RNA in a cell, we need to manage these
- tRNA are generally not interesting either (expect when forming dicistronic tRNA-mRNA)
:::

## What is an RNA transcript?

::: {.columns}

::: {.column width="40%"}


- Known transcribed regions are defined as genes
    + Transcribed from DNA $\implies$ start to finish

::: {.fragment}

- From the complete RNA transcript:
    + Exons form *mature transcript*
    + Introns are *spliced* out
    
:::
    
:::

::: {.column width="60%"}

![Image courtesy of National Human Genome Research Institute](assets/splicing2.png)

:::

:::

::: {.fragment}

- ~38% of genes (~24,000) are transcribed into multiple **transcripts**
- Different transcripts $\implies$ Different proteins (or lncRNA etc.)

:::

::: {.notes}
- This definition of a gene is quite different to the original *'unit of inheritance'*
- It's probably worth noting that from a DNA-structural perspective, introns & exons are just long sequences
    + Splice junctions are defined by core sequence motifs, not any 3D structure like appears here 
- This is really where the name transcriptomics has developed
- Represents far greater complexity than just gene-level abundances
- Some genes even have multiple start sites
- The key point is we have one gene, multiple transcripts, multiple proteins then multiple modifications of multiple proteins
- Transcriptomics reflects a key inflection point in the development of informational complexity
:::


# How Do We Study Transcriptomics {background-color="#3c3c44"}

## Early Gene Expression Approaches

::: {.columns}

::: {.column}

- The first high-throughput platform was Microarrays
    + 3' Arrays (Affymetrix)
- Gene-centric approach
    + Could analyse 10-15,000 genes
    + Abundance only analysis
- Many statistical tools developed in this context
    + `limma` has been maintained by Gordon Smyth (WEHI) for >20years
  
    
:::

::: {.column}

![Image courtesy of Affymetrix](https://s3.amazonaws.com/static.wd7.us/2/22/Affymetrix-microarray.jpg)

:::

:::

::: {.notes}
- The 20yrs younger nerd got excited by this technology
- Limma stands for linear models for microarrays
- Still heavily used in modern approaches
:::

## Microarrays

::: {.columns}

::: {.column}

![Image courtesy of Affymetrix](assets/affy_design.png)

Known sequences at known locations

:::

::: {.column}

::: {.fragment}

![Image courtesy of Affymetrix](https://s3.amazonaws.com/iphonecde/_GNCHIP3.GIF)

Labelled cDNA binds complementary target

:::

:::

:::

::: {.fragment}

- We analysed fluorescence NOT sequence data

:::

::: {.notes}

- Known sequences were placed at specific co-ordinates
- RNA was converted to cDNA (more stable) labelled with fluorophores
- cDNA binds to perfectly matching complementary sequence
- Laser excites fluorophores: brighter signal $\implies$ more original RNA
- Were indirectly measuring RNA abundance
- RNA degrades from one-specific end (i.e. 3') so probes targets 3' end of a gene for intact RNA
:::

## Bulk RNA Sequencing

- Modern approaches involve sequencing the transcriptome
    + Short Reads (Illumina) < 300bases: Quantitative
    + Long Reads (Nanopore, PacBio); Semi-Quantitative

. . .

- Short reads still dominate
    + Used in the PROPHECY transcriptomics layer

. . .
    
- Mature RNA transcripts may be short (8nt) or long (350,375nt)

::: {.notes}
- Bulk RNA-Seq really has become the more correct term now that we have single-cell RNA-seq
    + Just means we have RNA from multiple cells
- Transcribed RNA (pre-processing) can be up to 450,000 nucleotides
- Longest mature transcript is *XACT* (lncRNA)
- Shortest codes for a 2aa component of the T cell receptor
:::


## Bulk RNA Sequencing

- All cells from each sample (or tissue) are lysed and RNA extracted
- RNA is fragmented (250-500bp)
    + RNA Quality is assessed (RIN Score)
- RNA fragments are prepared for sequencing (library preparation)
    + Converted to cDNA
    + Add sequencing adapters
    + PCR amplification
    
. . .    
    
- Sequencing $\implies$ ~30-50m reads/sample

. . .

How do we put this all back together and quantify?

::: {.notes}
- A complete sequencing run is what we call a library and in RNA-Seq we usually have 30m to 50m reads
- Imagine chopping up the screen-plays to all Star Wars movies then having to figure out where it came from?
- "Use the Force" might be really hard to figure out, but "Use the Force Luke" might be easier
:::

## Bulk RNA Sequencing

![](assets/bd020139.f1.jpeg)

## Two Approaches To Alignment

::: {.columns}

::: {.column width="40%"}
![](assets/alignment.jpeg)
:::

::: {.column width="60%"}
1. Alignment to a reference genome
    - Most align to one location $\implies$ can see where <span style="color:#769b79;">&#10003;</span>
    - Brilliant for gene-level counts <span style="color:#769b79;">&#10003;</span>
    - Can call variants (allelic bias) <span style="color:#769b79;">&#10003;</span>
    - Not helpful for transcript-level <span style="color:red;">&#10007;</span>

::: {.fragment}

2. Alignment to a reference transcriptome
    - Obtain transcript & gene-level counts <span style="color:#769b79;">&#10003;</span>
    - Many exons shared with multiple transcripts <br>$\implies$ can model uncertainty <span style="color:#769b79;">&#10003;</span>
    - Don't know where each read has aligned <span style="color:red;">&#10007;</span>
        + No variant calling
:::

:::

:::

::: {.notes}
- Aligning to a linear reference uses conventional aligners (Burrows-Wheeler transform)
- Aligning to a transcriptome incorporates graph-based strategies
- Modern transcriptome alignments (salmon , kallisto) include uncertainty about which transcript a read comes from
    + Can manage this using modelling strategies
:::

## Limitations

- Both approaches use 1 reference sequence $\implies$ linear reference
- If 1 of each of our chromosomes matched exactly $\implies$ the other copy won't

. . .

- Current reference genome is GRCh38
    + Transcripts are sequences derived from this
    + Anchored to GRCh38 using co-ordinates
    
. . .    
    
- Includes 517 extra sequences (scaffolds, patches, haplotypes)
    + Do not exist in reality
    + Some genes have transcripts defined on scaffolds not chromosomes

::: {.notes}
- Reality is that no copies of either chromosome match the existing reference
    + 93% of GRCh38 is taken from 11 individuals
- GRC is the Genome Reference Consortium
    + Numbering didn't start at GRCh1 $\implies$ GRCh37 took over from hg19
    + hg1 was release in 2000
- Extra sequences are bits they know exist, but don't know what to do with
:::

## How Do We Manage Diversity?

Given that no real person matches the reference genome

. . .

- Can we improve the reference in general?
- Can we improve for a given study cohort

. . .

Two emerging approaches

. . .

1. Modify a linear reference using representative variants
2. Align to a *reference graph* which represents appropriate diversity

## Modifying a Reference

- Tools exist for modifying a reference genome
    + Unable to analyse at the transcript level $\implies$ only gene-level
    
. . .

- BODL is developing the software to produce a variant-modified transcriptome
    + Will enable counts at the transcript-level + uncertainty measures

. . .

- Current testing uses 1000 Genomes Project variants
    + Shown to significantly improve mappings in other datasets
    + Poorly representative of Australian Indigenous variation

::: {.notes}
- We're currently discussing how we manage this for the PROPHECY dataset
- How do we define improvement
- How do we manage data access between -omics layers
:::


## Graph Based Approaches

::: {.columns}

::: {.column width="55%"}

- Telomere-to-telomere (T2T) assemblies contain no scaffolds
    + Just chr1-22, X, Y & MT
- Individual assemblies $\implies$ *haplotype resolved*
    + Separate out both copies of chr1 etc.
    
::: {.fragment fragment-index=1}

- Can construct a reference graph
    + Shared sequences between chromosome pairs are joined
    + Variation represented as bubbles

:::

::: {.fragment fragment-index=2}
- For >1 individual would be a *pangenome graph*
:::

:::

::: {.column width="45%"}

::: {.fragment fragment-index=1}

![Sibbesen, J et al, *Haplotype-aware pantranscriptome analyses using spliced pangenome graphs* Nat Methods, 2023](assets/reference_graph.png){fig-align="left"}
:::

:::

:::

::: {.notes}
- Beyond my expertise of experience
- Currently have a student learning the tools
- Hardip is actively working in this space with NCIG
:::

## Graph Based Approaches

- A reference pan-genome graph is now available[^3]
    + Contains 47 haplotype resolved diploid T2T assemblies
    + Representative of diversity?
    
. . .
    
- Explicitly align to the *complete set of diversity* [(in the reference)]{.fragment}
  
. . .

- <span style="color:red;"> Are the days of the linear reference dead? &#128561;</span>

. . .

- How will this impact the PROPHECY transcriptomics layer?
    + e.g. How do we integrate annotated regulatory elements from a linear reference

[^3]: Liao W et al, *A draft human pangenome reference* Nature, 2023

::: {.notes}
- This is an area of active discussion amongst the Black Ochre Data Labs
- How will these approaches impact integration across layers?
- How will it impact our ability to access existing resources, all based on linear genomes
- Monica's work, using a public, non-Indigenous T2D dataset will help us understand how known genes behave under this new approach
:::



# How Do We Make Key Discoveries? {background-color="#3c3c44"}

## Differential Gene Expression

Once we have counts (gene or transcript-level)

- Identify any technical issues (GC bias, failed samples etc.)
- Fit standard statistical models (GLM)
    + Fairly simple in small Treatment vs Control comparisons
    + Less straightforward with large, complex designs
    
. . .

- Sequencing generally done in batches of $\leq$ 96 $\implies$ batch effects
    + Is identifiable technical noise which masks true biology
    + Lead to <span style="color:red;">&#8679;</span> false discoveries <span style="color:red">&#8681;</span> true discoveries
    + Active area of development for large cohort studies (Terry Speed)
    

<!-- - Batch effects, GC bias etc -->

::: {.notes}
- Most analysis uses Negative Binomial models (like Poisson with more flexibility in variance modelling)
- For simple comparisons we often assume effect size (change in expression) is only relevant if large enough
    + This is not as easy in complex models
    + These ar also observational studies $\implies$ we get what we get
    + We will find correlations between expression patterns and predictor variables $\implies$ doesn't elucidate mechanism
- 96-well plates are used in sample preparation
    + Lots of downstream protocols developed based on this, just like Roman chariot wheels determined train-track width
    + Key strategies for managing batch effects are
        + using consistent reagents from the same batch
        + including technical replicates across batches 
        + control RNA-sequences with known concentrations
:::

## Differential Gene Expression

::: {.columns}

::: {.column}
![*Typical MA plot* (Pederson, unpublished)](https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/master/docs/figure/dge_analysis.Rmd/plotMA-1.png?raw=true)
:::

::: {.column}
![*Typical Volcano plot* (Pederson, unpublished)](https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/master/docs/figure/dge_analysis.Rmd/plotVolcano-1.png?raw=true){width="80%"}
:::

:::

::: {.notes}
- In the context of more complex models, less useful plots
:::


## Network approaches

- DGE takes "significantly DE" genes and joins to try & form a story
    + i.e. big changes in a few genes $\implies$ biological consequences
    
. . .

- Network approaches look for larger shifts amongst correlated genes
    + i.e. small changes across an entire pathway $\implies$ biological consequences
- Far more flexibility with parameters $\implies$ *reproducibility*?
    + Recent research is improving this markedly

## WGCNA

::: {.columns}

::: {.column}

- Multiple approaches with WGCNA the biggest player
    + Form correlation network
    + Identify modules within correlation network
    + Compare to predictor variables
    + Identify underlying biology
:::

::: {.column}
![Image from Huang, J et al, *Analysis of functional hub genes identifies CDC45 as an oncogene in non-small cell lung cancer - a short report* Cellular Oncology, 2019](assets/13402_2019_438_Fig1_HTML.png){width="70%"}
:::

:::

::: {.notes}
- WGCNA is Weighted Gene Correlation Network Analysis
- The top panel shows a dendrogram based on correlations with modules at the bottom
- The bottom panel compares modules to predictor variable
:::

## Supervised Approaches

- Principal Component Analysis (PCA) is an un-supervised approach
    + Components maximises variance
    + Mainly used for QC & visualisation in transcriptomics
    
. . .

- Projection onto Latent Space (PLS) is a *supervised* approach
    + Components maximise covariance with predictor variables
    + Alternative approach to identifying groups of correlated genes

. . .

- No $p$-values &#128561; $\implies$ How can we ever publish?

::: {.notes}
- PLS is also known as Partial Least Squares
    + Supported & enabled by mix-omics (Kim-Anh Le Cao)
    + Excellent support for integration of multi-omics data
:::

## eQTL and TWAS

- Expression Quantitative Trait Loci (eQTL)
    + If RNA abundance is a quantitative trait $\implies$ which variants are associated with this?
    + Is there an association with phenotype (e.g. CVD development)
    
. . .

- Transcriptome-Wide Association Studies (TWAS)
    + Integrates analysis eQTL with GWAS

## scRNA-Seq 

::: {.columns}

::: {.column}

- Retains the connection between transcript and cell-of-origin
- Huge numbers of 'failure to detect' expression (i.e. zero counts)
- Uses clustering to identify cell types within a sample
- Pseudo-bulk clusters for DGE analysis

:::

::: {.column}
![Image from: Amezquita, R et al *OSCA: Orchestrating Single Cell analysis with Bioconductor* Bioconductor, 2022](https://bioconductor.org/books/3.15/OSCA.basic/reduced-dimensions_files/figure-html/tsne-brain-1.png)
:::

:::

## Spatial Transcriptomics

- Cells are held in place $\implies$ transcripts identified within a region
- The current hot area in transcriptomics  $\implies$ Nature Methods "Method of the Year, 2020"
- Single-cell resolution is arguably here

![](assets/716c1856c750044a946968defe719a56.jpg)   

    
::: {.notes}
- Neither technique applicable to PROPHECY Datasets $\implies$ don't really address initial questions
:::

## Pathway & Functional Analysis

::: {.columns}

::: {.column width="45%"}

- Look for biologically relevant signals in DE genes or network modules
    + Enriched pathways
    + Common transcription factors
    + Drug target signals
- Compare to public datasets <br>(if appropriate)

::: {.fragment}

- Interpretation is key
    + Researchers love to invent stories

:::

:::

::: {.column width="55%"}
    
![*Enriched pathways combining RNA-Seq with AR and H3K27ac ChIP-seq* (Pederson, Unpublished)](https://smped.github.io/GRAVI_PRJNA509779/assets/AR_E2_E2DHT-H3K27ac_E2_E2DHT/AR_Up-H3K27ac_Up_rnaseq_network.png)

:::

:::

::: {.notes}
- Often just fancy versions of Fisher's Exact Test
- Correlations to known treatments or perturbations can be informative
- Lots of redundancy in pathways & gene set collections
    + Often use networks to identify "communities" of highly similar gene-sets
    
- Famous story of a transcriptomic dataset, where the values and gene labels were out by one
    + (Keith Baggerley & Duke University)
    + Even though the biology was nonsense, researchers still saw patterns & explanations
:::

## Acknowledgements {.smaller}

<br>

::: {.columns}

::: {.column width="30%"}
#### TKI / BODL Nerds

- Alex Brown
- Jimmy Breen
- Sam Buckberry
- Liza Kretzschmar
- Yassine Souilmi
- Bastien Llamas
- Holly Martin
- Claudia Floreani
- Natasha Howard
- Anelle du Preez
- Kaashifah Bruce
- Amanda Satour-Richards
- Justine Clarke
- Katharine Brown
:::

::: {.column width="30%"}
#### NCIG

- Hardip Patel

#### ALIGN
- Johanna Barclay 
- Annalee Stearne
- Louise Lyons

#### Students
- Nhi Hin
- Jacqueline Rehn
- Nora Liu
- Lachlan Baer
- Megan Monaghan
- Monica Guilhaus

:::

::: {.column width="30%"}

#### Bioinformatics Hub
- David Adelson
- Gary Glonek
- Dan Kortschak
- Nathan Watson-Haigh
- Rick Tearle
- Hien To
:::

:::

# Additional Material  {background-color="#3c3c44"}

## The Central Dogma

![From Frances Crick, *Ideas on Protein Synthesis*, Unpublished Note, Wellcome Library, 1956](assets/crick_dogma_note.png){fig-align="left"}

::: {.notes}
- Interestingly, Crick's perspective on the the Central Dogma wasn't that RNA must be translated into protein, but was more about the direction of information loss
:::


## *SEC14L2*: A Random Example

![](assets/sec14l2.png){width="1100px"}

::: {.notes}
I was looking at this one the other day:

- The canonical gene for *SEC14L2* is shown at the top, as is the canonical gene for *MTFP1*
- *SEC14L2* has 4 start sites & multiple combinations identified
- *MTFP1* has one start site, with a few combinations
- New discoveries show a start site halfway down *SEC14L2* but with some of *MTFP1* at the end
- What do we call this?
- Also, if you start on the other DNA strand and go in the other direction, you get *RNF215*, which also has another start site
- I sometimes think that if we knew what we know know, our entire way of defining genes etc. might be completely different
:::


## Long Reads

::: {.columns}

::: {.column width="60%"}

- Short read technology has <span style="color:#769b79;">&#8681;&#8681;&#8681;</span> error rates
- Long reads have <span style="color:red;">&#8679;&#8679;&#8679;</span> error rates

::: {.fragment fragment-index=1}

- Reads are circularised <br>$\implies$ errors corrected by repeat reads

:::

::: {.fragment fragment-index=2}


- Great for identification of novel transcripts <br>$\implies$ difficult to quantify
    + Creation of a custom reference transcriptome
    + Challenging to refer back to functional annotations in reference
    
:::
    
:::

::: {.column width="40%"}

::: {.fragment fragment-index=1}

![Peccoud J et al, *Untangling Heteroplasmy, Structure, and Evolution of an Atypical Mitochondrial Genome by PacBio Sequencing* Genetics 2017](assets/m_269fig2.jpeg)
:::

:::

:::
