---
title: "ALIGN Capacity Building"
subtitle: "The Why, What and How of Transcriptomics"
author: "Dr Stevie Pederson (They/Them)"
affiliation: "Black Ochre Data Labs"
date: "2023-08-02"
date-format: long
title-slide-attributes:
    data-background-color: "#3c3c44"
    data-background-opacity: "20%"
    data-background-size: "35%"
    data-background-position: "50% 50%"
editor: source
format: 
  revealjs:
    bibliography: references.bib
    theme: [custom.scss]
    code-line-numbers: false
    width: 1280
    height: 720
    sansfont: Times New Roman
    logo: assets/bodl_logo_white_background.jpg
    slide-number: c/t
knitr: 
  opts_chunk: 
    echo: true
    include: true
    warning: false
    message: false
    fig.align: left
---

##

::: {.columns}

::: {.column width="35%"}

### Acknowledgement of Country

:::

::: {.column width="65%"}

I would like to acknowledge that today I'm presenting this topic from Kaurna Country.<br><br>

I acknowledge the deep feelings of attachment and relationship of the Kaurna people to their Place.<br><br>

I also pay respects to the cultural authority of Aboriginal and Torres Strait Islander peoples from other areas of Australia online today, and pay my respects to Elders past, present and emerging. 


:::

:::

::: {.notes}
I also wish to express my gratitude for being involved in this space, and an awareness of the trust that has been placed in me
:::

## Who Am I? 

- Oldest person in Black Ochre Data Labs but still an ECR

. . .

- Unintentionally took the "winding path" approach to my PhD (2008-2018)
    + Undertaken in a T Cell  / Autoimmunity Research Group
    + Developed a novel Bayesian Statistical model for alternate transcript detection (using microarrays)

. . . 

- Co-ordinator of the University of Adelaide Bioinformatics Hub (2014-2020)
    + Analysed dozens of RNA-Seq datasets (+ more)
    + Developed multiple training workshops and undergrad/postgrad courses

. . .     
    
- Massive R nerd (5 Bioconductor packages)
    

::: {.notes}
- PhD was working in a T cell group looking at autoimmune disease
- Developed a Bayesian model for detection of alternate transcript expression
- Am the lead developer for two Bioconductor packages and have contributed to 3 more
- Currently sit on the Community Advisory Boards for Bioconductor
- Also working on my 6th package
:::

## Who Am I? 

- First discovered microarrays and R in 2002
    + Most statistical understanding has been learned through this lens
    + Primary training is in biology (e.g. genetics, biochemistry)

. . .

- Majority of experience in bulk tissue (i.e. not single-cell)

. . .

- Also spent two years researching ER-dependent Breast Cancers
    + Looking at Transcription Factors and DNA state (ChIP-Seq)
    + Data Integration (RNA-Seq, ChIP-Seq, HiC etc)

::: {.notes}
- Statistical training has been very different to epidemiology & study design
:::

# Why Transcriptomics? {background-color="#3c3c44"}

## Why?

- Transcriptomics is essentially the analysis of RNA
- Provides insight into a **highly dynamic** component of biology
- Long history of identifying biomarkers and establishing mechanisms
    - Is essentially a hypothesis-generating step
    - Often need to test hypotheses using conventional experimental approaches
    
## Why?

PROPHECY $\implies$ Preventing Renal OPHtalmic Events in CommunitY

. . .

- Key focus is Type 2 Diabetes and complications (CVD and CKD)

. . .

- Multiple 'omics layers: DNA (Genomics), DNA-methylation (EpiGenomics), protein (Proteomics) etc

. . .

- Whole blood $\implies$ large immune component ("white blood cells")
- Can we see markers of the different stages of Type 2 Diabetes <br>(i.e. complications developing)
    
. . . 

- By integrating with other -omics layers:
    + Can we establish clear hypotheses into underlying mechanisms? <br>[$\implies$ clinical intervention]{.fragment .fade-in}
    
::: {.notes}
- Also have lipidomics, metabolomics, retinal scans, ECGs, imflammatory markers
:::
    
# How Does Biology Even Work? {background-color="#3c3c44"}

::: {.notes}
- If only any of us could answer this question
- As soon as we think we understand something, we find the exception
- So much redundancy built-in + so much complexity
:::

## DNA

::: {.columns}

::: {.column width='56%'}

![Image created using www.biorender.com](assets/cell_structure.png){width="1000px"}

:::

::: {.column width='44%'}

- Most are familiar with DNA
- Directly inherited from parents<br>$\implies$ unique people
- Stays in the nucleus
- The same in every cell [(roughly)]{.fragment .fade-in}

::: {.fragment}

- Generally static [(until cell division)]{.fragment .fade-in}

:::

:::


:::

::: {.notes}
- Most of us a familiar with DNA
- We all have our own unique DNA, passed down from our parents
- DNA lives in the nucleus of most cells
- Chromosomes replicate and divide with cell division
:::

## DNA

::: {.columns}

::: {.column width="55%"}
    
![Made using www.biorender.com](assets/dna_breakdown.png){fig-align="left"}

:::

::: {.column width="45%"}

Each chromosome is a string of millions of nucleotides

- 4 nucleotides (or bases): A, C, G, T
- Double-stranded molecule, i.e. double helix
    + Adds incredible stability 
    + Provides error handling mechanism
- Amazing amount of 3D structure & packaging

::: {.fragment}
- Only forms the classic ![](assets/sister_chromatids.jpg){width="4%"} shape during cell division
:::

:::

:::

::: {.notes}
- Full names for bases Adenine, Cytosine, Guanine, Thymine
- All chromosomes in a human cell would be about 2m long if stretched out
- The sequence is static, but a lot happens to the DNA and to package the DNA
- Essentially the same static molecule having things done to it
:::

## DNA

We often think of there being [**one unique reference sequence**]{.underline}

- This is very far from reality 
- Each copy of Chromosome 1 is different to the other copy within each of us
    + Both copies are (essentially) the same between all of our cells
    
. . .    
    
- Same for all other chromosomes (1-22 + X)
- Scale this out to millions of people...
- Some regions are relatively invariant (needed for survival)
    
## DNA
    
- Unique combinations of bases are ours and ours alone, for each of us
    + Inherited from parents (+ a few spontaneous changes)
    
. . .    
    
- We often refer to these unique sequences as *variants*
    + Single nucleotides (SNVs)
    + Missing or extra nucleotides (InDels) 
    + Larger structural variants (SVs) and repetitive elements (REs)
    
. . .    
    
- Collections of variants within a population $\implies$ *genetic diversity*

::: {.notes}
- There are large runs of DNA which rarely change between individuals
- I have a mutation in my haemoglobin not shared by either parent
    + My HbA1c cannot be measured, whereas both of theirs can
    + Was reported to me as a serious discovery during a public health study (NWHS)
    + My blood responds poorly to low-oxygen situations: like respiratory infections
    + Also no plans to climb Mt Everest
- Having "variants" implicitly ties back to this idea of a single reference
:::



## The Central Dogma of Biology

::: {.columns}

::: {.column width="40%"}
<br>
<iframe src="https://giphy.com/embed/mRI1hW0ZBVUly" width="450px" height="450px" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/biology-primer-mRI1hW0ZBVUly">via GIPHY</a></p>

:::

::: {.column width="60%"}

DNA is essentially a giant book of instructions


::: {.fragment}

- Some regions are known as genes
    + These are *transcribed* into RNA
    + Many RNA types (mRNA + miRNA, ncRNA, rRNA etc)
    
:::

::: {.fragment}
    
- mRNA molecules are *translated into proteins*

:::

::: {.fragment}

- Proteins *do stuff* i.e. pretty much everything
   + Keratin $\implies$ Hair structure
   + Haemoglobin $\implies$ Oxygen transport
   + Antibodies $\implies$ Detect alien molecules
   
:::

:::

:::

::: {.notes}
- DNA can almost be considered as a giant set of personalised instructions
- Most regions have some kind of function $\implies$ some help determine structure
    + Others are transcribed
- Transcribed regions (i.e. RNA) is known as the **Transcriptome**
- Also enzymes which breakdown food are proteins
:::

## The Central Dogma

![From Frances Crick, *Ideas on Protein Synthesis*, Unpublished Note, Wellcome Library, 1956](assets/crick_dogma_note.png)

::: {.notes}
- Interestingly, Crick's perspective on the the Central Dogma wasn't that RNA must be translated into protein, but was more about the direction of information loss
:::

## Proteins

Proteins are the molecular machines that run our bodies

. . .

- Incredibly dynamic system
    + Some are stable $\implies$ Some degrade rapidly
    
. . .

- Each cell-type (e.g. neuron, skin cell) will make a different set of proteins
    + Some structural proteins may be common between unrelated cell-types
    + Many proteins exist *outside* of cells

. . .

- Proteins can be heavily modified
    + Phosphorylation, Sumoylation, Ubiquitination etc
    + Can be single molecules or bind to multiple partners
- Different functions or activities for every state


::: {.notes}
- Another key feature of proteins which sets them apart from DNA & RNA is their role outside cells
- By dogma, Nucleic Acids don't
:::

## Proteins

*Proteomics* is the study of proteins

. . . 

- Is very limited technique
    + Tens of Thousands exist 
    + Can usually only identify hundreds (maybe 1-2000) from a sample
    
. . .    
    
- We don't understand much of the complexity
    + Single modifications can have huge consequences

. . .

- Still a very immature field
    + Technology is extremely variable and rapidly developing

::: {.notes}
- Proteomics is still really immature but can still shed important light on underlying biology
- Most proteomic data is produced using Mass Spectrometry
    + Fragments the proteins and we detect spectra mass on mass & charge spectra
- Some tissues are dominated by a small set of proteins (e.g. albumin in blood is >70% of all protein)
- Being difficult study makes it harder for us to study $\implies$ we often look elsewhere that's easier
- This is where transcriptomics comes in
:::

# What is Transcripomics? {background-color="#3c3c44"}

## Transcriptomics

*Transcriptomics is the study of RNA molecules within a cell or tissue*

. . .

- Not all genes are transcribed (i.e. expressed) in every cell
    + ~60,000 annotated genes in the genome
    + ~10-15,000 detectable genes in a given cell-type
    
. . .

- Transcriptomes are *cell-type specific*
    + Related cell-types will often have great similarity, e.g. T-cell sub-types
    
. . .

- Often take a cell-line (or cell-type) and expose to a treatment
    + Low variability $\implies$ treatment response is easy(ish) to spot

## Transcriptomics

- Transcriptomics is often an abundance-focussed analysis
    + Represents a *stable state* of a cell
    + Differential Gene Expression [$\implies$ *detectable response to stimuli*]{.fragment .fade-in}

. . .    
  
- Traditionally focussed on mRNA
- Common early assumption $\implies$ mRNA abundance reflects protein levels
    + Is sometimes true ...
    
    
::: {.notes}
- RNA molecules which are translated into proteins are generally referred to as messenger RNA (mRNA)
- Despite many mRNAs representing an intermediary state, this reveals a lot of information about the cell, e.g. what it's responding to, how it's maintaining itself
- Some genes are transcribed without protein product
    + Enables rapid response if protein is needed quickly
- Great examples of rapid transcriptional response are in T cells which encounter their target antigen or see pro-inflammatory cytokines
    + Huge & rapid transcriptional consequences
:::
    
## Transcriptomics
    
Our classic viewpoint is DNA $\implies$ mRNA $\implies$ Protein

::: {.columns}
   
::: {.column width="55%"}

::: {.incremental}
    
- Many other types of RNA play key roles
    + miRNAs bind to and degrade target mRNAs
    + lncRNAs form highly complex structures <br>$\implies$can silence chromosomal regions (e.g. *XIST*)
    + rRNAs form key units of ribosomes <br>$\implies$ protein translation

:::

:::

::: {.column width="45%"}

::: {.fragment}

![A 2kb region of Xist. Taken from Fang et al., “Probing Xist RNA Structure in Cells Using Targeted Structure-Seq”](https://github.com/UofABioinformaticsHub/transcriptomics_applications/blob/master/lectures/figures/Xist.png?raw=true)

:::

:::

:::

::: {.notes}
- Many of the functional RNAs also provide vital insights, but these are often poorly understood
- We generally ignore Ribosomal RNAs, and given they make up ~80% of RNA in a cell, we need to manage these
:::

## What is an RNA transcript?

::: {.columns}

::: {.column width="40%"}


- Known transcribed regions are defined as genes
    + Transcribed from DNA $\implies$ start to finish

::: {.fragment}

- From the complete RNA transcript:
    + Introns are *spliced* out
    + Exons form *mature transcript*
    
:::
    
:::

::: {.column width="60%"}

![Image courtesy of National Human Genome Research Institute](assets/splicing2.png)

:::

:::

::: {.fragment}

- ~38% of genes (~24,000) are transcribed into multiple **transcripts**
- Different transcripts $\implies$ Different proteins (or lncRNA)

:::

::: {.notes}
- This definition of a gene is quite different to the original *'unit of inheritance'*
- It's probably worth noting that from a DNA-structural perspective, introns & exons are just long sequences
    + Splice junctions are defined by core sequence motifs, not any 3D structure like appears here 
- This is really where the name transcriptomics has developed
- Represents far greater complexity than just gene-level abundances
- Some genes even have multiple start sites
- The key point is we have one gene, multiple transcripts, multiple proteins then multiple modifications of multiple proteins
- Transcriptomics reflects a key inflection point in the development of informational complexity
:::

## *SEC14L2*: A Random Example

![](assets/sec14l2.png){width="1100px"}

::: {.notes}
I was looking at this one the other day:

- The canonical gene for *SEC14L2* is shown at the top, as is the canonical gene for *MTFP1*
- *SEC14L2* has 4 start sites & multiple combinations identified
- *MTFP1* has one start site, with a few combinations
- New discoveries show a start site halfway down *SEC14L2* but with some of *MTFP1* at the end
- What do we call this?
- Also, if you start on the other DNA strand and go in the other direction, you get *RNF215*, which also has another start site
- I sometimes think that if we knew what we know know, our entire way of defining genes etc might be completely different
:::

# How Do We Study Transcriptomics {background-color="#3c3c44"}

## Early Gene Expression Approaches

::: {.columns}

::: {.column}

- The first high-throughput platform was Microarrays
    + 3' Arrays (Affymetrix)
- Gene-centric approach
    + Could analyse 10-15,000 genes
- Many statistical tools developed in this context
    + `limma` has been maintained by Gordon Smyth (WEHI) for >20years
  
    
:::

::: {.column}

![Image courtesy of Affymetrix](https://s3.amazonaws.com/static.wd7.us/2/22/Affymetrix-microarray.jpg)

:::

:::

::: {.notes}
- The 20yrs younger nerd got excited by this technology
- Limma stands for linear models for microarrays
- Still heavily used in modern approaches
:::

## Microarrays

::: {.columns}

::: {.column}

![Image courtesy of Affymetrix](assets/affy_design.png)

Known sequences at known locations

:::

::: {.column}

::: {.fragment}

![Image courtesy of Affymetrix](https://s3.amazonaws.com/iphonecde/_GNCHIP3.GIF)

Labelled cDNA binds complementary target

:::

:::

:::

::: {.notes}

- Known sequences were placed at specific co-ordinates
- RNA was converted to cDNA (more stable) labelled with fluorophores
- cDNA binds to perfectly matching complementary sequence
- Laser excites fluorophores: brighter signal $\implies$ more original RNA
- Were indirectly measuring RNA abundance
- RNA degrades from one-specific end (i.e. 3') so probes targets 3' end of a gene for intact RNA
:::

## Bulk RNA Sequencing

- Modern approaches involve sequencing the transcriptome
    + Short Reads (Illumina) < 300bases: Quantitative
    + Long Reads (Nanopore, PacBio); Semi-Quantitative

. . .

- Short reads still dominate
    + Used in the PROPHECY transcriptomics layer

. . .
    
- Mature RNA transcripts may be short (8nt) or long (350,375nt)

::: {.notes}
- Transcribed RNA (pre-processing) can be up to 350,000 nucleotides
- Longest transcript is *XACT* (lncRNA)
- Shortest codes for a 2aa component of the T cell receptor
:::


## Bulk RNA Sequencing

- All cells from each sample (or tissue) are lysed and RNA extracted
- RNA is fragmented (250-500bp)
    + RNA Quality is assessed (RIN Score)
- RNA fragments are prepared for sequencing (library preparation)
    + Converted to cDNA
    + Add sequencing adapters
    + PCR amplification
- Sequencing

. . .

How do we put this all back together and quantify?

::: {.notes}
- Imagine chopping up the screen-plays to all 9 Star Wars movies then having to figure out where it came from?
- "Use the Force" might be really hard to figure out, but "Use the Force Luke" might be easier
:::

## Bulk RNA Sequencing

![](assets/bd020139.f1.jpeg)

## Two Approaches To Alignment

::: {.columns}

::: {.column width="40%"}
![](assets/alignment.jpeg)
:::

::: {.column width="60%"}
1. Alignment to a reference genome
    - Most align to one location $\implies$ can see where <span style="color:#769b79;">&#10003;</span>
    - Brilliant for gene-level counts <span style="color:#769b79;">&#10003;</span>
    - Not helpful for transcript-level <span style="color:red;">&#10007;</span>

::: {.fragment}

2. Alignment to a reference transcriptome
    - Obtain transcript & gene-level counts <span style="color:#769b79;">&#10003;</span>
    - Many exons shared with multiple transcripts
    - Many reads align to multiple transcripts $\implies$ significant uncertainty
    - Can manage this uncertainty in modelling <span style="color:#769b79;">&#10003;</span>
    - Don't know where each read has aligned <span style="color:red;">&#10007;</span>
:::

:::

:::

::: {.notes}
- Aligning to a linear reference uses conventional aligners (Burrows-Wheeler transform)
- Aligning to a transcriptome incorporates graph-based strategies
:::

## Limitations



## Alternative Approaches

- Reference graphs

## Long Reads

- Great for identification of novel transcripts
- Uses circularisation


# How Do We Make Key Discoveries? {background-color="#3c3c44"}

## Differential Gene Expression

- Batch effects, GC bias etc

## Pathway & Functional Analysis

## WGCNA

## Mix-Omics



