<!DOCTYPE html>
<html lang="en"><head>
<script src="20231107_ALIGN_Transcriptomics_files/libs/clipboard/clipboard.min.js"></script>
<script src="20231107_ALIGN_Transcriptomics_files/libs/quarto-html/tabby.min.js"></script>
<script src="20231107_ALIGN_Transcriptomics_files/libs/quarto-html/popper.min.js"></script>
<script src="20231107_ALIGN_Transcriptomics_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="20231107_ALIGN_Transcriptomics_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="20231107_ALIGN_Transcriptomics_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="20231107_ALIGN_Transcriptomics_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="20231107_ALIGN_Transcriptomics_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.353">

  <meta name="author" content="Dr Stevie Pederson (They/Them)">
  <meta name="dcterms.date" content="2023-11-07">
  <title>ALIGN Capacity Building</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="20231107_ALIGN_Transcriptomics_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="20231107_ALIGN_Transcriptomics_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
  </style>
  <link rel="stylesheet" href="20231107_ALIGN_Transcriptomics_files/libs/revealjs/dist/theme/quarto.css">
  <link href="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" data-background-color="#3c3c44" data-background-opacity="20%" data-background-position="50% 50%" data-background-size="35%" class="quarto-title-block center">
  <h1 class="title">ALIGN Capacity Building</h1>
  <p class="subtitle">The Why, What and How of Transcriptomics</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Dr Stevie Pederson (They/Them) 
</div>
</div>
</div>

  <p class="date">November 7, 2023</p>
</section>
<section id="section" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:35%;">
<h3 id="acknowledgement-of-country">Acknowledgement of Country</h3>
</div><div class="column" style="width:65%;">
<p>I would like to acknowledge that I’m presenting today from Kaurna Country.<br><br></p>
<p>I acknowledge the deep feelings of attachment and relationship of the Kaurna people to their Place.<br><br></p>
<p>I also pay respects to the cultural authority of Aboriginal and Torres Strait Islander peoples from other areas of Australia online today, and pay my respects to Elders past, present and emerging.</p>
</div>
</div>
<aside class="notes">
<p>I also wish to express my gratitude for being involved in this space, and an awareness of the trust that has been placed in me</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="who-am-i" class="slide level2">
<h2>Who Am I?</h2>
<ul>
<li>Oldest person in Black Ochre Data Labs but still an ECR</li>
</ul>
<div class="fragment">
<ul>
<li>Unintentionally took the “winding path” approach to my PhD (2008-2018)
<ul>
<li>Undertaken in a T Cell / Autoimmunity Research Group</li>
<li>Developed a novel Bayesian Statistical model for alternate transcript detection (using microarrays)</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Coordinator of the University of Adelaide Bioinformatics Hub (2014-2020)
<ul>
<li>Analysed dozens of RNA-Seq datasets (+ more)</li>
<li>Developed multiple training workshops and undergrad/postgraduate courses</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Massive R nerd (5 Bioconductor packages)</li>
</ul>
<aside class="notes">
<ul>
<li>PhD was working in a T cell group looking at autoimmune disease</li>
<li>Developed a Bayesian model for detection of alternate transcript expression</li>
<li>Am the lead developer for two Bioconductor packages and have contributed to 3 more</li>
<li>Currently sit on the Community Advisory Boards for Bioconductor</li>
<li>Also working on my 6th package</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="who-am-i-1" class="slide level2">
<h2>Who Am I?</h2>
<ul>
<li>First discovered microarrays and R in 2002
<ul>
<li>Most statistical understanding has been learned through this lens</li>
<li>Primary training is in biology (e.g.&nbsp;genetics, biochemistry)</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Majority of experience in bulk tissue (i.e.&nbsp;not single-cell)</li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Also spent two years researching ER-dependent Breast Cancers
<ul>
<li>Looking at Transcription Factors and DNA state (ChIP-Seq)</li>
<li>Data Integration (RNA-Seq, ChIP-Seq, HiC etc.)</li>
</ul></li>
</ul>
<aside class="notes">
<ul>
<li>Statistical training has been very different to epidemiology &amp; study design</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section>
<section id="why-transcriptomics" class="title-slide slide level1 center" data-background-color="#3c3c44">
<h1>Why Transcriptomics?</h1>

</section>
<section id="why" class="slide level2">
<h2>Why?</h2>
<ul>
<li>Transcriptomics is essentially the analysis of RNA</li>
<li>Provides insight into a <strong>highly dynamic</strong> component of biology</li>
<li>Long history of identifying biomarkers and establishing mechanisms
<ul>
<li>Is essentially a hypothesis-generating step</li>
<li>Often need to test hypotheses using conventional experimental approaches</li>
</ul></li>
</ul>
</section>
<section id="why-1" class="slide level2">
<h2>Why?</h2>
<p>PROPHECY <span class="math inline">\(\implies\)</span> Preventing Renal OPHtalmic Events in CommunitY</p>
<div class="fragment">
<ul>
<li>Key focus is Type 2 Diabetes and complications (CVD and CKD)</li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Multiple ’omics layers: DNA (Genomics), DNA-methylation (Epigenomics), protein (Proteomics), metabolites (Metabolomics) etc</li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Whole blood <span class="math inline">\(\implies\)</span> large immune component (“white blood cells”)</li>
<li>Can we see markers of the different stages of Type 2 Diabetes <br>(i.e.&nbsp;complications developing)</li>
</ul>
</div>
<div class="fragment">
<ul>
<li>By integrating with other -omics layers:
<ul>
<li>Can we establish clear hypotheses into underlying mechanisms? <br><span class="fragment fade-in"><span class="math inline">\(\implies\)</span> clinical intervention</span></li>
</ul></li>
</ul>
<aside class="notes">
<ul>
<li>Also have lipidomics, metabolomics, retinal scans, ECGs, inflammatory markers</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section></section>
<section>
<section id="how-does-biology-even-work" class="title-slide slide level1 center" data-background-color="#3c3c44">
<h1>How Does Biology Even Work?</h1>
<aside class="notes">
<ul>
<li>If only any of us could answer this question</li>
<li>As soon as we think we understand something, we find the exception</li>
<li>So much redundancy built-in + so much complexity</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="dna" class="slide level2">
<h2>DNA</h2>
<div class="columns">
<div class="column" style="width:56%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="assets/cell_structure.png" width="1000"></p>
<figcaption>Image created using www.biorender.com</figcaption>
</figure>
</div>
</div><div class="column" style="width:44%;">
<ul>
<li>Most are familiar with DNA and cells</li>
<li>DNA stays in the nucleus</li>
<li>Directly inherited from parents<br><span class="math inline">\(\implies\)</span> unique people</li>
<li>The same in every cell <span class="fragment fade-in">(roughly)</span></li>
</ul>
<div class="fragment">
<ul>
<li>Generally static <span class="fragment fade-in">(until cell division)</span></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Lots of interaction with other molecules: DNA-methylation, Histones etc.</li>
</ul>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>Most of us a familiar with DNA and the idea that our bodies are made of trillions of cells (28-36 trillion)</li>
<li>We all have our own unique DNA, passed down from our parents</li>
<li>DNA lives in the nucleus of most cells (not RBC)</li>
<li>Chromosomes replicate and divide with cell division</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="dna-1" class="slide level2">
<h2>DNA</h2>
<div class="columns">
<div class="column" style="width:55%;">
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="assets/dna_breakdown.png"></p>
<figcaption>Made using www.biorender.com</figcaption>
</figure>
</div>
</div><div class="column" style="width:45%;">
<p>Each chromosome is a string of millions of nucleotides</p>
<ul>
<li>4 nucleotides (or bases): A, C, G, T</li>
<li>Double-stranded molecule, i.e.&nbsp;double helix
<ul>
<li>Adds incredible stability</li>
<li>Provides error handling mechanism</li>
</ul></li>
<li>Amazing amount of 3D structure &amp; packaging</li>
</ul>
<div class="fragment">
<ul>
<li>Only forms the classic <img data-src="assets/sister_chromatids.jpg" style="width:4.0%"> shape during cell division</li>
</ul>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>Full names for bases Adenine, Cytosine, Guanine, Thymine</li>
<li>All chromosomes in a human cell would be about 2m long if stretched out</li>
<li>The sequence is static, but a lot happens to the DNA and to package the DNA</li>
<li>Essentially the same static molecule having things done to it</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="dna-2" class="slide level2">
<h2>DNA</h2>
<p>We often think of there being <u><strong>one unique reference sequence</strong></u></p>
<ul>
<li>This is very far from reality</li>
<li>Each copy of Chromosome 1 is different to the other copy within each of us
<ul>
<li>Both copies are (essentially) the same between all of our cells</li>
<li>No value judgements on similarity to reference <span class="math inline">\(\implies\)</span> just who we are</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Same for all other chromosomes (1-22 + X)</li>
<li>Scale this out to billions of people…</li>
<li>Some regions are relatively invariant (needed for survival)</li>
</ul>
</div>
</section>
<section id="dna-3" class="slide level2">
<h2>DNA</h2>
<ul>
<li>Unique combinations of bases are ours and ours alone, for each of us
<ul>
<li>Inherited from parents (+ a few spontaneous changes)</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>We often refer to these unique sequences as <em>variants</em>
<ul>
<li>Single nucleotides (SNVs)</li>
<li>Missing or extra nucleotides (InDels)</li>
<li>Larger structural variants (SVs) and repetitive elements (REs)</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Collections of variants within a population <span class="math inline">\(\implies\)</span> <em>genetic diversity</em></li>
</ul>
<aside class="notes">
<ul>
<li>There are large runs of DNA which rarely change between individuals</li>
<li>I have a mutation in one of my haemoglobin not shared by either parent
<ul>
<li>My HbA1c cannot be measured, whereas both of theirs can</li>
<li>Was communicated to me as a reportable discovery during a public health study (NWHS, 20 years ago)</li>
<li>My blood responds poorly to low-oxygen situations: like respiratory infections</li>
<li>Also no plans to climb Mt Everest</li>
</ul></li>
<li>Having “variants” implicitly ties back to this idea of a single reference
<ul>
<li>Not mutations at all!!! Just diversity</li>
</ul></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="the-central-dogma-of-biology" class="slide level2">
<h2>The Central Dogma of Biology</h2>
<div class="columns">
<div class="column" style="width:40%;">
<br> <iframe src="https://giphy.com/embed/mRI1hW0ZBVUly" width="450px" height="450px" frameborder="0" class="giphy-embed" allowfullscreen=""></iframe>
<p>
<a href="https://giphy.com/gifs/biology-primer-mRI1hW0ZBVUly">via GIPHY</a>
</p>
</div><div class="column" style="width:60%;">
<p>DNA is essentially a giant book of instructions</p>
<div class="fragment">
<ul>
<li>Some regions are known as genes
<ul>
<li><em>Transcribed</em> into RNA</li>
<li>Many RNA types (mRNA, miRNA, ncRNA, rRNA etc.)</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>mRNA molecules are <em>translated into proteins</em>
<ul>
<li>Groups of three bases encode a single amino acid</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Proteins <em>do stuff</em> i.e.&nbsp;pretty much everything
<ul>
<li>Keratin <span class="math inline">\(\implies\)</span> Hair structure</li>
<li>Haemoglobin <span class="math inline">\(\implies\)</span> Oxygen transport</li>
<li>Antibodies <span class="math inline">\(\implies\)</span> Detect “alien” molecules</li>
</ul></li>
</ul>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>DNA can almost be considered as a giant set of personalised instructions</li>
<li>Most regions have some kind of function <span class="math inline">\(\implies\)</span> some help determine structure
<ul>
<li>Others are transcribed</li>
</ul></li>
<li>Transcribed regions (i.e.&nbsp;RNA) is known as the <strong>Transcriptome</strong></li>
<li>Groups of 3 bases are known as codons and these code for amino acids
<ul>
<li>Amino acids are key structural element of proteins</li>
<li>Each amino acid can have multiple codons (or codes)</li>
</ul></li>
<li>Also enzymes which breakdown food are proteins</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="the-central-dogma" class="slide level2">
<h2>The Central Dogma</h2>

<img data-src="assets/crick_dogma_note.png" class="r-stretch quarto-figure-left"><p class="caption">From Frances Crick, <em>Ideas on Protein Synthesis</em>, Unpublished Note, Wellcome Library, 1956</p><aside class="notes">
<ul>
<li>Interestingly, Crick’s perspective on the the Central Dogma wasn’t that RNA must be translated into protein, but was more about the direction of information loss</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="proteins" class="slide level2">
<h2>Proteins</h2>
<p>Proteins are the molecular machines that run our bodies</p>
<div class="fragment">
<ul>
<li>Incredibly dynamic system
<ul>
<li>Some are stable <span class="math inline">\(\implies\)</span> some degrade rapidly</li>
<li>Some exist only within cells <span class="math inline">\(\implies\)</span> some are exported</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Each cell-type (e.g.&nbsp;neuron, skin cell) will make a different set of proteins
<ul>
<li>Some structural proteins may be common between unrelated cell-types</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Proteins can be heavily modified
<ul>
<li>Phosphorylation, Sumoylation, Ubiquitination etc.</li>
<li>Can be single molecules or bind to multiple partners</li>
</ul></li>
<li>Different functions or activities for every state</li>
</ul>
<aside class="notes">
<ul>
<li>Proteins DO THINGS
<ul>
<li>Stomach enzymes like pepsin digest other proteins, others like lactase cut sugars</li>
<li>Cell surface receptors (ACE2)</li>
</ul></li>
<li>Another key feature of proteins which sets them apart from DNA &amp; RNA is their role outside cells</li>
<li>A common structural protein is Actin</li>
<li>By dogma, Nucleic Acids don’t exist outside cells (apart from when they do)
<ul>
<li>tRNA-mRNA molecules are extracellular signalling molecules in plants</li>
</ul></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="proteins-1" class="slide level2">
<h2>Proteins</h2>
<p><em>Proteomics</em> is the study of proteins</p>
<div class="fragment">
<ul>
<li>Is very limited technique
<ul>
<li>Tens of Thousands exist</li>
<li>Can usually only identify hundreds (maybe 1-2000) from a sample</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>We don’t understand much of the complexity
<ul>
<li>Single modifications can have huge consequences</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Compared to DNA/RNA sequencing is a very immature field
<ul>
<li>Technology is extremely variable and rapidly developing</li>
</ul></li>
</ul>
<aside class="notes">
<ul>
<li>Proteomics is still really immature but can still shed important light on underlying biology</li>
<li>Most proteomic data is produced using Mass Spectrometry
<ul>
<li>Fragments the proteins and we detect spectra mass on mass &amp; charge spectra</li>
</ul></li>
<li>Some tissues are dominated by a small set of proteins (e.g.&nbsp;albumin in blood is &gt;70% of all protein)</li>
<li>Being difficult study makes it harder for us to study <span class="math inline">\(\implies\)</span> we often look elsewhere that’s easier</li>
<li>This is where transcriptomics comes in</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section></section>
<section>
<section id="what-is-transcripomics" class="title-slide slide level1 center" data-background-color="#3c3c44">
<h1>What is Transcripomics?</h1>

</section>
<section id="transcriptomics" class="slide level2">
<h2>Transcriptomics</h2>
<p><em>Transcriptomics is the study of RNA molecules within a cell or tissue</em></p>
<div class="fragment">
<ul>
<li>Not all genes are transcribed (i.e.&nbsp;expressed) in every cell
<ul>
<li>~60,000 annotated genes in the genome</li>
<li>~10-15,000 detectable genes in a given cell-type</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Transcriptomes are <em>cell-type specific</em>
<ul>
<li>Related cell-types will often have great similarity, e.g.&nbsp;T-cell sub-types</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Often take a cell-line (or cell-type) and expose to a treatment
<ul>
<li>Low variability <span class="math inline">\(\implies\)</span> treatment response is easy(ish) to spot</li>
</ul></li>
</ul>
</div>
</section>
<section id="transcriptomics-1" class="slide level2">
<h2>Transcriptomics</h2>
<ul>
<li>Transcriptomics is often an abundance-focussed analysis
<ul>
<li>Represents a <em>stable state</em> of a cell</li>
<li>Differential Gene Expression <span class="fragment fade-in"><span class="math inline">\(\implies\)</span> <em>detectable response to stimuli</em></span></li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Traditionally focussed on mRNA</li>
<li>Common early assumption <span class="math inline">\(\implies\)</span> mRNA abundance reflects protein levels
<ul>
<li>Is sometimes true …</li>
</ul></li>
</ul>
<aside class="notes">
<ul>
<li>RNA molecules which are translated into proteins are generally referred to as messenger RNA (mRNA)</li>
<li>Despite many mRNA representing an intermediary state, this reveals a lot of information about the cell, e.g.&nbsp;what it’s responding to, how it’s maintaining itself</li>
<li>Some genes are transcribed without protein product
<ul>
<li>Enables rapid response if protein is needed quickly</li>
</ul></li>
<li>Great examples of rapid transcriptional response are in T cells which encounter their target antigen or see pro-inflammatory cytokines
<ul>
<li>Huge &amp; rapid transcriptional consequences</li>
</ul></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="transcriptomics-2" class="slide level2">
<h2>Transcriptomics</h2>
<p>Our classic viewpoint is DNA <span class="math inline">\(\implies\)</span> mRNA <span class="math inline">\(\implies\)</span> Protein</p>
<div class="columns">
<div class="column" style="width:55%;">
<div class="fragment" data-fragment-index="1">
<p>Many other types of RNA play key roles</p>
<ul>
<li>miRNA bind to and degrade target mRNA</li>
</ul>
</div>
<div class="fragment" data-fragment-index="2">
<ul>
<li>lncRNA form highly complex structures <br><span class="math inline">\(\implies\)</span>can silence chromosomal regions (e.g.&nbsp;<em>XIST</em>)<sup>1</sup></li>
</ul>
</div>
<div class="fragment" data-fragment-index="3">
<ul>
<li>rRNA are key components of ribosomes</li>
</ul>
</div>
<div class="fragment" data-fragment-index="4">
<ul>
<li>tRNA interact with ribosomes <br><span class="math inline">\(\implies\)</span> translate mRNA into proteins<sup>2</sup></li>
</ul>
</div>
</div><div class="column" style="width:45%;">
<div class="r-stack">
<p><img data-src="https://github.com/UofABioinformaticsHub/transcriptomics_applications/blob/master/lectures/figures/Xist.png?raw=true" class="fragment fade-in-then-out" style="width:85.0%" data-fragment-index="2"></p>
<p><img data-src="https://cdn.pfizer.com/pfizercom/20191227_trna.gif" class="fragment" style="width:85.0%" data-fragment-index="5"></p>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>Many of the functional RNA also provide vital insights, but these are often poorly understood</li>
<li>We generally ignore Ribosomal RNA, and given they make up ~80% of RNA in a cell, we need to manage these</li>
<li>tRNA are generally not interesting either (expect when forming dicistronic tRNA-mRNA)</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<aside><ol class="aside-footnotes"><li id="fn1"><p>Taken from Fang et al., “Probing Xist RNA Structure in Cells Using Targeted Structure-Seq”</p></li><li id="fn2"><p>Animation courtesy of www.pfizer.com</p></li></ol></aside></section>
<section id="what-is-an-rna-transcript" class="slide level2">
<h2>What is an RNA transcript?</h2>
<div class="columns">
<div class="column" style="width:40%;">
<ul>
<li>Known transcribed regions are defined as genes
<ul>
<li>Transcribed from DNA <span class="math inline">\(\implies\)</span> start to finish</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>From the complete RNA transcript:
<ul>
<li>Introns are <em>spliced</em> out</li>
<li>Exons form <em>mature transcript</em></li>
</ul></li>
</ul>
</div>
</div><div class="column" style="width:60%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="assets/splicing2.png"></p>
<figcaption>Image courtesy of National Human Genome Research Institute</figcaption>
</figure>
</div>
</div>
</div>
<div class="fragment">
<ul>
<li>~38% of genes (~24,000) are transcribed into multiple <strong>transcripts</strong></li>
<li>Different transcripts <span class="math inline">\(\implies\)</span> Different proteins (or lncRNA etc.)</li>
</ul>
</div>
<aside class="notes">
<ul>
<li>This definition of a gene is quite different to the original <em>‘unit of inheritance’</em></li>
<li>It’s probably worth noting that from a DNA-structural perspective, introns &amp; exons are just long sequences
<ul>
<li>Splice junctions are defined by core sequence motifs, not any 3D structure like appears here</li>
</ul></li>
<li>This is really where the name transcriptomics has developed</li>
<li>Represents far greater complexity than just gene-level abundances</li>
<li>Some genes even have multiple start sites</li>
<li>The key point is we have one gene, multiple transcripts, multiple proteins then multiple modifications of multiple proteins</li>
<li>Transcriptomics reflects a key inflection point in the development of informational complexity</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="sec14l2-a-random-example" class="slide level2">
<h2><em>SEC14L2</em>: A Random Example</h2>

<img data-src="assets/sec14l2.png" width="1100" class="r-stretch"><aside class="notes">
<p>I was looking at this one the other day:</p>
<ul>
<li>The canonical gene for <em>SEC14L2</em> is shown at the top, as is the canonical gene for <em>MTFP1</em></li>
<li><em>SEC14L2</em> has 4 start sites &amp; multiple combinations identified</li>
<li><em>MTFP1</em> has one start site, with a few combinations</li>
<li>New discoveries show a start site halfway down <em>SEC14L2</em> but with some of <em>MTFP1</em> at the end</li>
<li>What do we call this?</li>
<li>Also, if you start on the other DNA strand and go in the other direction, you get <em>RNF215</em>, which also has another start site</li>
<li>I sometimes think that if we knew what we know know, our entire way of defining genes etc. might be completely different</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section></section>
<section>
<section id="how-do-we-study-transcriptomics" class="title-slide slide level1 center" data-background-color="#3c3c44">
<h1>How Do We Study Transcriptomics</h1>

</section>
<section id="early-gene-expression-approaches" class="slide level2">
<h2>Early Gene Expression Approaches</h2>
<div class="columns">
<div class="column">
<ul>
<li>The first high-throughput platform was Microarrays
<ul>
<li>3’ Arrays (Affymetrix)</li>
</ul></li>
<li>Gene-centric approach
<ul>
<li>Could analyse 10-15,000 genes</li>
</ul></li>
<li>Many statistical tools developed in this context
<ul>
<li><code>limma</code> has been maintained by Gordon Smyth (WEHI) for &gt;20years</li>
</ul></li>
</ul>
</div><div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="https://s3.amazonaws.com/static.wd7.us/2/22/Affymetrix-microarray.jpg"></p>
<figcaption>Image courtesy of Affymetrix</figcaption>
</figure>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>The 20yrs younger nerd got excited by this technology</li>
<li>Limma stands for linear models for microarrays</li>
<li>Still heavily used in modern approaches</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="microarrays" class="slide level2">
<h2>Microarrays</h2>
<div class="columns">
<div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="assets/affy_design.png"></p>
<figcaption>Image courtesy of Affymetrix</figcaption>
</figure>
</div>
<p>Known sequences at known locations</p>
</div><div class="column">
<div class="fragment">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="https://s3.amazonaws.com/iphonecde/_GNCHIP3.GIF"></p>
<figcaption>Image courtesy of Affymetrix</figcaption>
</figure>
</div>
<p>Labelled cDNA binds complementary target</p>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>Known sequences were placed at specific co-ordinates</li>
<li>RNA was converted to cDNA (more stable) labelled with fluorophores</li>
<li>cDNA binds to perfectly matching complementary sequence</li>
<li>Laser excites fluorophores: brighter signal <span class="math inline">\(\implies\)</span> more original RNA</li>
<li>Were indirectly measuring RNA abundance</li>
<li>RNA degrades from one-specific end (i.e.&nbsp;3’) so probes targets 3’ end of a gene for intact RNA</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="bulk-rna-sequencing" class="slide level2">
<h2>Bulk RNA Sequencing</h2>
<ul>
<li>Modern approaches involve sequencing the transcriptome
<ul>
<li>Short Reads (Illumina) &lt; 300bases: Quantitative</li>
<li>Long Reads (Nanopore, PacBio); Semi-Quantitative</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Short reads still dominate
<ul>
<li>Used in the PROPHECY transcriptomics layer</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Mature RNA transcripts may be short (8nt) or long (350,375nt)</li>
</ul>
<aside class="notes">
<ul>
<li>Bulk RNA-Seq really has become the more correct term now that we have single-cell RNA-seq
<ul>
<li>Just means we have RNA from multiple cells</li>
</ul></li>
<li>Transcribed RNA (pre-processing) can be up to 350,000 nucleotides</li>
<li>Longest transcript is <em>XACT</em> (lncRNA)</li>
<li>Shortest codes for a 2aa component of the T cell receptor</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="bulk-rna-sequencing-1" class="slide level2">
<h2>Bulk RNA Sequencing</h2>
<ul>
<li>All cells from each sample (or tissue) are lysed and RNA extracted</li>
<li>RNA is fragmented (250-500bp)
<ul>
<li>RNA Quality is assessed (RIN Score)</li>
</ul></li>
<li>RNA fragments are prepared for sequencing (library preparation)
<ul>
<li>Converted to cDNA</li>
<li>Add sequencing adapters</li>
<li>PCR amplification</li>
</ul></li>
<li>Sequencing</li>
</ul>
<div class="fragment">
<p>How do we put this all back together and quantify?</p>
<aside class="notes">
<ul>
<li>Imagine chopping up the screen-plays to all 9 Star Wars movies then having to figure out where it came from?</li>
<li>“Use the Force” might be really hard to figure out, but “Use the Force Luke” might be easier</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="bulk-rna-sequencing-2" class="slide level2">
<h2>Bulk RNA Sequencing</h2>

<img data-src="assets/bd020139.f1.jpeg" class="r-stretch"></section>
<section id="two-approaches-to-alignment" class="slide level2">
<h2>Two Approaches To Alignment</h2>
<div class="columns">
<div class="column" style="width:40%;">
<p><img data-src="assets/alignment.jpeg"></p>
</div><div class="column" style="width:60%;">
<ol type="1">
<li>Alignment to a reference genome
<ul>
<li>Most align to one location <span class="math inline">\(\implies\)</span> can see where <span style="color:#769b79;">✓</span></li>
<li>Brilliant for gene-level counts <span style="color:#769b79;">✓</span></li>
<li>Not helpful for transcript-level <span style="color:red;">✗</span></li>
</ul></li>
</ol>
<div class="fragment">
<ol start="2" type="1">
<li>Alignment to a reference transcriptome
<ul>
<li>Obtain transcript &amp; gene-level counts <span style="color:#769b79;">✓</span></li>
<li>Many exons shared with multiple transcripts</li>
<li>Reads align to multiple transcripts <span class="math inline">\(\implies\)</span> uncertainty</li>
<li>Can manage this uncertainty in modelling <span style="color:#769b79;">✓</span></li>
<li>Don’t know where each read has aligned <span style="color:red;">✗</span></li>
</ul></li>
</ol>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>Aligning to a linear reference uses conventional aligners (Burrows-Wheeler transform)</li>
<li>Aligning to a transcriptome incorporates graph-based strategies</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="limitations" class="slide level2">
<h2>Limitations</h2>
<ul>
<li>Both approaches use 1 reference sequence <span class="math inline">\(\implies\)</span> linear reference</li>
<li>If 1 of each of our chromosomes matched exactly <span class="math inline">\(\implies\)</span> the other copy won’t</li>
</ul>
<div class="fragment">
<ul>
<li>Current reference genome is GRCh38
<ul>
<li>Transcripts are sequences derived from this</li>
<li>Anchored to GRCh38 using co-ordinates</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Includes 517 extra sequences (scaffolds, patches, haplotypes)
<ul>
<li>Do not exist in reality</li>
<li>Some genes have transcripts defined on scaffolds not chromosomes</li>
</ul></li>
</ul>
<aside class="notes">
<ul>
<li>Reality is that no copies of either chromosome match the existing reference
<ul>
<li>93% of GRCh38 is taken from 11 individuals</li>
</ul></li>
<li>GRC is the Genome Reference Consortium
<ul>
<li>Numbering didn’t start at GRCh1 <span class="math inline">\(\implies\)</span> GRCh37 took over from hg19</li>
<li>hg1 was release in 2000</li>
</ul></li>
<li>Extra sequences are bits they know exist, but don’t know what to do with</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="how-do-we-manage-diversity" class="slide level2">
<h2>How Do We Manage Diversity?</h2>
<p>Given that no real person matches the reference genome</p>
<div class="fragment">
<ul>
<li>Can we improve the reference in general?</li>
<li>Can we improve for a given study cohort</li>
</ul>
</div>
<div class="fragment">
<p>Two emerging approaches</p>
</div>
<div class="fragment">
<ol type="1">
<li>Modify a linear reference using representative variants</li>
<li>Align to a <em>reference graph</em> which represents appropriate diversity</li>
</ol>
</div>
</section>
<section id="modifying-a-reference" class="slide level2">
<h2>Modifying a Reference</h2>
<ul>
<li>Tools exist for modifying a reference genome
<ul>
<li>Unable to analyse at the transcript level <span class="math inline">\(\implies\)</span> only gene-level</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>BODL is developing the software to produce a variant-modified transcriptome
<ul>
<li>Will enable counts at the transcript-level + uncertainty measures</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Current approach uses 1000 Genomes Project variants
<ul>
<li>Shown to significantly improve mappings in other datasets</li>
<li>Poorly representative of Australian Indigenous variation</li>
</ul></li>
</ul>
<aside class="notes">
<ul>
<li>We’re currently discussing how we manage this for the PROPHECY dataset</li>
<li>How do we define improvement</li>
<li>How do we manage data access between -omics layers</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="graph-based-approaches" class="slide level2">
<h2>Graph Based Approaches</h2>
<div class="columns">
<div class="column" style="width:55%;">
<ul>
<li>Telomere-to-telomere (T2T) assemblies contain no scaffolds
<ul>
<li>Just chr1-22, X, Y &amp; MT</li>
</ul></li>
<li>Individual assemblies <span class="math inline">\(\implies\)</span> <em>haplotype resolved</em>
<ul>
<li>Separate out both copies of chr1 etc.</li>
</ul></li>
</ul>
<div class="fragment" data-fragment-index="1">
<ul>
<li>Can construct a reference graph
<ul>
<li>Shared sequences between chromosome pairs are joined</li>
<li>Variation represented as bubbles</li>
</ul></li>
</ul>
</div>
<div class="fragment" data-fragment-index="2">
<ul>
<li>For &gt;1 individual would be a <em>pangenome graph</em></li>
</ul>
</div>
</div><div class="column" style="width:45%;">
<div class="fragment" data-fragment-index="1">
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="assets/reference_graph.png"></p>
<figcaption>Sibbesen, J et al, <em>Haplotype-aware pantranscriptome analyses using spliced pangenome graphs</em> Nat Methods, 2023</figcaption>
</figure>
</div>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>Beyond my expertise of experience</li>
<li>Currently have a student learning the tools</li>
<li>Hardip is actively working in this space with NCIG</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="graph-based-approaches-1" class="slide level2">
<h2>Graph Based Approaches</h2>
<ul>
<li>A reference pan-genome is now available<sup>1</sup>
<ul>
<li>Contains 47 haplotype resolved diploid T2T assemblies</li>
<li>Representative of diversity?</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Explicitly align to the <em>complete set of diversity</em> <span class="fragment">(in the reference)</span></li>
</ul>
</div>
<div class="fragment">
<ul>
<li><span style="color:red;"> Are the days of the linear reference dead? 😱</span></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>How will this impact the PROPHECY transcriptomics layer?
<ul>
<li>e.g.&nbsp;How do we integrate annotated regulatory elements from a linear reference</li>
</ul></li>
</ul>
<aside class="notes">
<ul>
<li>This is an area of active discussion amongst the Black Ochre Data Labs</li>
<li>How will these approaches impact integration across layers?</li>
<li>How will it impact our ability to access existing resources, all based on linear genomes</li>
<li>Monica’s work, using a public, non-Indigenous T2D dataset will help us understand how known genes behave under this new approach</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
<aside><ol class="aside-footnotes"><li id="fn3"><p>Liao W et al, <em>A draft human pangenome reference</em> Nature, 2023</p></li></ol></aside></section>
<section id="long-reads" class="slide level2">
<h2>Long Reads</h2>
<div class="columns">
<div class="column" style="width:60%;">
<ul>
<li>Short read technology has <span style="color:#769b79;">⇩⇩⇩</span> error rates</li>
<li>Long reads have <span style="color:red;">⇧⇧⇧</span> error rates</li>
</ul>
<div class="fragment" data-fragment-index="1">
<ul>
<li>Reads are circularised <br><span class="math inline">\(\implies\)</span> errors corrected by repeat reads</li>
</ul>
</div>
<div class="fragment" data-fragment-index="2">
<ul>
<li>Great for identification of novel transcripts <br><span class="math inline">\(\implies\)</span> difficult to quantify
<ul>
<li>Creation of a custom reference transcriptome</li>
<li>Challenging to refer back to functional annotations in reference</li>
</ul></li>
</ul>
</div>
</div><div class="column" style="width:40%;">
<div class="fragment" data-fragment-index="1">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="assets/m_269fig2.jpeg"></p>
<figcaption>Peccoud J et al, <em>Untangling Heteroplasmy, Structure, and Evolution of an Atypical Mitochondrial Genome by PacBio Sequencing</em> Genetics 2017</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section></section>
<section>
<section id="how-do-we-make-key-discoveries" class="title-slide slide level1 center" data-background-color="#3c3c44">
<h1>How Do We Make Key Discoveries?</h1>

</section>
<section id="differential-gene-expression" class="slide level2">
<h2>Differential Gene Expression</h2>
<p>Once we have counts (gene or transcript-level)</p>
<ul>
<li>Identify any technical issues (GC bias, failed samples etc.)</li>
<li>Fit standard statistical models (GLM)
<ul>
<li>Fairly simple in small Treatment vs Control comparisons</li>
<li>Less straightforward with large, complex designs</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Sequencing generally done in batches of <span class="math inline">\(\leq\)</span> 96 <span class="math inline">\(\implies\)</span> batch effects
<ul>
<li>Is identifiable technical noise which masks true biology</li>
<li>Lead to <span style="color:red;">⇧</span> false discoveries <span style="color:red">⇩</span> true discoveries</li>
<li>Active area of development for large cohort studies (Terry Speed)</li>
</ul></li>
</ul>
<!-- - Batch effects, GC bias etc -->
<aside class="notes">
<ul>
<li>Most analysis uses Negative Binomial models (like Poisson with more flexibility in variance modelling)</li>
<li>For simple comparisons we often assume effect size (change in expression) is only relevant if large enough
<ul>
<li>This is not as easy in complex models</li>
<li>These ar also observational studies <span class="math inline">\(\implies\)</span> we get what we get</li>
<li>We will find correlations between expression patterns and predictor variables <span class="math inline">\(\implies\)</span> doesn’t elucidate mechanism</li>
</ul></li>
<li>96-well plates are used in sample preparation
<ul>
<li>Lots of downstream protocols developed based on this, just like Roman chariot wheels determined train-track width</li>
<li>Key strategies for managing batch effects are
<ul>
<li>using consistent reagents from the same batch</li>
<li>including technical replicates across batches</li>
<li>control RNA-sequences with known concentrations</li>
</ul></li>
</ul></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="differential-gene-expression-1" class="slide level2">
<h2>Differential Gene Expression</h2>
<div class="columns">
<div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/master/docs/figure/dge_analysis.Rmd/plotMA-1.png?raw=true"></p>
<figcaption><em>Typical MA plot</em> (Pederson, unpublished)</figcaption>
</figure>
</div>
</div><div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/master/docs/figure/dge_analysis.Rmd/plotVolcano-1.png?raw=true" style="width:80.0%"></p>
<figcaption><em>Typical Volcano plot</em> (Pederson, unpublished)</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="network-approaches" class="slide level2">
<h2>Network approaches</h2>
<ul>
<li>DGE takes “significantly DE” genes and joins to try &amp; form a story
<ul>
<li>i.e.&nbsp;big changes in a few genes <span class="math inline">\(\implies\)</span> biological consequences</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Network approaches look for larger shifts amongst correlated genes
<ul>
<li>i.e.&nbsp;small changes across an entire pathway <span class="math inline">\(\implies\)</span> biological consequences</li>
</ul></li>
<li>Far more flexibility with parameters <span class="math inline">\(\implies\)</span> <em>reproducibility</em>?
<ul>
<li>Recent research is improving this markedly</li>
</ul></li>
</ul>
</div>
</section>
<section id="wgcna" class="slide level2">
<h2>WGCNA</h2>
<div class="columns">
<div class="column">
<ul>
<li>Multiple approaches with WGCNA the biggest player
<ul>
<li>Form correlation network</li>
<li>Identify modules within correlation network</li>
<li>Compare to predictor variables</li>
<li>Identify underlying biology</li>
</ul></li>
</ul>
</div><div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="assets/13402_2019_438_Fig1_HTML.png" style="width:70.0%"></p>
<figcaption>Image from Huang, J et al, <em>Analysis of functional hub genes identifies CDC45 as an oncogene in non-small cell lung cancer - a short report</em> Cellular Oncology, 2019</figcaption>
</figure>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>WGCNA is Weighted Gene Correlation Network Analysis</li>
<li>The top panel shows a dendrogram based on correlations with modules at the bottom</li>
<li>The bottom panel compares modules to predictor variable</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="supervised-approaches" class="slide level2">
<h2>Supervised Approaches</h2>
<ul>
<li>Principal Component Analysis (PCA) is an un-supervised approach
<ul>
<li>mainly used for QC &amp; visualisation in transcriptomics</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Projection onto Latent Space (PLS) is a <em>supervised</em> approach
<ul>
<li>Maximises correlations with predictor variables</li>
<li>Alternative approach to identifying groups of correlated genes</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>No <span class="math inline">\(p\)</span>-values 😱 <span class="math inline">\(\implies\)</span> How can we ever publish?</li>
</ul>
<aside class="notes">
<ul>
<li>PLS is also known as Partial Least Squares
<ul>
<li>Supported &amp; enabled by mix-omics (Kim-Anh Le Cao)</li>
<li>Excellent support for integration of multi-omics data</li>
</ul></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="eqtl-and-twas" class="slide level2">
<h2>eQTL and TWAS</h2>
<ul>
<li>Expression Quantitative Trait Loci (eQTL)
<ul>
<li>If RNA abundance is a quantitative trait <span class="math inline">\(\implies\)</span> which variants are associated with this?</li>
<li>Is there an association with phenotype (e.g.&nbsp;CVD development)</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Transcriptome-Wide Association Studies (TWAS)
<ul>
<li>Integrates analysis eQTL with GWAS</li>
</ul></li>
</ul>
</div>
</section>
<section id="scrna-seq" class="slide level2">
<h2>scRNA-Seq</h2>
<div class="columns">
<div class="column">
<ul>
<li>Retains the connection between transcript and cell-of-origin</li>
<li>Huge numbers of ‘failure to detect’ expression (i.e.&nbsp;zero counts)</li>
<li>Uses clustering to identify cell types</li>
<li>Pseudo-bulk clusters for DGE analysis</li>
</ul>
</div><div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="https://bioconductor.org/books/3.15/OSCA.basic/reduced-dimensions_files/figure-html/tsne-brain-1.png"></p>
<figcaption>Image from: Amezquita, R et al <em>OSCA: Orchestrating Single Cell analysis with Bioconductor</em> Bioconductor, 2022</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="spatial-transcriptomics" class="slide level2">
<h2>Spatial Transcriptomics</h2>
<ul>
<li>Cells are held in place <span class="math inline">\(\implies\)</span> transcripts identified within a region</li>
<li>The current hot area in transcriptomics <span class="math inline">\(\implies\)</span> Nature Methods “Method of the Year, 2020”</li>
<li>Single-cell resolution is arguably here</li>
</ul>

<img data-src="assets/716c1856c750044a946968defe719a56.jpg" class="r-stretch"><aside class="notes">
<ul>
<li>Neither technique applicable to PROPHECY Datasets <span class="math inline">\(\implies\)</span> don’t really address initial questions</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="pathway-functional-analysis" class="slide level2">
<h2>Pathway &amp; Functional Analysis</h2>
<div class="columns">
<div class="column">
<ul>
<li>Look for biologically relevant signals in DE genes or network modules
<ul>
<li>Enriched pathways</li>
<li>Common transcription factors</li>
<li>Drug target signals</li>
</ul></li>
<li>Compare to public datasets (if appropriate)</li>
</ul>
<div class="fragment">
<ul>
<li>Interpretation is key
<ul>
<li>Researchers love to invent stories &amp; patterns</li>
</ul></li>
</ul>
</div>
</div><div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="https://smped.github.io/GRAVI_PRJNA509779/assets/AR_E2_E2DHT-H3K27ac_E2_E2DHT/AR_Up-H3K27ac_Up_rnaseq_network.png"></p>
<figcaption><em>Enriched pathways combining RNA-Seq with AR and H3K27ac ChIP-seq</em> (Pederson, Unpublished)</figcaption>
</figure>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>Often just fancy versions of Fisher’s Exact Test</li>
<li>Correlations to known treatments or perturbations can be informative</li>
<li>Famous story of a transcriptomic dataset, where the values and gene labels were out by one
<ul>
<li>(Keith Baggerley &amp; Duke University)</li>
<li>Even though the biology was nonsense, researchers still saw patterns &amp; explanations</li>
</ul></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="acknowledgements" class="slide level2 smaller">
<h2>Acknowledgements</h2>
<p><br></p>
<div class="columns">
<div class="column" style="width:30%;">
<h4 id="tki-bodl-nerds">TKI / BODL Nerds</h4>
<ul>
<li>Alex Brown</li>
<li>Jimmy Breen</li>
<li>Sam Buckberry</li>
<li>Liza Kretzschmar</li>
<li>Yassine Souilmi</li>
<li>Bastien Llamas</li>
<li>Holly Martin</li>
<li>Claudia Floreani</li>
<li>Natasha Howard</li>
<li>Anelle du Preez</li>
<li>Kaashifah Bruce</li>
<li>Amanda Satour-Richards</li>
<li>Justine Clarke</li>
<li>Katharine Brown</li>
</ul>
</div><div class="column" style="width:30%;">
<h4 id="ncig">NCIG</h4>
<ul>
<li>Hardip Patel</li>
</ul>
<h4 id="align">ALIGN</h4>
<ul>
<li>Johanna Barclay</li>
<li>Annalee Stearne</li>
<li>Louise Lyons</li>
</ul>
<h4 id="students">Students</h4>
<ul>
<li>Nhi Hin</li>
<li>Jacqueline Rehn</li>
<li>Nora Liu</li>
<li>Lachlan Baer</li>
<li>Megan Monaghan</li>
<li>Monica Guilhaus</li>
</ul>
</div><div class="column" style="width:30%;">
<h4 id="bioinformatics-hub">Bioinformatics Hub</h4>
<ul>
<li>David Adelson</li>
<li>Gary Glonek</li>
<li>Dan Kortschak</li>
<li>Nathan Watson-Haigh</li>
<li>Rick Tearle</li>
<li>Hien To</li>
</ul>
</div>
</div>

<img src="assets/bodl_logo_white_background.jpg" class="slide-logo r-stretch"><div class="footer footer-default">

</div>
</section></section>

    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="20231107_ALIGN_Transcriptomics_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="20231107_ALIGN_Transcriptomics_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1280,

        height: 720,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>